{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Скачивание файлов близкородственной бактерии"
      ],
      "metadata": {
        "id": "9l4mfp8udffD"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bGAKTNXRdDCt",
        "outputId": "9961db4b-4056-48de-fbfe-295e579e9ffd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Entrez Direct has been successfully downloaded and installed.\n",
            "\n",
            "In order to complete the configuration process, please execute the following:\n",
            "\n",
            "  echo \"export PATH=\\${PATH}:/root/edirect\" >> ${HOME}/.bashrc\n",
            "\n",
            "or manually edit the PATH variable assignment in your .bashrc file.\n",
            "\n",
            "Would you like to do that automatically now? [y/N]\n",
            "y\n",
            "OK, done.\n",
            "\n",
            "To activate EDirect for this terminal session, please execute the following:\n",
            "\n",
            "export PATH=${PATH}:${HOME}/edirect\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!sh -c \"$(curl -fsSL ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh)\""
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format gb  >  T_oleivorans_MIL_1.gbk"
      ],
      "metadata": {
        "id": "tdiKn0-mduZ-"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Установка програм "
      ],
      "metadata": {
        "id": "GxMG_d0Pd3lc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install biopython"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MhDe8g4Md1xI",
        "outputId": "1feac314-0c83-45d6-bdfd-b2142ae80c10"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting biopython\n",
            "  Downloading biopython-1.79-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.6 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.6 MB 4.4 MB/s \n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from biopython) (1.21.6)\n",
            "Installing collected packages: biopython\n",
            "Successfully installed biopython-1.79\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get update"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0pMmDNhreA33",
        "outputId": "cb3d776e-e41c-4786-8e98-cbbea017e57d"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Get:1 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]\n",
            "Ign:2 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n",
            "Get:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease [1,581 B]\n",
            "Hit:4 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n",
            "Get:5 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n",
            "Get:6 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease [15.9 kB]\n",
            "Get:7 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ Packages [95.2 kB]\n",
            "Hit:8 http://archive.ubuntu.com/ubuntu bionic InRelease\n",
            "Get:9 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages [983 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n",
            "Hit:12 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [83.3 kB]\n",
            "Hit:14 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease\n",
            "Get:15 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1,554 kB]\n",
            "Hit:16 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [3,467 kB]\n",
            "Get:18 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main Sources [2,165 kB]\n",
            "Get:19 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [1,208 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [1,248 kB]\n",
            "Get:21 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [3,035 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,332 kB]\n",
            "Get:23 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main amd64 Packages [1,109 kB]\n",
            "Fetched 17.5 MB in 6s (2,703 kB/s)\n",
            "Reading package lists... Done\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install ncbi-blast+"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XPD8s3DbeC5B",
        "outputId": "39fa4138-65f6-45e4-c1a9-aa6f29f06158"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  ncbi-data\n",
            "The following NEW packages will be installed:\n",
            "  ncbi-blast+ ncbi-data\n",
            "0 upgraded, 2 newly installed, 0 to remove and 38 not upgraded.\n",
            "Need to get 13.1 MB of archives.\n",
            "After this operation, 66.7 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 ncbi-data all 6.1.20170106-2 [3,645 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/universe amd64 ncbi-blast+ amd64 2.6.0-1 [9,446 kB]\n",
            "Fetched 13.1 MB in 3s (5,118 kB/s)\n",
            "Selecting previously unselected package ncbi-data.\n",
            "(Reading database ... 123941 files and directories currently installed.)\n",
            "Preparing to unpack .../ncbi-data_6.1.20170106-2_all.deb ...\n",
            "Unpacking ncbi-data (6.1.20170106-2) ...\n",
            "Selecting previously unselected package ncbi-blast+.\n",
            "Preparing to unpack .../ncbi-blast+_2.6.0-1_amd64.deb ...\n",
            "Unpacking ncbi-blast+ (2.6.0-1) ...\n",
            "Setting up ncbi-data (6.1.20170106-2) ...\n",
            "Setting up ncbi-blast+ (2.6.0-1) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!blastn -help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EbJLs3ZteTv2",
        "outputId": "7a47e0b6-7704-4aa7-92cc-2666e8547201"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "USAGE\n",
            "  blastn [-h] [-help] [-import_search_strategy filename]\n",
            "    [-export_search_strategy filename] [-task task_name] [-db database_name]\n",
            "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\n",
            "    [-negative_gilist filename] [-entrez_query entrez_query]\n",
            "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\n",
            "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\n",
            "    [-out output_file] [-evalue evalue] [-word_size int_value]\n",
            "    [-gapopen open_penalty] [-gapextend extend_penalty]\n",
            "    [-perc_identity float_value] [-qcov_hsp_perc float_value]\n",
            "    [-max_hsps int_value] [-xdrop_ungap float_value] [-xdrop_gap float_value]\n",
            "    [-xdrop_gap_final float_value] [-searchsp int_value]\n",
            "    [-sum_stats bool_value] [-penalty penalty] [-reward reward] [-no_greedy]\n",
            "    [-min_raw_gapped_score int_value] [-template_type type]\n",
            "    [-template_length int_value] [-dust DUST_options]\n",
            "    [-filtering_db filtering_database]\n",
            "    [-window_masker_taxid window_masker_taxid]\n",
            "    [-window_masker_db window_masker_db] [-soft_masking soft_masking]\n",
            "    [-ungapped] [-culling_limit int_value] [-best_hit_overhang float_value]\n",
            "    [-best_hit_score_edge float_value] [-window_size int_value]\n",
            "    [-off_diagonal_range int_value] [-use_index boolean] [-index_name string]\n",
            "    [-lcase_masking] [-query_loc range] [-strand strand] [-parse_deflines]\n",
            "    [-outfmt format] [-show_gis] [-num_descriptions int_value]\n",
            "    [-num_alignments int_value] [-line_length line_length] [-html]\n",
            "    [-max_target_seqs num_sequences] [-num_threads int_value] [-remote]\n",
            "    [-version]\n",
            "\n",
            "DESCRIPTION\n",
            "   Nucleotide-Nucleotide BLAST 2.6.0+\n",
            "\n",
            "OPTIONAL ARGUMENTS\n",
            " -h\n",
            "   Print USAGE and DESCRIPTION;  ignore all other parameters\n",
            " -help\n",
            "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\n",
            " -version\n",
            "   Print version number;  ignore other arguments\n",
            "\n",
            " *** Input query options\n",
            " -query <File_In>\n",
            "   Input file name\n",
            "   Default = `-'\n",
            " -query_loc <String>\n",
            "   Location on the query sequence in 1-based offsets (Format: start-stop)\n",
            " -strand <String, `both', `minus', `plus'>\n",
            "   Query strand(s) to search against database/subject\n",
            "   Default = `both'\n",
            "\n",
            " *** General search options\n",
            " -task <String, Permissible values: 'blastn' 'blastn-short' 'dc-megablast'\n",
            "                'megablast' 'rmblastn' >\n",
            "   Task to execute\n",
            "   Default = `megablast'\n",
            " -db <String>\n",
            "   BLAST database name\n",
            "    * Incompatible with:  subject, subject_loc\n",
            " -out <File_Out>\n",
            "   Output file name\n",
            "   Default = `-'\n",
            " -evalue <Real>\n",
            "   Expectation value (E) threshold for saving hits \n",
            "   Default = `10'\n",
            " -word_size <Integer, >=4>\n",
            "   Word size for wordfinder algorithm (length of best perfect match)\n",
            " -gapopen <Integer>\n",
            "   Cost to open a gap\n",
            " -gapextend <Integer>\n",
            "   Cost to extend a gap\n",
            " -penalty <Integer, <=0>\n",
            "   Penalty for a nucleotide mismatch\n",
            " -reward <Integer, >=0>\n",
            "   Reward for a nucleotide match\n",
            " -use_index <Boolean>\n",
            "   Use MegaBLAST database index\n",
            "   Default = `false'\n",
            " -index_name <String>\n",
            "   MegaBLAST database index name (deprecated; use only for old style indices)\n",
            "\n",
            " *** BLAST-2-Sequences options\n",
            " -subject <File_In>\n",
            "   Subject sequence(s) to search\n",
            "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\n",
            "   db_soft_mask, db_hard_mask\n",
            " -subject_loc <String>\n",
            "   Location on the subject sequence in 1-based offsets (Format: start-stop)\n",
            "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\n",
            "   db_soft_mask, db_hard_mask, remote\n",
            "\n",
            " *** Formatting options\n",
            " -outfmt <String>\n",
            "   alignment view options:\n",
            "     0 = Pairwise,\n",
            "     1 = Query-anchored showing identities,\n",
            "     2 = Query-anchored no identities,\n",
            "     3 = Flat query-anchored showing identities,\n",
            "     4 = Flat query-anchored no identities,\n",
            "     5 = BLAST XML,\n",
            "     6 = Tabular,\n",
            "     7 = Tabular with comment lines,\n",
            "     8 = Seqalign (Text ASN.1),\n",
            "     9 = Seqalign (Binary ASN.1),\n",
            "    10 = Comma-separated values,\n",
            "    11 = BLAST archive (ASN.1),\n",
            "    12 = Seqalign (JSON),\n",
            "    13 = Multiple-file BLAST JSON,\n",
            "    14 = Multiple-file BLAST XML2,\n",
            "    15 = Single-file BLAST JSON,\n",
            "    16 = Single-file BLAST XML2,\n",
            "    17 = Sequence Alignment/Map (SAM),\n",
            "    18 = Organism Report\n",
            "   \n",
            "   Options 6, 7, 10 and 17 can be additionally configured to produce\n",
            "   a custom format specified by space delimited format specifiers.\n",
            "   The supported format specifiers for options 6, 7 and 10 are:\n",
            "   \t    qseqid means Query Seq-id\n",
            "   \t       qgi means Query GI\n",
            "   \t      qacc means Query accesion\n",
            "   \t   qaccver means Query accesion.version\n",
            "   \t      qlen means Query sequence length\n",
            "   \t    sseqid means Subject Seq-id\n",
            "   \t sallseqid means All subject Seq-id(s), separated by a ';'\n",
            "   \t       sgi means Subject GI\n",
            "   \t    sallgi means All subject GIs\n",
            "   \t      sacc means Subject accession\n",
            "   \t   saccver means Subject accession.version\n",
            "   \t   sallacc means All subject accessions\n",
            "   \t      slen means Subject sequence length\n",
            "   \t    qstart means Start of alignment in query\n",
            "   \t      qend means End of alignment in query\n",
            "   \t    sstart means Start of alignment in subject\n",
            "   \t      send means End of alignment in subject\n",
            "   \t      qseq means Aligned part of query sequence\n",
            "   \t      sseq means Aligned part of subject sequence\n",
            "   \t    evalue means Expect value\n",
            "   \t  bitscore means Bit score\n",
            "   \t     score means Raw score\n",
            "   \t    length means Alignment length\n",
            "   \t    pident means Percentage of identical matches\n",
            "   \t    nident means Number of identical matches\n",
            "   \t  mismatch means Number of mismatches\n",
            "   \t  positive means Number of positive-scoring matches\n",
            "   \t   gapopen means Number of gap openings\n",
            "   \t      gaps means Total number of gaps\n",
            "   \t      ppos means Percentage of positive-scoring matches\n",
            "   \t    frames means Query and subject frames separated by a '/'\n",
            "   \t    qframe means Query frame\n",
            "   \t    sframe means Subject frame\n",
            "   \t      btop means Blast traceback operations (BTOP)\n",
            "   \t    staxid means Subject Taxonomy ID\n",
            "   \t  ssciname means Subject Scientific Name\n",
            "   \t  scomname means Subject Common Name\n",
            "   \tsblastname means Subject Blast Name\n",
            "   \t sskingdom means Subject Super Kingdom\n",
            "   \t   staxids means unique Subject Taxonomy ID(s), separated by a ';'\n",
            "   \t\t\t (in numerical order)\n",
            "   \t sscinames means unique Subject Scientific Name(s), separated by a ';'\n",
            "   \t scomnames means unique Subject Common Name(s), separated by a ';'\n",
            "   \tsblastnames means unique Subject Blast Name(s), separated by a ';'\n",
            "   \t\t\t (in alphabetical order)\n",
            "   \tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\n",
            "   \t\t\t (in alphabetical order) \n",
            "   \t    stitle means Subject Title\n",
            "   \tsalltitles means All Subject Title(s), separated by a '<>'\n",
            "   \t   sstrand means Subject Strand\n",
            "   \t     qcovs means Query Coverage Per Subject\n",
            "   \t   qcovhsp means Query Coverage Per HSP\n",
            "   \t    qcovus means Query Coverage Per Unique Subject (blastn only)\n",
            "   When not provided, the default value is:\n",
            "   'qaccver saccver pident length mismatch gapopen qstart qend sstart send\n",
            "   evalue bitscore', which is equivalent to the keyword 'std'\n",
            "   The supported format specifier for option 17 is:\n",
            "   \t        SQ means Include Sequence Data\n",
            "   \t        SR means Subject as Reference Seq\n",
            "   Default = `0'\n",
            " -show_gis\n",
            "   Show NCBI GIs in deflines?\n",
            " -num_descriptions <Integer, >=0>\n",
            "   Number of database sequences to show one-line descriptions for\n",
            "   Not applicable for outfmt > 4\n",
            "   Default = `500'\n",
            "    * Incompatible with:  max_target_seqs\n",
            " -num_alignments <Integer, >=0>\n",
            "   Number of database sequences to show alignments for\n",
            "   Default = `250'\n",
            "    * Incompatible with:  max_target_seqs\n",
            " -line_length <Integer, >=1>\n",
            "   Line length for formatting alignments\n",
            "   Not applicable for outfmt > 4\n",
            "   Default = `60'\n",
            " -html\n",
            "   Produce HTML output?\n",
            "\n",
            " *** Query filtering options\n",
            " -dust <String>\n",
            "   Filter query sequence with DUST (Format: 'yes', 'level window linker', or\n",
            "   'no' to disable)\n",
            "   Default = `20 64 1'\n",
            " -filtering_db <String>\n",
            "   BLAST database containing filtering elements (i.e.: repeats)\n",
            " -window_masker_taxid <Integer>\n",
            "   Enable WindowMasker filtering using a Taxonomic ID\n",
            " -window_masker_db <String>\n",
            "   Enable WindowMasker filtering using this repeats database.\n",
            " -soft_masking <Boolean>\n",
            "   Apply filtering locations as soft masks\n",
            "   Default = `true'\n",
            " -lcase_masking\n",
            "   Use lower case filtering in query and subject sequence(s)?\n",
            "\n",
            " *** Restrict search or results\n",
            " -gilist <String>\n",
            "   Restrict search of database to list of GI's\n",
            "    * Incompatible with:  negative_gilist, seqidlist, remote, subject,\n",
            "   subject_loc\n",
            " -seqidlist <String>\n",
            "   Restrict search of database to list of SeqId's\n",
            "    * Incompatible with:  gilist, negative_gilist, remote, subject,\n",
            "   subject_loc\n",
            " -negative_gilist <String>\n",
            "   Restrict search of database to everything except the listed GIs\n",
            "    * Incompatible with:  gilist, seqidlist, remote, subject, subject_loc\n",
            " -entrez_query <String>\n",
            "   Restrict search with the given Entrez query\n",
            "    * Requires:  remote\n",
            " -db_soft_mask <String>\n",
            "   Filtering algorithm ID to apply to the BLAST database as soft masking\n",
            "    * Incompatible with:  db_hard_mask, subject, subject_loc\n",
            " -db_hard_mask <String>\n",
            "   Filtering algorithm ID to apply to the BLAST database as hard masking\n",
            "    * Incompatible with:  db_soft_mask, subject, subject_loc\n",
            " -perc_identity <Real, 0..100>\n",
            "   Percent identity\n",
            " -qcov_hsp_perc <Real, 0..100>\n",
            "   Percent query coverage per hsp\n",
            " -max_hsps <Integer, >=1>\n",
            "   Set maximum number of HSPs per subject sequence to save for each query\n",
            " -culling_limit <Integer, >=0>\n",
            "   If the query range of a hit is enveloped by that of at least this many\n",
            "   higher-scoring hits, delete the hit\n",
            "    * Incompatible with:  best_hit_overhang, best_hit_score_edge\n",
            " -best_hit_overhang <Real, (>0 and <0.5)>\n",
            "   Best Hit algorithm overhang value (recommended value: 0.1)\n",
            "    * Incompatible with:  culling_limit\n",
            " -best_hit_score_edge <Real, (>0 and <0.5)>\n",
            "   Best Hit algorithm score edge value (recommended value: 0.1)\n",
            "    * Incompatible with:  culling_limit\n",
            " -max_target_seqs <Integer, >=1>\n",
            "   Maximum number of aligned sequences to keep \n",
            "   Not applicable for outfmt <= 4\n",
            "   Default = `500'\n",
            "    * Incompatible with:  num_descriptions, num_alignments\n",
            "\n",
            " *** Discontiguous MegaBLAST options\n",
            " -template_type <String, `coding', `coding_and_optimal', `optimal'>\n",
            "   Discontiguous MegaBLAST template type\n",
            "    * Requires:  template_length\n",
            " -template_length <Integer, Permissible values: '16' '18' '21' >\n",
            "   Discontiguous MegaBLAST template length\n",
            "    * Requires:  template_type\n",
            "\n",
            " *** Statistical options\n",
            " -dbsize <Int8>\n",
            "   Effective length of the database \n",
            " -searchsp <Int8, >=0>\n",
            "   Effective length of the search space\n",
            " -sum_stats <Boolean>\n",
            "   Use sum statistics\n",
            "\n",
            " *** Search strategy options\n",
            " -import_search_strategy <File_In>\n",
            "   Search strategy to use\n",
            "    * Incompatible with:  export_search_strategy\n",
            " -export_search_strategy <File_Out>\n",
            "   File name to record the search strategy used\n",
            "    * Incompatible with:  import_search_strategy\n",
            "\n",
            " *** Extension options\n",
            " -xdrop_ungap <Real>\n",
            "   X-dropoff value (in bits) for ungapped extensions\n",
            " -xdrop_gap <Real>\n",
            "   X-dropoff value (in bits) for preliminary gapped extensions\n",
            " -xdrop_gap_final <Real>\n",
            "   X-dropoff value (in bits) for final gapped alignment\n",
            " -no_greedy\n",
            "   Use non-greedy dynamic programming extension\n",
            " -min_raw_gapped_score <Integer>\n",
            "   Minimum raw gapped score to keep an alignment in the preliminary gapped and\n",
            "   traceback stages\n",
            " -ungapped\n",
            "   Perform ungapped alignment only?\n",
            " -window_size <Integer, >=0>\n",
            "   Multiple hits window size, use 0 to specify 1-hit algorithm\n",
            " -off_diagonal_range <Integer, >=0>\n",
            "   Number of off-diagonals to search for the 2nd hit, use 0 to turn off\n",
            "   Default = `0'\n",
            "\n",
            " *** Miscellaneous options\n",
            " -parse_deflines\n",
            "   Should the query and subject defline(s) be parsed?\n",
            " -num_threads <Integer, >=1>\n",
            "   Number of threads (CPUs) to use in the BLAST search\n",
            "   Default = `1'\n",
            "    * Incompatible with:  remote\n",
            " -remote\n",
            "   Execute search remotely?\n",
            "    * Incompatible with:  gilist, seqidlist, negative_gilist, subject_loc,\n",
            "   num_threads\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Прочитать подробнее можно тут https://www.ncbi.nlm.nih.gov/books/NBK279690/ "
      ],
      "metadata": {
        "id": "fJL5gxFlhnw7"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Предсказание рибосомальной рнк"
      ],
      "metadata": {
        "id": "w5_RP4Yafn_i"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "from Bio import SeqIO\n",
        "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
        "import re\n",
        "import pandas as pd"
      ],
      "metadata": {
        "id": "_JQjcWvNZSIQ"
      },
      "execution_count": 7,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Находим координаты начала, конца, стренд рРНК у близкородственной бактерии\n",
        "coords = []\n",
        "for feature in SeqIO.read(\"T_oleivorans_MIL_1.gbk\", \"genbank\").features:\n",
        "  if feature.type == 'rRNA':\n",
        "    coords.append([int(feature.location.start), int(feature.location.end), feature.strand])"
      ],
      "metadata": {
        "id": "tUcLCBYMbZ0B"
      },
      "execution_count": 8,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Находим последовательности рРНК, которые находятся по полученным координатам\n",
        "sequences = SeqIO.read(\"T_oleivorans_MIL_1.gbk\", \"genbank\").seq\n",
        "sequencesRNA = []\n",
        "for coord in coords:\n",
        "  if coord[2] == 1:\n",
        "    sequencesRNA.append([coord[0],coord[1],str(sequences[coord[0]:coord[1]]),'plus'])\n",
        "  else:\n",
        "    sequencesRNA.append([coord[0],coord[1],str(sequences[coord[0]:coord[1]]),'minus'])"
      ],
      "metadata": {
        "id": "ESGZwyVYbenI"
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Записываем полученные последовательности в fasta файл\n",
        "with open('rRNA.fasta','w') as file:\n",
        "  for sequence in sequencesRNA:\n",
        "    file.write(f'>rRNA_{sequence[0]}...{sequence[1]}\\n')\n",
        "    file.write(sequence[2] + '\\n')"
      ],
      "metadata": {
        "id": "n9xzcq-1belw"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#Скачиваем файл скаффолдов\n",
        "!wget https://raw.githubusercontent.com/miamib34ch/hse22_hw2/main/data/scaffolds.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eYUt5oEYb1uJ",
        "outputId": "da610964-58c9-48ec-c531-e4f3f08c2959"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 20:17:21--  https://raw.githubusercontent.com/miamib34ch/hse22_hw2/main/data/scaffolds.fa\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3965905 (3.8M) [text/plain]\n",
            "Saving to: ‘scaffolds.fa’\n",
            "\n",
            "scaffolds.fa        100%[===================>]   3.78M  --.-KB/s    in 0.09s   \n",
            "\n",
            "2022-10-19 20:17:23 (43.8 MB/s) - ‘scaffolds.fa’ saved [3965905/3965905]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Делаем выравнивание \n",
        "!blastn -query rRNA.fasta -subject scaffolds.fa -outfmt '10 qseqid sseqid pident sstrand evalue' > allignment_stats.txt\n",
        "!blastn -query rRNA.fasta -subject scaffolds.fa -outfmt 7"
      ],
      "metadata": {
        "id": "fng793_WepS2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "dedc7d3d-cbe7-42d2-f8c9-cbf83029b701"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_341494...343033\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_341494...343033\tscaffold1_cov231\t100.000\t1539\t0\t0\t1\t1539\t2527358\t2528896\t0.0\t2843\n",
            "rRNA_341494...343033\tscaffold1_cov231\t99.935\t1539\t1\t0\t1\t1539\t846845\t845307\t0.0\t2837\n",
            "rRNA_341494...343033\tscaffold1_cov231\t100.000\t64\t0\t0\t1\t64\t1673342\t1673405\t7.22e-27\t119\n",
            "rRNA_341494...343033\tscaffold32_cov590\t100.000\t1399\t0\t0\t134\t1532\t1399\t1\t0.0\t2584\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_343487...346374\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 5 hits found\n",
            "rRNA_343487...346374\tscaffold1_cov231\t100.000\t2887\t0\t0\t1\t2887\t2529147\t2532033\t0.0\t5332\n",
            "rRNA_343487...346374\tscaffold1_cov231\t99.965\t2887\t1\t0\t1\t2887\t845056\t842170\t0.0\t5326\n",
            "rRNA_343487...346374\tscaffold35_cov590\t100.000\t1535\t0\t0\t1353\t2887\t1815\t281\t0.0\t2835\n",
            "rRNA_343487...346374\tscaffold41_cov598\t100.000\t1051\t0\t0\t215\t1265\t1051\t1\t0.0\t1941\n",
            "rRNA_343487...346374\tscaffold49_cov624\t100.000\t127\t0\t0\t1\t127\t127\t1\t1.30e-61\t235\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_346568...346684\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_346568...346684\tscaffold1_cov231\t100.000\t116\t0\t0\t1\t116\t841988\t841873\t5.67e-57\t215\n",
            "rRNA_346568...346684\tscaffold1_cov231\t100.000\t116\t0\t0\t1\t116\t2532215\t2532330\t5.67e-57\t215\n",
            "rRNA_346568...346684\tscaffold1_cov231\t100.000\t60\t0\t0\t57\t116\t1673623\t1673682\t7.65e-26\t111\n",
            "rRNA_346568...346684\tscaffold35_cov590\t100.000\t99\t0\t0\t1\t99\t99\t1\t1.60e-47\t183\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_2580484...2580600\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_2580484...2580600\tscaffold1_cov231\t98.276\t116\t2\t0\t1\t116\t841873\t841988\t1.23e-53\t204\n",
            "rRNA_2580484...2580600\tscaffold1_cov231\t98.276\t116\t2\t0\t1\t116\t2532330\t2532215\t1.23e-53\t204\n",
            "rRNA_2580484...2580600\tscaffold1_cov231\t96.667\t60\t2\t0\t1\t60\t1673682\t1673623\t1.66e-22\t100\n",
            "rRNA_2580484...2580600\tscaffold35_cov590\t98.990\t99\t1\t0\t18\t116\t1\t99\t7.44e-46\t178\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_2580781...2583668\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 5 hits found\n",
            "rRNA_2580781...2583668\tscaffold1_cov231\t100.000\t2887\t0\t0\t1\t2887\t2532033\t2529147\t0.0\t5332\n",
            "rRNA_2580781...2583668\tscaffold1_cov231\t99.965\t2887\t1\t0\t1\t2887\t842170\t845056\t0.0\t5326\n",
            "rRNA_2580781...2583668\tscaffold35_cov590\t100.000\t1535\t0\t0\t1\t1535\t281\t1815\t0.0\t2835\n",
            "rRNA_2580781...2583668\tscaffold41_cov598\t100.000\t1051\t0\t0\t1623\t2673\t1\t1051\t0.0\t1941\n",
            "rRNA_2580781...2583668\tscaffold49_cov624\t100.000\t127\t0\t0\t2761\t2887\t1\t127\t1.30e-61\t235\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_2583918...2585457\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_2583918...2585457\tscaffold1_cov231\t99.935\t1539\t1\t0\t1\t1539\t845307\t846845\t0.0\t2837\n",
            "rRNA_2583918...2585457\tscaffold1_cov231\t99.870\t1539\t2\t0\t1\t1539\t2528896\t2527358\t0.0\t2832\n",
            "rRNA_2583918...2585457\tscaffold1_cov231\t100.000\t64\t0\t0\t1476\t1539\t1673405\t1673342\t7.22e-27\t119\n",
            "rRNA_2583918...2585457\tscaffold32_cov590\t99.929\t1399\t1\t0\t8\t1406\t1\t1399\t0.0\t2579\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_3418645...3418761\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_3418645...3418761\tscaffold1_cov231\t100.000\t116\t0\t0\t1\t116\t841873\t841988\t5.67e-57\t215\n",
            "rRNA_3418645...3418761\tscaffold1_cov231\t100.000\t116\t0\t0\t1\t116\t2532330\t2532215\t5.67e-57\t215\n",
            "rRNA_3418645...3418761\tscaffold1_cov231\t100.000\t60\t0\t0\t1\t60\t1673682\t1673623\t7.65e-26\t111\n",
            "rRNA_3418645...3418761\tscaffold35_cov590\t100.000\t99\t0\t0\t18\t116\t1\t99\t1.60e-47\t183\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_3418942...3421829\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 5 hits found\n",
            "rRNA_3418942...3421829\tscaffold1_cov231\t100.000\t2887\t0\t0\t1\t2887\t2532033\t2529147\t0.0\t5332\n",
            "rRNA_3418942...3421829\tscaffold1_cov231\t99.965\t2887\t1\t0\t1\t2887\t842170\t845056\t0.0\t5326\n",
            "rRNA_3418942...3421829\tscaffold35_cov590\t100.000\t1535\t0\t0\t1\t1535\t281\t1815\t0.0\t2835\n",
            "rRNA_3418942...3421829\tscaffold41_cov598\t100.000\t1051\t0\t0\t1623\t2673\t1\t1051\t0.0\t1941\n",
            "rRNA_3418942...3421829\tscaffold49_cov624\t100.000\t127\t0\t0\t2761\t2887\t1\t127\t1.30e-61\t235\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_3422079...3423618\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_3422079...3423618\tscaffold1_cov231\t99.935\t1539\t1\t0\t1\t1539\t845307\t846845\t0.0\t2837\n",
            "rRNA_3422079...3423618\tscaffold1_cov231\t99.870\t1539\t2\t0\t1\t1539\t2528896\t2527358\t0.0\t2832\n",
            "rRNA_3422079...3423618\tscaffold1_cov231\t100.000\t64\t0\t0\t1476\t1539\t1673405\t1673342\t7.22e-27\t119\n",
            "rRNA_3422079...3423618\tscaffold32_cov590\t99.929\t1399\t1\t0\t8\t1406\t1\t1399\t0.0\t2579\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_3423993...3424109\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_3423993...3424109\tscaffold1_cov231\t100.000\t116\t0\t0\t1\t116\t841873\t841988\t5.67e-57\t215\n",
            "rRNA_3423993...3424109\tscaffold1_cov231\t100.000\t116\t0\t0\t1\t116\t2532330\t2532215\t5.67e-57\t215\n",
            "rRNA_3423993...3424109\tscaffold1_cov231\t100.000\t60\t0\t0\t1\t60\t1673682\t1673623\t7.65e-26\t111\n",
            "rRNA_3423993...3424109\tscaffold35_cov590\t100.000\t99\t0\t0\t18\t116\t1\t99\t1.60e-47\t183\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_3424290...3427177\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 5 hits found\n",
            "rRNA_3424290...3427177\tscaffold1_cov231\t100.000\t2887\t0\t0\t1\t2887\t2532033\t2529147\t0.0\t5332\n",
            "rRNA_3424290...3427177\tscaffold1_cov231\t99.965\t2887\t1\t0\t1\t2887\t842170\t845056\t0.0\t5326\n",
            "rRNA_3424290...3427177\tscaffold35_cov590\t100.000\t1535\t0\t0\t1\t1535\t281\t1815\t0.0\t2835\n",
            "rRNA_3424290...3427177\tscaffold41_cov598\t100.000\t1051\t0\t0\t1623\t2673\t1\t1051\t0.0\t1941\n",
            "rRNA_3424290...3427177\tscaffold49_cov624\t100.000\t127\t0\t0\t2761\t2887\t1\t127\t1.30e-61\t235\n",
            "# BLASTN 2.6.0+\n",
            "# Query: rRNA_3427427...3428966\n",
            "# Database: User specified sequence set (Input: scaffolds.fa)\n",
            "# Fields: query acc.ver, subject acc.ver, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score\n",
            "# 4 hits found\n",
            "rRNA_3427427...3428966\tscaffold1_cov231\t99.935\t1539\t1\t0\t1\t1539\t845307\t846845\t0.0\t2837\n",
            "rRNA_3427427...3428966\tscaffold1_cov231\t99.870\t1539\t2\t0\t1\t1539\t2528896\t2527358\t0.0\t2832\n",
            "rRNA_3427427...3428966\tscaffold1_cov231\t100.000\t64\t0\t0\t1476\t1539\t1673405\t1673342\t7.22e-27\t119\n",
            "rRNA_3427427...3428966\tscaffold32_cov590\t99.929\t1399\t1\t0\t8\t1406\t1\t1399\t0.0\t2579\n",
            "# BLAST processed 12 queries\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Возьмём с самым низким evalue из каждого промежутка координат\n",
        "names = ['qseqid', 'sseqid', '% identity', 'sstrand', 'evalue']\n",
        "stats = pd.read_csv('allignment_stats.txt', sep=',', header=None, names=names)\n",
        "stats = stats.sort_values('evalue', ascending=True).drop_duplicates('qseqid')\n",
        "stats"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 414
        },
        "id": "6FcyiOVqo0uI",
        "outputId": "eb0ebe9c-717d-49ab-d744-80626cfeda44"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                    qseqid             sseqid  % identity sstrand  \\\n",
              "0     rRNA_341494...343033   scaffold1_cov231     100.000    plus   \n",
              "22  rRNA_2583918...2585457   scaffold1_cov231      99.935    plus   \n",
              "30  rRNA_3418942...3421829   scaffold1_cov231     100.000   minus   \n",
              "35  rRNA_3422079...3423618   scaffold1_cov231      99.935    plus   \n",
              "43  rRNA_3424290...3427177   scaffold1_cov231     100.000   minus   \n",
              "48  rRNA_3427427...3428966   scaffold1_cov231      99.935    plus   \n",
              "20  rRNA_2580781...2583668  scaffold41_cov598     100.000    plus   \n",
              "4     rRNA_343487...346374   scaffold1_cov231     100.000    plus   \n",
              "39  rRNA_3423993...3424109   scaffold1_cov231     100.000    plus   \n",
              "27  rRNA_3418645...3418761   scaffold1_cov231     100.000   minus   \n",
              "9     rRNA_346568...346684   scaffold1_cov231     100.000   minus   \n",
              "14  rRNA_2580484...2580600   scaffold1_cov231      98.276   minus   \n",
              "\n",
              "          evalue  \n",
              "0   0.000000e+00  \n",
              "22  0.000000e+00  \n",
              "30  0.000000e+00  \n",
              "35  0.000000e+00  \n",
              "43  0.000000e+00  \n",
              "48  0.000000e+00  \n",
              "20  0.000000e+00  \n",
              "4   0.000000e+00  \n",
              "39  5.670000e-57  \n",
              "27  5.670000e-57  \n",
              "9   5.670000e-57  \n",
              "14  1.230000e-53  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-341e7be1-ac37-48f7-94c0-c39ec278a045\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>qseqid</th>\n",
              "      <th>sseqid</th>\n",
              "      <th>% identity</th>\n",
              "      <th>sstrand</th>\n",
              "      <th>evalue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>rRNA_341494...343033</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>rRNA_2583918...2585457</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>99.935</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>rRNA_3418942...3421829</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>rRNA_3422079...3423618</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>99.935</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>rRNA_3424290...3427177</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>rRNA_3427427...3428966</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>99.935</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>rRNA_2580781...2583668</td>\n",
              "      <td>scaffold41_cov598</td>\n",
              "      <td>100.000</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>rRNA_343487...346374</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>rRNA_3423993...3424109</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>plus</td>\n",
              "      <td>5.670000e-57</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>rRNA_3418645...3418761</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>5.670000e-57</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>rRNA_346568...346684</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>5.670000e-57</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>rRNA_2580484...2580600</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>98.276</td>\n",
              "      <td>minus</td>\n",
              "      <td>1.230000e-53</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-341e7be1-ac37-48f7-94c0-c39ec278a045')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-341e7be1-ac37-48f7-94c0-c39ec278a045 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-341e7be1-ac37-48f7-94c0-c39ec278a045');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Оставим только строки, со стрендами, которые совпадают со стрендами близкородственной бактерии\n",
        "for sequence in sequencesRNA:\n",
        "  for line in stats.iterrows():\n",
        "    if line[1][0] == f'rRNA_{sequence[0]}...{sequence[1]}' and sequence[3] != line[1][3]:\n",
        "      stats = stats.drop(line[0])\n",
        "stats"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 231
        },
        "id": "UCApvlBSsDFM",
        "outputId": "c2463d7d-4608-47a7-dcad-26b8d0ed11b2"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                    qseqid            sseqid  % identity sstrand        evalue\n",
              "0     rRNA_341494...343033  scaffold1_cov231     100.000    plus  0.000000e+00\n",
              "30  rRNA_3418942...3421829  scaffold1_cov231     100.000   minus  0.000000e+00\n",
              "43  rRNA_3424290...3427177  scaffold1_cov231     100.000   minus  0.000000e+00\n",
              "4     rRNA_343487...346374  scaffold1_cov231     100.000    plus  0.000000e+00\n",
              "27  rRNA_3418645...3418761  scaffold1_cov231     100.000   minus  5.670000e-57\n",
              "14  rRNA_2580484...2580600  scaffold1_cov231      98.276   minus  1.230000e-53"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-4013cbb7-0994-4a30-aa6f-fbd5722b6794\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>qseqid</th>\n",
              "      <th>sseqid</th>\n",
              "      <th>% identity</th>\n",
              "      <th>sstrand</th>\n",
              "      <th>evalue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>rRNA_341494...343033</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>rRNA_3418942...3421829</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>rRNA_3424290...3427177</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>rRNA_343487...346374</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>plus</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>rRNA_3418645...3418761</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>minus</td>\n",
              "      <td>5.670000e-57</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>rRNA_2580484...2580600</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>98.276</td>\n",
              "      <td>minus</td>\n",
              "      <td>1.230000e-53</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4013cbb7-0994-4a30-aa6f-fbd5722b6794')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-4013cbb7-0994-4a30-aa6f-fbd5722b6794 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-4013cbb7-0994-4a30-aa6f-fbd5722b6794');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Скачиваем файл полученный в основной части, чтобы дополнить аннотацию\n",
        "!wget https://raw.githubusercontent.com/miamib34ch/hse22_hw2/main/data/GENOME.gbk\n",
        "genome = SeqIO.to_dict(SeqIO.parse(\"/content/GENOME.gbk\", \"genbank\"))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rSjUCJ-Xj6nG",
        "outputId": "52a16875-c7bb-4e44-feb9-5826e26f794e"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 20:20:08--  https://raw.githubusercontent.com/miamib34ch/hse22_hw2/main/data/GENOME.gbk\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 7346795 (7.0M) [text/plain]\n",
            "Saving to: ‘GENOME.gbk’\n",
            "\n",
            "GENOME.gbk          100%[===================>]   7.01M  --.-KB/s    in 0.1s    \n",
            "\n",
            "2022-10-19 20:20:10 (68.0 MB/s) - ‘GENOME.gbk’ saved [7346795/7346795]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#Разбиваем полученные данные для более удобной работы с ними\n",
        "stats['Start'] = 0\n",
        "stats['End'] = 0\n",
        "\n",
        "for line in stats.iterrows():\n",
        "  cords = stats.at[line[0],\"qseqid\"].split('...')\n",
        "  stats.at[line[0],\"Start\"] = cords[0].split('_')[1]\n",
        "  stats.at[line[0],\"End\"] = cords[1]\n",
        "  if stats.at[line[0],\"sstrand\"] == 'minus':\n",
        "    stats.at[line[0],\"sstrand\"] = '-'\n",
        "  else: \n",
        "    stats.at[line[0],\"sstrand\"] = '+'\n",
        "\n",
        "stats"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 231
        },
        "id": "vaC-41mvkRE4",
        "outputId": "be00467b-97da-44ba-8b63-83592f401b75"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                    qseqid            sseqid  % identity sstrand  \\\n",
              "0     rRNA_341494...343033  scaffold1_cov231     100.000       +   \n",
              "30  rRNA_3418942...3421829  scaffold1_cov231     100.000       -   \n",
              "43  rRNA_3424290...3427177  scaffold1_cov231     100.000       -   \n",
              "4     rRNA_343487...346374  scaffold1_cov231     100.000       +   \n",
              "27  rRNA_3418645...3418761  scaffold1_cov231     100.000       -   \n",
              "14  rRNA_2580484...2580600  scaffold1_cov231      98.276       -   \n",
              "\n",
              "          evalue    Start      End  \n",
              "0   0.000000e+00   341494   343033  \n",
              "30  0.000000e+00  3418942  3421829  \n",
              "43  0.000000e+00  3424290  3427177  \n",
              "4   0.000000e+00   343487   346374  \n",
              "27  5.670000e-57  3418645  3418761  \n",
              "14  1.230000e-53  2580484  2580600  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5ab110e6-3b78-48fa-a899-54781589b279\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>qseqid</th>\n",
              "      <th>sseqid</th>\n",
              "      <th>% identity</th>\n",
              "      <th>sstrand</th>\n",
              "      <th>evalue</th>\n",
              "      <th>Start</th>\n",
              "      <th>End</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>rRNA_341494...343033</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>+</td>\n",
              "      <td>0.000000e+00</td>\n",
              "      <td>341494</td>\n",
              "      <td>343033</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>rRNA_3418942...3421829</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>-</td>\n",
              "      <td>0.000000e+00</td>\n",
              "      <td>3418942</td>\n",
              "      <td>3421829</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>rRNA_3424290...3427177</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>-</td>\n",
              "      <td>0.000000e+00</td>\n",
              "      <td>3424290</td>\n",
              "      <td>3427177</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>rRNA_343487...346374</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>+</td>\n",
              "      <td>0.000000e+00</td>\n",
              "      <td>343487</td>\n",
              "      <td>346374</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>rRNA_3418645...3418761</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>100.000</td>\n",
              "      <td>-</td>\n",
              "      <td>5.670000e-57</td>\n",
              "      <td>3418645</td>\n",
              "      <td>3418761</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>rRNA_2580484...2580600</td>\n",
              "      <td>scaffold1_cov231</td>\n",
              "      <td>98.276</td>\n",
              "      <td>-</td>\n",
              "      <td>1.230000e-53</td>\n",
              "      <td>2580484</td>\n",
              "      <td>2580600</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5ab110e6-3b78-48fa-a899-54781589b279')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-5ab110e6-3b78-48fa-a899-54781589b279 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-5ab110e6-3b78-48fa-a899-54781589b279');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#переделываем аннотацию\n",
        "for id in genome:\n",
        "  for feature in genome[id].features:\n",
        "    for line in stats.iterrows():\n",
        "      if (str(feature.location) == f\"[{line[1][5]}:{line[1][6]}]({line[1][3]})\"):\n",
        "        genome[id].annotations['molecule_type'] = 'rRna'"
      ],
      "metadata": {
        "id": "zRRe-K24l-5K"
      },
      "execution_count": 18,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#записываем результаты в генбанковский файл\n",
        "SeqIO.write(genome.values(), \"GENOME.gbk\", \"genbank\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lTogcsmc4dPz",
        "outputId": "d9939e70-b39d-4bfb-95ad-68a6b0f82eb0"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "69"
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ]
    }
  ]
}