{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "2kNe42gZn4Oc"
      },
      "source": [
        "ÐŸÐ¾Ð»Ñ‹Ð³Ð°Ð»Ð¾Ð² Ð‘Ð¾Ð³Ð´Ð°Ð½ Ð”Ð—1\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CLLeQw9Mxab4"
      },
      "source": [
        "## Ð¡ÑÑ‹Ð»ÐºÐ° Ð½Ð° Ð´Ð¸ÑÐº Ñ ÐºÐ°Ñ€Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð½Ñ‹Ð¼Ð¸ Ñ‡Ñ‚ÐµÐ½Ð¸ÑÐ¼Ð¸"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "iX7NcoC_CLY0"
      },
      "source": [
        "\n",
        "| | SRA accession  | Stage    |\n",
        "|---:|:-------------|:-----------|\n",
        "| 1 | SRR5836473  | 8 cell       |\n",
        "| 2 | SRR3824222 | Epiblast    |\n",
        "| 3 | SRR5836475  | ICM       |\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "RQAK_lggxsXi"
      },
      "source": [
        "[Bam-Ñ„Ð°Ð¹Ð»Ñ‹ Ð¸ Ð¾Ñ‚Ñ‡ÐµÑ‚Ñ‹](https://drive.google.com/drive/folders/1athn52a93obOwfuHynDCRPRhjAPB80aO?usp=sharing)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5l0a7C6JHe40"
      },
      "source": [
        "## Ð£ÑÑ‚Ð°Ð½Ð¾Ð²ÐºÐ° FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "esYQuHKO8nAg",
        "outputId": "a23d7833-dd63-40f4-ca52-03adb37c97f2"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2023-02-07 16:48:39--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: â€˜fastqc_v0.11.9.zipâ€™\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M   121KB/s    in 38s     \n",
            "\n",
            "2023-02-07 16:49:18 (264 KB/s) - â€˜fastqc_v0.11.9.zipâ€™ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_zEx3USYPKoR"
      },
      "source": [
        "ÐœÐ¾Ð¶Ð½Ð¾ Ð¿Ð¾Ð»ÑƒÑ‡Ð°Ñ‚ÑŒ Ð´Ð°Ð½Ð½Ñ‹Ðµ ÑÐµÐºÐ²ÐµÐ½Ð¸Ñ€Ð¾Ð²Ð¸Ñ Ñ ENA. ÐÐ°Ð¿Ñ€Ð¸Ð¼ÐµÑ€, https://www.ebi.ac.uk/ena/browser/view/SRR5836473"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c5QMGKJYxrza",
        "outputId": "8f0ce104-491a-4a70-abd6-9f8843b89365"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2023-02-07 16:50:49--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
            "           => â€˜SRR5836473_1.fastq.gzâ€™\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
            "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
            "Length: 6856386252 (6.4G) (unauthoritative)\n",
            "\n",
            "SRR5836473_1.fastq. 100%[===================>]   6.38G  28.1MB/s    in 3m 48s  \n",
            "\n",
            "2023-02-07 16:54:39 (28.7 MB/s) - â€˜SRR5836473_1.fastq.gzâ€™ saved [6856386252]\n",
            "\n",
            "--2023-02-07 16:54:39--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
            "           => â€˜SRR5836473_2.fastq.gzâ€™\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_2.fastq.gz ... 8127008133\n",
            "==> PASV ... done.    ==> RETR SRR5836473_2.fastq.gz ... done.\n",
            "Length: 8127008133 (7.6G) (unauthoritative)\n",
            "\n",
            "SRR5836473_2.fastq. 100%[===================>]   7.57G  29.0MB/s    in 4m 32s  \n",
            "\n",
            "2023-02-07 16:59:13 (28.4 MB/s) - â€˜SRR5836473_2.fastq.gzâ€™ saved [8127008133]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "n4KxvlAPv4kw"
      },
      "source": [
        "Ð¡Ð¾Ð·Ð´Ð°Ñ‘Ð¼ Ð¾Ñ‚Ñ‡Ñ‘Ñ‚Ñ‹"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yjup-N1wv6cB",
        "outputId": "565ed930-2b4e-45b3-ba5a-37e619996e3a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Started analysis of SRR5836473_1.fastq\n",
            "Approx 5% complete for SRR5836473_1.fastq\n",
            "Approx 10% complete for SRR5836473_1.fastq\n",
            "Approx 15% complete for SRR5836473_1.fastq\n",
            "Approx 20% complete for SRR5836473_1.fastq\n",
            "Approx 25% complete for SRR5836473_1.fastq\n",
            "Approx 30% complete for SRR5836473_1.fastq\n",
            "Approx 35% complete for SRR5836473_1.fastq\n",
            "Approx 40% complete for SRR5836473_1.fastq\n",
            "Approx 45% complete for SRR5836473_1.fastq\n",
            "Approx 50% complete for SRR5836473_1.fastq\n",
            "Approx 55% complete for SRR5836473_1.fastq\n",
            "Approx 60% complete for SRR5836473_1.fastq\n",
            "Approx 65% complete for SRR5836473_1.fastq\n",
            "Approx 70% complete for SRR5836473_1.fastq\n",
            "Approx 75% complete for SRR5836473_1.fastq\n",
            "Approx 80% complete for SRR5836473_1.fastq\n",
            "Approx 85% complete for SRR5836473_1.fastq\n",
            "Approx 90% complete for SRR5836473_1.fastq\n",
            "Approx 95% complete for SRR5836473_1.fastq\n",
            "Analysis complete for SRR5836473_1.fastq\n"
          ]
        }
      ],
      "source": [
        "!gunzip SRR5836473_1.fastq.gz\n",
        "!FastQC/fastqc  /content/SRR5836473_1.fastq\n",
        "!rm SRR5836473_1.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lopV-n96wAm_",
        "outputId": "0b915289-a2e4-4a4a-8b70-11e0637c4976"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Started analysis of SRR5836473_2.fastq\n",
            "Approx 5% complete for SRR5836473_2.fastq\n",
            "Approx 10% complete for SRR5836473_2.fastq\n",
            "Approx 15% complete for SRR5836473_2.fastq\n",
            "Approx 20% complete for SRR5836473_2.fastq\n",
            "Approx 25% complete for SRR5836473_2.fastq\n",
            "Approx 30% complete for SRR5836473_2.fastq\n",
            "Approx 35% complete for SRR5836473_2.fastq\n",
            "Approx 40% complete for SRR5836473_2.fastq\n",
            "Approx 45% complete for SRR5836473_2.fastq\n",
            "Approx 50% complete for SRR5836473_2.fastq\n",
            "Approx 55% complete for SRR5836473_2.fastq\n",
            "Approx 60% complete for SRR5836473_2.fastq\n",
            "Approx 65% complete for SRR5836473_2.fastq\n",
            "Approx 70% complete for SRR5836473_2.fastq\n",
            "Approx 75% complete for SRR5836473_2.fastq\n",
            "Approx 80% complete for SRR5836473_2.fastq\n",
            "Approx 85% complete for SRR5836473_2.fastq\n",
            "Approx 90% complete for SRR5836473_2.fastq\n",
            "Approx 95% complete for SRR5836473_2.fastq\n",
            "Analysis complete for SRR5836473_2.fastq\n"
          ]
        }
      ],
      "source": [
        "!gunzip SRR5836473_2.fastq.gz\n",
        "!FastQC/fastqc  /content/SRR5836473_2.fastq\n",
        "!rm SRR5836473_2.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "48ot2hG-wMte"
      },
      "outputs": [],
      "source": [
        "!rm /content/SRR5836473_1_fastqc.html\n",
        "!rm /content/SRR5836473_2_fastqc.html"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ugk3bV0LI5El"
      },
      "source": [
        "# Ð£ÑÑ‚Ð°Ð½Ð¾Ð²ÐºÐ° Bismark Ñ Ð·Ð°Ð²Ð¸ÑÐ¸Ð¼Ð¾ÑÑ‚ÑÐ¼Ð¸ (+bowtie2, samtools)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uSOZLyn5n5Nl",
        "outputId": "c2851f9b-7b68-40e2-e05c-53ccd1c67863"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "â¬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...\n",
            "ðŸ“¦ Installing...\n",
            "ðŸ“Œ Adjusting configuration...\n",
            "ðŸ©¹ Patching environment...\n",
            "â² Done in 0:00:23\n",
            "ðŸ” Restarting kernel...\n"
          ]
        }
      ],
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "npb--DNtn52o",
        "outputId": "72cc1996-f03d-49b2-eb85-526d28ba2165"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "âœ¨ðŸ°âœ¨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "import condacolab\n",
        "condacolab.check()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MfBSN4YuoCL6",
        "outputId": "7c0980ed-8710-4c54-8500-d31c39930766"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bismark\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bismark-0.24.0             |       hdfd78af_0         6.6 MB  bioconda\n",
            "    bowtie2-2.5.1              |   py38h77f66f0_0        14.8 MB  bioconda\n",
            "    ca-certificates-2022.12.7  |       ha878542_0         143 KB  conda-forge\n",
            "    certifi-2022.12.7          |     pyhd8ed1ab_0         147 KB  conda-forge\n",
            "    colorama-0.4.6             |     pyhd8ed1ab_0          25 KB  conda-forge\n",
            "    conda-22.11.1              |   py38h578d9bd_1         905 KB  conda-forge\n",
            "    hisat2-2.2.1               |       h87f3376_4        16.5 MB  bioconda\n",
            "    k8-0.2.5                   |       hd03093a_2         1.7 MB  bioconda\n",
            "    libhwloc-2.8.0             |       h32351e8_1         3.0 MB  conda-forge\n",
            "    minimap2-2.24              |       h7132678_1         484 KB  bioconda\n",
            "    openssl-3.0.8              |       h0b41bf4_0         2.5 MB  conda-forge\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    pluggy-1.0.0               |     pyhd8ed1ab_5          16 KB  conda-forge\n",
            "    ruamel.yaml-0.17.21        |   py38h0a891b7_2         172 KB  conda-forge\n",
            "    ruamel.yaml.clib-0.2.7     |   py38h1de0b5d_1         143 KB  conda-forge\n",
            "    samtools-1.3.1             |                0         1.5 MB  bioconda\n",
            "    tbb-2021.7.0               |       h924138e_1         1.5 MB  conda-forge\n",
            "    tqdm-4.64.1                |     pyhd8ed1ab_0          82 KB  conda-forge\n",
            "    zlib-1.2.13                |       h166bdaf_4          92 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        64.6 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bismark            bioconda/noarch::bismark-0.24.0-hdfd78af_0 None\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.1-py38h77f66f0_0 None\n",
            "  colorama           conda-forge/noarch::colorama-0.4.6-pyhd8ed1ab_0 None\n",
            "  hisat2             bioconda/linux-64::hisat2-2.2.1-h87f3376_4 None\n",
            "  k8                 bioconda/linux-64::k8-0.2.5-hd03093a_2 None\n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.8.0-h32351e8_1 None\n",
            "  minimap2           bioconda/linux-64::minimap2-2.24-h7132678_1 None\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5 None\n",
            "  pluggy             conda-forge/noarch::pluggy-1.0.0-pyhd8ed1ab_5 None\n",
            "  ruamel.yaml        conda-forge/linux-64::ruamel.yaml-0.17.21-py38h0a891b7_2 None\n",
            "  ruamel.yaml.clib   conda-forge/linux-64::ruamel.yaml.clib-0.2.7-py38h1de0b5d_1 None\n",
            "  samtools           bioconda/linux-64::samtools-1.3.1-0 None\n",
            "  tbb                conda-forge/linux-64::tbb-2021.7.0-h924138e_1 None\n",
            "  tqdm               conda-forge/noarch::tqdm-4.64.1-pyhd8ed1ab_0 None\n",
            "  zlib               conda-forge/linux-64::zlib-1.2.13-h166bdaf_4 None\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2022.9.24-ha878542_0 --> 2022.12.7-ha878542_0 None\n",
            "  certifi                            2022.9.24-pyhd8ed1ab_0 --> 2022.12.7-pyhd8ed1ab_0 None\n",
            "  conda                               22.9.0-py38h578d9bd_2 --> 22.11.1-py38h578d9bd_1 None\n",
            "  openssl                                  3.0.7-h0b41bf4_1 --> 3.0.8-h0b41bf4_0 None\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "pluggy-1.0.0         | 16 KB     | : 100% 1.0/1 [00:00<00:00,  7.78it/s]\n",
            "ruamel.yaml.clib-0.2 | 143 KB    | : 100% 1.0/1 [00:00<00:00, 17.63it/s]\n",
            "ruamel.yaml-0.17.21  | 172 KB    | : 100% 1.0/1 [00:00<00:00,  6.51it/s]\n",
            "bismark-0.24.0       | 6.6 MB    | : 100% 1.0/1 [00:02<00:00,  2.03s/it]               \n",
            "hisat2-2.2.1         | 16.5 MB   | : 100% 1.0/1 [00:04<00:00,  4.96s/it]               \n",
            "k8-0.2.5             | 1.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.45it/s]\n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:03<00:00,  3.46s/it]              \n",
            "tbb-2021.7.0         | 1.5 MB    | : 100% 1.0/1 [00:00<00:00, 10.52it/s]\n",
            "zlib-1.2.13          | 92 KB     | : 100% 1.0/1 [00:00<00:00, 18.22it/s]\n",
            "libhwloc-2.8.0       | 3.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.21it/s]\n",
            "ca-certificates-2022 | 143 KB    | : 100% 1.0/1 [00:00<00:00, 22.42it/s]\n",
            "conda-22.11.1        | 905 KB    | : 100% 1.0/1 [00:00<00:00,  4.05it/s]\n",
            "certifi-2022.12.7    | 147 KB    | : 100% 1.0/1 [00:00<00:00, 24.67it/s]\n",
            "tqdm-4.64.1          | 82 KB     | : 100% 1.0/1 [00:00<00:00, 10.77it/s]\n",
            "perl-5.32.1          | 14.4 MB   | : 100% 1.0/1 [00:07<00:00,  7.15s/it]               \n",
            "openssl-3.0.8        | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  4.06it/s]\n",
            "samtools-1.3.1       | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.30it/s]\n",
            "minimap2-2.24        | 484 KB    | : 100% 1.0/1 [00:00<00:00,  6.25it/s]\n",
            "colorama-0.4.6       | 25 KB     | : 100% 1.0/1 [00:00<00:00, 20.51it/s]\n",
            "Preparing transaction: \\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Retrieving notices: ...working... done\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda bismark"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WX0zmvL9__e6",
        "outputId": "1dac196d-12c4-45bf-c840-5daf604f0444"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following NEW packages will be installed:\n",
            "  libncurses5\n",
            "0 upgraded, 1 newly installed, 0 to remove and 28 not upgraded.\n",
            "Need to get 96.9 kB of archives.\n",
            "After this operation, 330 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 libncurses5 amd64 6.2-0ubuntu2 [96.9 kB]\n",
            "Fetched 96.9 kB in 0s (237 kB/s)\n",
            "Selecting previously unselected package libncurses5:amd64.\n",
            "(Reading database ... 129554 files and directories currently installed.)\n",
            "Preparing to unpack .../libncurses5_6.2-0ubuntu2_amd64.deb ...\n",
            "Unpacking libncurses5:amd64 (6.2-0ubuntu2) ...\n",
            "Setting up libncurses5:amd64 (6.2-0ubuntu2) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install libncurses5"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "13VM_exnoHEl",
        "outputId": "70dd49d1-a343-4095-877e-85ca20dd5797"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
            "\n",
            "Options:\n",
            "  -b       output BAM\n",
            "  -C       output CRAM (requires -T)\n",
            "  -1       use fast BAM compression (implies -b)\n",
            "  -u       uncompressed BAM output (implies -b)\n",
            "  -h       include header in SAM output\n",
            "  -H       print SAM header only (no alignments)\n",
            "  -c       print only the count of matching records\n",
            "  -o FILE  output file name [stdout]\n",
            "  -U FILE  output reads not selected by filters to FILE [null]\n",
            "  -t FILE  FILE listing reference names and lengths (see long help) [null]\n",
            "  -L FILE  only include reads overlapping this BED FILE [null]\n",
            "  -r STR   only include reads in read group STR [null]\n",
            "  -R FILE  only include reads with read group listed in FILE [null]\n",
            "  -q INT   only include reads with mapping quality >= INT [0]\n",
            "  -l STR   only include reads in library STR [null]\n",
            "  -m INT   only include reads with number of CIGAR operations consuming\n",
            "           query sequence >= INT [0]\n",
            "  -f INT   only include reads with all bits set in INT set in FLAG [0]\n",
            "  -F INT   only include reads with none of the bits set in INT set in FLAG [0]\n",
            "  -x STR   read tag to strip (repeatable) [null]\n",
            "  -B       collapse the backward CIGAR operation\n",
            "  -s FLOAT integer part sets seed of random number generator [0];\n",
            "           rest sets fraction of templates to subsample [no subsampling]\n",
            "  -@, --threads INT\n",
            "           number of BAM/CRAM compression threads [0]\n",
            "  -?       print long help, including note about region specification\n",
            "  -S       ignored (input format is auto-detected)\n",
            "      --input-fmt-option OPT[=VAL]\n",
            "               Specify a single input file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
            "               Specify output format (SAM, BAM, CRAM)\n",
            "      --output-fmt-option OPT[=VAL]\n",
            "               Specify a single output file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -T, --reference FILE\n",
            "               Reference sequence FASTA FILE [null]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! samtools view"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# ÐžÑÐ½Ð¾Ð²Ð½Ð°Ñ Ñ‡Ð°ÑÑ‚ÑŒ"
      ],
      "metadata": {
        "id": "em0FAacEXnkm"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lBUia_8DP-gu"
      },
      "source": [
        "ÐœÐ¾Ð¶Ð½Ð¾ Ð·Ð°Ð³Ñ€ÑƒÐ·Ð¸Ñ‚ÑŒ Ñ„Ð°Ð¹Ð» Ñ google drive Ð½Ð°Ð¿Ñ€ÑÐ¼ÑƒÑŽ:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cAlLzmWkm1DW",
        "outputId": "3d36015e-8658-4201-cb55-c788ebe04aa9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ],
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SrGGCxZQ2Ao3",
        "outputId": "1f4730f5-bb35-4151-d66d-e576661ed0b9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 5 files...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort /content/drive/MyDrive/temp/cell8/SRR5836473_1_bismark_bt2_pe.bam -o sorted_8cell.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nPh11hgOs6bO",
        "outputId": "0c964e04-867f-4fe3-94e3-dece57c8e978"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 11 files...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort /content/drive/MyDrive/temp/epiblast/SRR3824222_1_bismark_bt2_pe.bam -o sorted_epiblast.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dRR0o077s7__",
        "outputId": "91ca21bc-2593-4683-e994-f62be83c671d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 7 files...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort /content/drive/MyDrive/temp/ICM/SRR5836475_1_bismark_bt2_pe.bam -o sorted_icm.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "GErvoOM-3Axj"
      },
      "outputs": [],
      "source": [
        "! samtools index  sorted_8cell.bam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools index  sorted_epiblast.bam"
      ],
      "metadata": {
        "id": "f9QcnX-WX8Vg"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools index  sorted_icm.bam"
      ],
      "metadata": {
        "id": "ydJC3KiKX8tQ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Lw--6KavBoXp"
      },
      "source": [
        "## Ð§Ð¸ÑÐ»Ð¾ Ñ€Ð¸Ð´Ð¾Ð², Ð·Ð°ÐºÐ°Ñ€Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð½Ñ‹Ñ… Ð½Ð° ÑƒÑ‡Ð°ÑÑ‚ÐºÐ°Ñ… 11347700-11367700; 40185800-40195800"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lAqz7CMqZjAL",
        "outputId": "b0ee02f1-f308-4b70-8631-b14618a9fe77"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1090\n",
            "464\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_icm.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_icm.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hj5mBPNkZnoQ",
        "outputId": "6d49a93d-059b-419e-f030-a48c1df924f6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1456\n",
            "630\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Odx1QahXZmiN",
        "outputId": "ac380b53-5e9c-4005-ca93-3931bbc0062e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2328\n",
            "1062\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CliRCSUQTmV-"
      },
      "source": [
        "## Ð”ÐµÐ´ÑƒÐ¿Ð»Ð¸ÐºÐ°Ñ†Ð¸Ñ"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls *bt2_pe.bam | xargs -P 4 -tI{} deduplicate_bismark  --bam  --paired  -o s_{} {} \n",
        "# Ð´ÐµÐ´ÑƒÐ¿Ð»Ð¸ÐºÐ°Ñ†Ð¸Ñ Ð²ÑÐµÑ… Ñ„Ð°Ð¹Ð»Ð¾Ð² ÑÑ€Ð°Ð·Ñƒ"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dWpmkaZfa9wu",
        "outputId": "e8a8ac1d-f362-4299-dbe0-433ef76f17ea"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "deduplicate_bismark --bam --paired -o s_SRR3824222_1_bismark_bt2_pe.bam SRR3824222_1_bismark_bt2_pe.bam \n",
            "deduplicate_bismark --bam --paired -o s_SRR5836473_1_bismark_bt2_pe.bam SRR5836473_1_bismark_bt2_pe.bam \n",
            "deduplicate_bismark --bam --paired -o s_SRR5836475_1_bismark_bt2_pe.bam SRR5836475_1_bismark_bt2_pe.bam \n",
            "Output filename was given as: s_SRR3824222_1_bismark_bt2_pe.bam\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "Output filename was given as: s_SRR5836473_1_bismark_bt2_pe.bam\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR3824222_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "Output filename was given as: s_SRR5836475_1_bismark_bt2_pe.bam\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836473_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836475_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836473_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t\n",
            "\n",
            "Now testing Bismark result file SRR5836475_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t\n",
            "\n",
            "Now testing Bismark result file SRR3824222_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "...passed!\n",
            "Output file is: s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "...passed!\n",
            "Output file is: s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:\t2850231\n",
            "Total number duplicated alignments removed:\t521904 (18.31%)\n",
            "Duplicated alignments were found at:\t230432 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 2328327 (81.69% of total)\n",
            "\n",
            "\n",
            "Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:\t4159998\n",
            "Total number duplicated alignments removed:\t377882 (9.08%)\n",
            "Duplicated alignments were found at:\t229833 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 3782116 (90.92% of total)\n",
            "\n",
            "\n",
            "Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:\t7039016\n",
            "Total number duplicated alignments removed:\t205258 (2.92%)\n",
            "Duplicated alignments were found at:\t164796 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 6833758 (97.08% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:**\t\n",
        "**Total count of deduplicated leftover sequences: 2328327 (81.69% of total)**\n",
        "\n",
        "\n",
        "**Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:**\t\n",
        "**Total count of deduplicated leftover sequences: 3782116 (90.92% of total)**\n",
        "\n",
        "\n",
        "**Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:**\t\n",
        "**Total count of deduplicated leftover sequences: 6833758 (97.08% of total)**\n"
      ],
      "metadata": {
        "id": "eX6HVi0jidZ-"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "LKBD3cwo1wLx"
      },
      "source": [
        "## ÐšÐ¾Ð»Ð»Ð¸Ð½Ð³ Ð¼ÐµÑ‚Ð¸Ð»Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¸Ñ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4c5BEvKpDw5K",
        "outputId": "713b3653-d2cb-4e9f-9167-151d2dd82f5f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file /content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as /content/drive/MyDrive/temp/mouse_chr/\n",
            "\n",
            "Checking file >>/content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >/content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file /content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "\n",
            "Processed 3782116 lines in total\n",
            "Total number of methylation call strings processed: 7564232\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t177157754\n",
            "\n",
            "Total methylated C's in CpG context:\t1914374\n",
            "Total methylated C's in CHG context:\t297421\n",
            "Total methylated C's in CHH context:\t1066806\n",
            "\n",
            "Total C to T conversions in CpG context:\t6262904\n",
            "Total C to T conversions in CHG context:\t39194423\n",
            "Total C to T conversions in CHH context:\t128421826\n",
            "\n",
            "C methylated in CpG context:\t23.4%\n",
            "C methylated in CHG context:\t0.8%\n",
            "C methylated in CHH context:\t0.8%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz CpG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz CHG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz CHG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz CHH_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz CHH_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_SRR5836475_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\t/content/CpG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_SRR5836475_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_SRR5836475_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_SRR5836475_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_SRR5836475_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/drive/MyDrive/temp/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/drive/MyDrive/temp/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_SRR5836475_1_bismark_bt2_pe.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_SRR5836475_1_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2053827 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder /content/drive/MyDrive/temp/mouse_chr /content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder /content/drive/MyDrive/temp/mouse_chr /content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zkcv6XQlmE8u",
        "outputId": "1b247775-81b7-4333-840e-e3425d6825d9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file /content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as /content/drive/MyDrive/temp/mouse_chr/\n",
            "\n",
            "Checking file >>/content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >/content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file /content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "\n",
            "Processed 2328327 lines in total\n",
            "Total number of methylation call strings processed: 4656654\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t104224209\n",
            "\n",
            "Total methylated C's in CpG context:\t2101489\n",
            "Total methylated C's in CHG context:\t269204\n",
            "Total methylated C's in CHH context:\t944695\n",
            "\n",
            "Total C to T conversions in CpG context:\t2731177\n",
            "Total C to T conversions in CHG context:\t22964928\n",
            "Total C to T conversions in CHH context:\t75212716\n",
            "\n",
            "C methylated in CpG context:\t43.5%\n",
            "C methylated in CHG context:\t1.2%\n",
            "C methylated in CHH context:\t1.2%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz CpG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz CHG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz CHG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz CHH_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz CHH_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_SRR5836473_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\t/content/CpG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_SRR5836473_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_SRR5836473_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_SRR5836473_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_SRR5836473_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/drive/MyDrive/temp/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/drive/MyDrive/temp/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_SRR5836473_1_bismark_bt2_pe.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_SRR5836473_1_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 1736778 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder /content/drive/MyDrive/temp/mouse_chr /content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DKkBthNcmFRR",
        "outputId": "7743e5bc-8e5f-4ff5-8e1f-60bc59281bfe"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.0 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file /content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as /content/drive/MyDrive/temp/mouse_chr/\n",
            "\n",
            "Checking file >>/content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >/content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file /content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "Processed lines: 4000000\n",
            "Processed lines: 4500000\n",
            "Processed lines: 5000000\n",
            "Processed lines: 5500000\n",
            "Processed lines: 6000000\n",
            "Processed lines: 6500000\n",
            "\n",
            "Processed 6833758 lines in total\n",
            "Total number of methylation call strings processed: 13667516\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t330808061\n",
            "\n",
            "Total methylated C's in CpG context:\t11544557\n",
            "Total methylated C's in CHG context:\t1572954\n",
            "Total methylated C's in CHH context:\t3695988\n",
            "\n",
            "Total C to T conversions in CpG context:\t3465254\n",
            "Total C to T conversions in CHG context:\t71995231\n",
            "Total C to T conversions in CHH context:\t238534077\n",
            "\n",
            "C methylated in CpG context:\t76.9%\n",
            "C methylated in CHG context:\t2.1%\n",
            "C methylated in CHH context:\t1.5%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz CpG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz CHG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz CHG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz CHH_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz CHH_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_SRR3824222_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\t/content/CpG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_SRR3824222_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_SRR3824222_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_SRR3824222_1_bismark_bt2_pe.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_SRR3824222_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/drive/MyDrive/temp/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/drive/MyDrive/temp/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_SRR3824222_1_bismark_bt2_pe.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_SRR3824222_1_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2249610 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report --alignment_report /content/SRR5836475_1_bismark_bt2_PE_report.txt --mbias_report /content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
        "! bismark2report --alignment_report /content/SRR5836473_1_bismark_bt2_PE_report.txt --mbias_report /content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.M-bias.txt\n",
        "! bismark2report --alignment_report /content/SRR3824222_1_bismark_bt2_PE_report.txt --mbias_report /content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.M-bias.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qdGsWDXgPxLL",
        "outputId": "1496cbab-9b31-45b2-d36d-51cfc084cd1a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> /content/SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report /content/SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> /content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
            "Processing M-bias report /content/s_SRR5836475_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> /content/SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report /content/SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> /content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
            "Processing M-bias report /content/s_SRR5836473_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> /content/SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report /content/SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> /content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
            "Processing M-bias report /content/s_SRR3824222_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## ÐŸÐ¾ÑÑ‚Ñ€Ð¾Ð¹Ñ‚Ðµ Ð³Ð¸ÑÑ‚Ð¾Ð³Ñ€Ð°Ð¼Ð¼Ñƒ Ñ€Ð°ÑÐ¿Ñ€ÐµÐ´ÐµÐ»ÐµÐ½Ð¸Ñ Ð¼ÐµÑ‚Ð¸Ð»Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð¸Ñ Ñ†Ð¸Ñ‚Ð¾Ð·Ð¸Ð½Ð¾Ð² Ð¿Ð¾ Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ðµ"
      ],
      "metadata": {
        "id": "s2WeHYA2U9On"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! ls *bedGraph.gz | xargs -tI{} gzip -d {}\n",
        "#! ls *cov.gz | xargs -tI{} gzip -d {}"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8OxwtZgVV6cI",
        "outputId": "1e259ba4-bd81-4016-8c06-2586204b0aa6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "gzip -d s_SRR3824222_1_bismark_bt2_pe.deduplicated.bedGraph.gz \n",
            "gzip -d s_SRR5836473_1_bismark_bt2_pe.deduplicated.bedGraph.gz \n",
            "gzip -d s_SRR5836475_1_bismark_bt2_pe.deduplicated.bedGraph.gz \n",
            "gzip -d s_SRR3824222_1_bismark_bt2_pe.deduplicated.bismark.cov.gz \n",
            "gzip -d s_SRR5836473_1_bismark_bt2_pe.deduplicated.bismark.cov.gz \n",
            "gzip -d s_SRR5836475_1_bismark_bt2_pe.deduplicated.bismark.cov.gz \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import matplotlib.pyplot as plt\n",
        "import seaborn as sns"
      ],
      "metadata": {
        "id": "y-78ZY0YVqkO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "ids = ['3824222', '5836473', '5836475']\n",
        "header = ['chr', 'start', 'ston', 'M%']\n",
        "for id in ids:\n",
        "  line = 's_SRR' + id + '_1_bismark_bt2_pe.deduplicated.bedGraph'\n",
        "  df = pd.read_csv(line, delimiter='\\t', skiprows=1, names = header)\n",
        "  fig = plt.figure(figsize=(10, 5))\n",
        "  a = sns.histplot(data=df, kde=True, x='M%', bins=10)\n",
        "  if (id == '5836473'):\n",
        "    a.set_title('8-cell')\n",
        "  if (id == '5836475'):\n",
        "    a.set_title('ICM')\n",
        "  if (id == '3824222'):\n",
        "    a.set_title('epiblast')\n",
        "  a.set(ylim=(0, 3500000)) \n",
        "  plt.ylabel('Ð§Ð°ÑÑ‚Ð¾Ñ‚Ð°')\n",
        "  plt.show()\n",
        "  fig.savefig('{}.png'.format(id))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "bzh9jFWwU_L6",
        "outputId": "8f758a4f-b323-4812-d6b8-e2ba68b0e219"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 720x360 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmEAAAFNCAYAAABIc7ibAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deZxcZZn3/89VS+9bOt3ZVyALIbIGDCJOEFFABBSURdxGxXFXnPFRnxkVHWd+LoPPIC4DiIoiouggIirIoogESAIJJCGQEJJ0tl7S+1Jdy/X7o6qbTugknaSrTlX39/16FV11zqmqq7tyur/c933u29wdEREREcmtUNAFiIiIiIxHCmEiIiIiAVAIExEREQmAQpiIiIhIABTCRERERAKgECYiIiISAIUwERkXzGytmS3L3P+ymf3sAMe+ZGZvyFlxIjIuRYIuQEQkF9z9uFy+n5n9GGhw93/N5fuKSOFQS5iIiIhIABTCRKSgmNk0M/u1mTWZ2WYz+0Rm+5fN7E4zu8PMOs1slZmdMOR5+3Yxluzv2H3e7zQze8zM2sxsp5ndYGZFmX1mZt82s0Yz6zCzZ8xssZldDbwT+KyZdZnZ77L4IxGRAqUQJiIFw8xCwO+A1cB04GzgU2b2pswhFwG/AmqBnwN3mVl0Py830mOTwKeBOuD0zHt+JLPvjcDrgPlANfAOoMXdbwRuA77h7hXu/pbD/qZFZMwqyBBmZrdk/s/z2REe/w4zW5cZmPvzbNcnIllzKlDv7l9x9353fxG4Cbg8s3+lu9/p7nHgOqAEWLqf1xrRse6+0t2Xu3vC3V8C/gf4h8zuOFAJLATM3de7+87R+VZFZKwryBAG/Bg4dyQHmtk84PPAGZmBuZ/KYl0ikl2zgWmZrsE2M2sDvgBMzuzfNnCgu6eABmDafl5rRMea2Xwzu8fMdplZB/AfpFvFcPcHgRuA7wKNZnajmVUd6TcpIuNDQYYwd/8rsGfoNjM72sz+aGYrzewRM1uY2fVB4Lvu3pp5bmOOyxWR0bMN2OzuNUNule5+fmb/zIEDM12XM4Ad+3mtkR77feA5YJ67V5EOfTaw092vd/dTgEWkuyX/ZWDX4XyDIjJ+FGQI248bgY9nfhn+M/C9zPb5wHwze9TMlpvZiFrQRCQvPQF0mtn/MbNSMwtnBsKfmtl/ipm9zcwipFu9Y8Dy/bzWSI+tBDqArsz/3H14YIeZnWpmr86MJesG+oBUZvdu4Kgj+3ZFZCwbEyHMzCqA1wC/MrOnSY/ZmJrZHQHmAcuAK4CbzKwmiDpF5Mi4exK4ADgR2Aw0AzeTHhQP8FvgMqAVeBfwtsyYr+GM9Nh/Bq4EOkmPP7tjyL6qzLZWYAvQAnwzs++HwKJMt+ldh/zNisiYZ+6F2WJuZnOAe9x9cWYMxgZ3nzrMcT8AHnf3H2UePwB8zt2fzGW9IpJdZvZl4Bh3vyroWkRERmJMtIS5ewew2czeDoNz9wzM+XMX6VYwzKyOdPfki0HUKSIiIjIgayHMzErM7AkzW52ZGuLaYY55b2bCxacztw+M8LVvBx4DFphZg5m9n/TEiO83s9XAWtJzAAH8CWgxs3XAQ8C/uHvLaHyPIiIiIocra92RZmZAubt3ZQat/g34pLsvH3LMe4El7v6xrBQhIiIikqeytoC3p9NdV+ZhNHMrzAFoIiIiIqMsq2PCMpePPw00Ave7++PDHHaJma3JrPk2c5j9IiIiImNOTq6OzEwJ8b+k5/F6dsj2iUCXu8fM7EPAZe7++mGefzVwNUB5efkpCxcu3PcQERERkbyzcuXKZnevH25fzqaoMLMvAj3u/q397A8De9y9erj9A5YsWeIrVqzIRokiIiIio8rMVrr7kuH2ZfPqyPqBSVHNrBQ4h/TSH0OPGTqv14XA+mzVIyIiIpJPsjYwn/SM9T/JtHCFgF+6+z1m9hVghbvfDXzCzC4EEqTXgnxvFusRERERyRsFN2O+uiNFRESkUATSHSkiIiIi+6cQJiIiIhIAhTARERGRACiEiYiIiARAIUxEREQkAAphIiIiIgFQCBMREREJgEKYiIiISAAUwkREREQCoBAmIiIiEgCFMBEREZEAKISJiIiIBEAhTERERCQACmEiIiIiAVAIExEREQmAQpiIiIhIABTCRERERAKgECYiIiISAIUwERERkQAohImIiIgEQCFMREREJAAKYSIiIiIBUAgTERERCYBCmIiIiEgAFMJEREREAqAQJiIiIhIAhTARERGRACiEiYiIiARAIUxEREQkAAphIiIiIgHIWggzsxIze8LMVpvZWjO7dphjis3sDjPbaGaPm9mcbNUjIiIikk+y2RIWA17v7icAJwLnmtnSfY55P9Dq7scA3wa+nsV6RERERPJG1kKYp3VlHkYzN9/nsIuAn2Tu3wmcbWaWrZpERERE8kVWx4SZWdjMngYagfvd/fF9DpkObANw9wTQDkzMZk0iIiIi+SCrIczdk+5+IjADOM3MFh/O65jZ1Wa2wsxWNDU1jW6RIiIiIgHIydWR7t4GPAScu8+u7cBMADOLANVAyzDPv9Hdl7j7kvr6+myXKyIiIpJ12bw6st7MajL3S4FzgOf2Oexu4D2Z+5cCD7r7vuPGRERERMacSBZfeyrwEzMLkw57v3T3e8zsK8AKd78b+CHwUzPbCOwBLs9iPSIiIiJ5I2shzN3XACcNs/2LQ+73AW/PVg0iIiIi+Uoz5ouIiIgEQCFMREREJAAKYSIiIiIBUAgTERERCYBCmIiIiEgAFMJEREREAqAQJiIiIhIAhTARERGRACiEiYiIiARAIUxEREQkAAphIiIiIgFQCBMREREJgEKYiIiISAAUwkREREQCoBAmIiIiEgCFMBEREZEAKISJiIiIBEAhTERERCQACmEiIiIiAVAIExEREQmAQpiIiIhIABTCRERERAKgECYiIiISAIUwERERkQAohImIiIgEQCFMREREJAAKYSIiIiIBUAgTERERCYBCmIiIiEgAFMJEREREApC1EGZmM83sITNbZ2ZrzeyTwxyzzMzazezpzO2L2apHREREJJ9EsvjaCeAz7r7KzCqBlWZ2v7uv2+e4R9z9gizWISIiIpJ3stYS5u473X1V5n4nsB6Ynq33ExERESkkORkTZmZzgJOAx4fZfbqZrTazP5jZcbmoR0RERCRo2eyOBMDMKoBfA59y9459dq8CZrt7l5mdD9wFzBvmNa4GrgaYNWtWlisWERERyb6stoSZWZR0ALvN3X+z735373D3rsz9e4GomdUNc9yN7r7E3ZfU19dns2QRERGRnMjm1ZEG/BBY7+7X7eeYKZnjMLPTMvW0ZKsmERERkXyRze7IM4B3Ac+Y2dOZbV8AZgG4+w+AS4EPm1kC6AUud3fPYk0iIiIieSFrIczd/wbYQY65AbghWzWIiIiI5CvNmC8iIiISAIUwERERkQAohImIiIgEQCFMREREJAAKYSIiIiIBUAgTERERCYBCmIiIiEgAFMJEREREAqAQJiIiIhIAhTAREREZd3762Eus3LIn0BoUwkRERGRcae3u56v3rOeeNTsDrUMhTERERMaV3z69nf5kirefMjPQOhTCREREZFz51coGFk+vYtG0qkDrUAgTERGRcePZ7e2s3dHBO5YE2woGCmEiIiIyjty5soGicIgLT5gWdCkKYSIiIjI+xBJJ7np6O288bjI1ZUVBl6MQJiIiIuPDA+sbaeuJ50VXJCiEiYiIyDjx1NZWiiIhzjimLuhSAIUwERERGSe2tPQwc0Ip4ZAFXQqgECYiIiLjxNY9PcyeWB50GYMUwkRERGTMc3e27ulhVm1Z0KUMUggTERGRMa+5q5+e/iSzJyqEiYiIiOTM1j3dAAphIiIiIrm0paUHgFm1GhMmIiIikjNbWnowg5m1pUGXMkghTERERMa8rXt6mFpVQnEkHHQpgxTCREREZMzb0tLNrDwaDwYKYSIiIjIObN3Tw+w8Gg8GCmEiIiIyxnXFEjR39aslTERERCSXtmaujMyn6SkgiyHMzGaa2UNmts7M1prZJ4c5xszsejPbaGZrzOzkbNUjIiIi49PgHGF51h0ZyeJrJ4DPuPsqM6sEVprZ/e6+bsgx5wHzMrdXA9/PfBUREREZFYNzhI2XljB33+nuqzL3O4H1wPR9DrsIuNXTlgM1ZjY1WzWJiIjI+LNlTw81ZVGqS6NBl7KXnIwJM7M5wEnA4/vsmg5sG/K4gVcGNczsajNbYWYrmpqaslWmiIiIjEFbW3qYnUcLdw8YcXekmb0ZOA4oGdjm7l8ZwfMqgF8Dn3L3jsMp0t1vBG4EWLJkiR/Oa4iIiMj4tGVPNyfOnBB0Ga8wopYwM/sBcBnwccCAtwOzR/C8KOkAdpu7/2aYQ7YDM4c8npHZJiIiInLEEskUO9r6mJVHyxUNGGl35Gvc/d1Aq7tfC5wOzD/QE8zMgB8C6939uv0cdjfw7sxVkkuBdnffOcKaRERERA6opbufZMqZUp1/IWyk3ZG9ma89ZjYNaAEONoD+DOBdwDNm9nRm2xeAWQDu/gPgXuB8YCPQA7xv5KWLiIiIHFhTZwyASZXFAVfySiMNYfeYWQ3wTWAV4MBNB3qCu/+NdNflgY5x4KMjrEFERETkkDR29gFQX6ghzN2/mrn7azO7Byhx9/bslSUiIiJy5AZawuor8i+EjXRg/qqB++4eUwATERGRQtDYkQlhedgSNtKB+QfsVhQRERHJR01dMapKIpREw0GX8gojHRO2wMzWDHlspId0HZ+FmkRERERGRVNnjElVJQc/MAAjDWGbgbdksxARERGR0dbYGcvL8WAw8hDW7+5bslqJiIiIyChr6oxx0qyaoMsY1kjHhH08q1WIiIiIjDJ3p7GzL29bwkYawp4xs28PLKJtZv9lZtVZrUxERETkCHTFEvTFU0yqKuwQdgvQAbwjc+sAfpStokRERESOVGNn/k5PASMfE3a0u18y5PG1Q5YiEhEREck7L0/Ump9XR460JazXzF478MDMzuDl9SRFRERE8s7gupF52h050pawDwM/GTIOrBV4T3ZKEhERETlyjXm8ZBGMPITtcvcTzKwKwN07sliTiIiIyBFr6owRDRs1ZdGgSxnWSLsj74V0+FIAExERkUIwMD2FWX6uvjjSECYiIiJSUJo6Y9Tn6ZJFMPLuyOPNbGgL2MDakVVZqElERETkiDV1xpgxoSzoMvZrxJO1unvVkFulApiIiIjks6bOWN7OEQbqjhQREZExKJ5M0dLdz6QxEMIuOfghIiIiIvmhpasfyN/Z8mHkIexLZja4BLmZTTCzW7JUk4iIiMgRGZyodQyEsOPdvW3ggbu3AidlpyQRERGRI9PY2QeMjZawkJlNGHhgZrWM/MpKERERkZx6ecmiwp+i4r+Ax8zsV6Snp7gU+FrWqhIRERE5AgNLFtVVFAVcyf6NKIS5+61mthI4K7Ppbe6+LntliYiIiBy+5q4Y1aVRiiPhoEvZrxF3Kbr7WjNrAkoAzGyWu2/NWmUiIiIih6mlqz+vW8FghGPCzOxCM3sB2Az8BXgJ+EMW6xIRERE5bM1dMSZW5O+gfBj5wPyvAkuB5919LnA2sDxrVYmIiIgcgZbuMdISBsTdvYX0VZIhd38IWJLFukREREQOW0tXjInl+d0SNtIxYW1mVgE8AtxmZo1Ad/bKEhERETk8iWSK1p44E8dIS9iFQA/wKeCPwEbggmwVJSIiInK49vSklyzK9zFhB2wJM7PNgO+7OfP108BRB3juLaSDWqO7Lx5m/zLgt6QH+wP8xt2/MrKyRURERIY3sG5kXXl+t4QdrDty6LgvAx7k5bnCDubHwA3ArQc45hF3V4uaiIiIjJqBEFbQLWGZwfiDzCyx77YDPPevZjbn8EsTEREROXQt3enZ8mvzvCVspGPCMLOjeLkrcrScbmarzewPZnbcAd77ajNbYWYrmpqaRrkEERERGUuaB7oj83xg/sHGhD1DekxYMVAGfGgU33sVMNvdu8zsfOAuYN5wB7r7jcCNAEuWLNl3jJqIiIjIoJauGJGQUVUSDbqUAzrYmLCB8Vp97r57NN/Y3TuG3L/XzL5nZnXu3jya7yMiIiLjS0tXP7XlRYRCo92BN7oONiZsS7be2MymALvd3c3sNNJdoyMabyYiIiKyPy3d+b9kERzCAt6HysxuB5YBdWbWAHwJiAK4+w+AS4EPm1kC6AUud3d1NYqIiMgRaS6AxbshiyHM3a84yP4bSE9hISIiIjJqWrpjzJlYFnQZBzXiqyNFRERECkFLV39BdEcqhImIiMiY0dOfoKc/mffrRoJCmIiIiIwhLy9ZpJYwERERkZxp6R5YskgtYSIiIiI509KVXrJIY8JEREREcmhw8e48XzcSFMJERERkDGnuHmgJUwgTERERyZk9Xf2UFYUpK8raVKijRiFMRERExoyW7v6CaAUDhTAREREZQ5q7YkwsgOkpQCFMRERExpCWAlk3EhTCREREZAxp6VZLmIiIiEhOuXtm3Ui1hImIiIjkTEdvgkTKC2KiVlAIExERkTGiKTNbvsaEiYiIiORQcyaE1aslTERERCR3mjozLWGVCmEiIiIiOdM82B2pECYiIiKSM81dMcIho6Y0GnQpI6IQJiIiImNCU2eMuooiQiELupQRUQgTERGRMaG5q79guiJBIUxERETGiOaumEKYiIiISK41dyqEiYiIiOSUu9Pc1U99gUxPAQphIiIiMgZ09CboT6YKZrZ8UAgTERGRMWBgySK1hImIiMiYNHPWbMws726Ll5wOwOUXv3nEz5k5a3agP8tIoO8uIiIiBaVh21auu29D0GW8wvO7O/nDs7v4zLd/ysQRDs6/5o0LslzVgaklTERERApeT38SgLKiwmlfyloIM7NbzKzRzJ7dz34zs+vNbKOZrTGzk7NVi4iIiIxtPf0JzKAkWjjtS9ms9MfAuQfYfx4wL3O7Gvh+FmsRERGRMaynP0lZNIxZYSxZBFkMYe7+V2DPAQ65CLjV05YDNWY2NVv1iIiIyNjVHUtQVlw4XZEQ7Jiw6cC2IY8bMttEREREDklPf5KyonDQZRySgug4NbOrzWyFma1oamoKuhwRERHJMwphh2Y7MHPI4xmZba/g7je6+xJ3X1JfX5+T4kRERKQwuDu9/cmCujISgg1hdwPvzlwluRRod/edAdYjIiIiBSiWSJF0L7iWsKxFRjO7HVgG1JlZA/AlIArg7j8A7gXOBzYCPcD7slWLiIiIjF0vzxGmEAaAu19xkP0OfDRb7y8iIiLjQ09/AiisiVqhQAbmi4iIiOxPobaEKYSJiIhIQRsIYeVqCRMRERHJnUJcsggUwkRERKTAFeKSRaAQJiIiIgWuK5agvMCWLAKFMBERESlw3bEEFQphIiIiIrnV1acQJiIiIpJT8WSKvkSKihKFMBEREZGc6YqlJ2pVS5iIiIhIDnX1KYSJiIiI5NxgS5i6I0VERERyR92RIiIiIgHo6ktQEgkRDRdepCm8ikVEREQyumKJguyKBIUwERERKWBdBTpRKyiEiYiISAHrLNCJWkEhTERERApUIpWiN55Ud6SIiIhILnXHkkBhXhkJCmEiIiJSoAp5olZQCBMREZECVchzhIFCmIiIiBSoQp4tHxTCREREpEB19SUoCocojoSDLuWwKISJiIhIQeqMxQu2KxIUwkRERKRAFfJs+aAQJiIiIgWqO5ZUS9hY0tEX570/eoJnt7cHXYqIiIjsRyrldBfwkkWgEPYKW1t6WLejg4u/+yg3PPgCiWQq6JJERERkH939CZzCvTISFMJeYfH0au779Os4d/EUvnXf83zg1hW4e9BliYiIyBCdBT5RKyiEDaumrIgbrjyZz523kIc3NPHQhsagSxIREZEh2nrjANSURQOu5PAphB3A+187l9kTy/jWn54nlVJrmIiISL5o6+nHDKpKFMKGZWbnmtkGM9toZp8bZv97zazJzJ7O3D6QzXoOVTQc4lNvmMe6nR384dldQZcjIiIiGa09capLooRDFnQphy1rIczMwsB3gfOARcAVZrZomEPvcPcTM7ebs1XP4brwhOnMm1TBf92/QYP0RURE8kRbT39Bd0VCdlvCTgM2uvuL7t4P/AK4KIvvlxXhkPGZN87nxaZufv/MzqDLERERGffcnbaeODVlRUGXckSyGcKmA9uGPG7IbNvXJWa2xszuNLOZWaznsL3puCnMrC3lVysagi5FRERk3OuOJUmknAlqCTsivwPmuPvxwP3AT4Y7yMyuNrMVZraiqakppwVm3p+3njidRzc1s6u9L+fvLyIiIi9r7ekHUEvYAWwHhrZszchsG+TuLe4eyzy8GThluBdy9xvdfYm7L6mvr89KsQfz1pNn4A53Pb394AeLiIhI1rT1FP70FJDdEPYkMM/M5ppZEXA5cPfQA8xs6pCHFwLrs1jPEZlbV86JM2v431XbNXmriIhIgNp6+wmHjMoCnqgVshjC3D0BfAz4E+lw9Ut3X2tmXzGzCzOHfcLM1prZauATwHuzVc9ouOTk6WzY3cm6nR1BlyIiIjJutfbEqSmNYla401NAlseEufu97j7f3Y92969ltn3R3e/O3P+8ux/n7ie4+1nu/lw26zlSFxw/jWjY+N9V6pIUEREJyliYngKCH5hfUCaUF7FswSR+u3oHSc2gLyIiknOplNPeW/jTU4BC2CG78IRpNHXGWPHSnqBLERERGXc6+uKknIKfngIUwg7ZWQsnURwJaRkjERGRAAwu3F2qlrBxp6I4wuvm1/PHZ3dpUW8REZEcGyvTU4BC2GE5/1VT2NXRx1Pb2oIuRUREZFxp6+mnKByirCgcdClHTCHsMJx97GSiYeMPWktSREQkp1p74tSUFf70FKAQdliqSqKcOa+ePzy7SxO3ioiI5Ii709jRR31lcdCljAqFsMN03uIpbG/r5Znt7UGXIiIiMi609cbpS6SYUlUSdCmjQiHsMJ2zaDKRkPH7NeqSFBGRA5s5azZmNiZuQdrV3gfA1OqxEcIKe9GlANWUFfHaeXXcs2YnnztvYeD/MEVEJH81bNvKdfdtOKTnuDs9/Um6Ygm6Y4nM1/Tj3niSeDJFIumYQSRkRDOD1cuKI1SVRKgtL6K2rIji6OgOYL/mjQtG9fUOxc72PorCIWrLC396ClAIOyIXnjCNa365mlVbWzlldm3Q5YiISB5ydyI1U3h2Rzu7O/ro7U/S058cXHll4P/hDSPlTjyZIp50evoT7DsTkgFlxWHKiiLp4BUx3CGRcnrjcXZ19NHTn9zrOTVlUaZWlTBtQilzJpZTUcCLXu/q6GNydfGYafgo3E8iD5yzaDLFkRC/W71TIUxERPbS0RfnzhUN/HT5FqZ/6GYeWN9IcSRERXGE0qIwxZH0iCDP/MeBkEE0HB1s1aoojlBREqG8OEJFcYSyaJhQ6MABJJVyOvri7Onup6W7n13tfbzU0sP6XZ0A1FcWM39SBQumVFJZUjhzbcWTKZq7Ypw6hv7eKoQdgcqSKK9fOIl71uzk3y5YRPggJ4aIiASvL57k4Q1N3L16O4+/uAcHwiHjqLpyzls8hXMXT2XKEYw56uiL8z9/2cSPH32J7v4kp8yewIqff5NPffHrTMjB1AqhkFFTVkRNWRFH1ae3uTvNXf281NLN5uZuHt3UwqObWphVW8aJM2uYM7Es71uXdnf04c4RfTb5RiFsP2bOmk3Dtq0HPa5swRnUX/x5Ko46ib4tq3NQ2aGZMXMW27ZuCboMERknevuTbNjdyfbWXlq6Y5w6p5aFUyrz5g/8n9bu4l/vepamzhh1FUW8fuEkSqJh4skUq7a28uXfrePae9Zx/uKpfHjZ0SyeXj3i1+6LJ/nZ8i3c8NBG2nrivOWEaXzwzLkcP6MG+8i91JZ/O4vf2YGZGfWVxdRXFnPqnFraevp5blcnz+5o5+7VO5hQFmXJ7PRndbCWtqAMDMpXCBsHRjqIMpFMceMjL/L6a77DG46dnIPKDk2QAyhFZPxIpZxfr2rgG3/aQFNnbK998ydXcOVps3jX6XMC6zFo7e7ni3ev5Xerd3Ds1Cq+eenxvPaYOiLhvScJ2NjYyZ0rt3Pb8i38/pmdnDmvjo8sO4alR9XuN0gmU85dT23nuvufZ3tbL6+bX89n37TgkAJcrtWUFbH0qImcOqeWFxo7WbW1jfvX7+bJLXtYOnci8ydX5E1wHrCro4+a0iilo3yhQZAUwo5QJBzi6PoKNjZ2sWxBPZGQZv0QkfFl254ePvrzVaxpaOekWTV85cLjmD2xnKrSCA9taOJ/VzXw5d+t409rd/Pty07MeUvGxsYu3v+TJ9nR1ss158znw8uOJhoe/nf1MZMq+dx5C/nIWUfzs+VbuOVvm7nipuWcMLOGc4+bwmlza5kzsYyOvgS7O/q4b+1u7lmzg8bOGMfPqOablx7Pa46py+n3dyTCIWPhlCoWTK5kU1M3yze38Me1u1jdUMLr5tfnzXxc7s7O9j5m1ZYFXcqoUggbBQunVPLcrk42NnaxcEpV0OWIyAi5O09ta+Ol5m7coTgaYtmCSQV99ViubWnp5sqbHqcrluD/XXYiF54wba/urHctnc1Vr57Fr1Y28OW713Luf/+V71xxEmfOq89JfY9ubObDP1tJNBziF1efzimzJ4zoeVUlUT6y7Bj+8Yy53LmygVsfe4mv//G5VxxXFA5x1sJ6Ljl5Bucsmpx3rUcjZWYcM6mCo+vLWbezg79vauGOJ7exeFoVrz2mbtSnuThUnX0JevqTY6orEhTCRsWs2jJqSqOsaWhXCBMpAO7OA+sb+f5fNrFyS+te+6pKIrxz6Wzed8YcJlWOrV/4o21zczdX3rSc3niS2z7w6v12v5kZ71gykyWzJ/CR21bxvh89yTfffjxvPWlGVuu7/Ymt/Ntdz3JUfTk/fM+pzDyMVpSSaJirls7mqqWzae6KseKlPexq76OqNEpNWZRTZtVSXVY4VxgejJlx3LRqjplUwROb9/DU1jY2N3dz1sJJHF1fEVhdW1t7AJhWXRpYDdmgEDYKzIxXzajmkReaaeqMjZk1rUTGot7+JNN2uIAAABWsSURBVP9y52ruWbOT6TWlXHvhcbxufj1hM3Z19PHjv2/mf/6yiTue3Mb33nkyS4+aGHTJeWlTUxdX3rSceNL5+QeWsmjawf8H9Kj6Cn75T6fzoVtX8uk7VtPUGeODZx416q1HyZTzn/eu5+a/beYf5tdzw5UnjcpUDHUVxZy7eOooVJj/iiNhzpxXz/zJlfx5/W7uWbOTYyZVsGx+blow97VhZyc1pVHqKsbGJK0DFMJGyaKpVTy2qYU1DW2cnYcD9EUkfXXVB29dwbM72vnsuQv44JlH7TU2aNbEMk6bW8sLuzv5p5+t5KqbH+dLb1nEVUtnF2w3UzZsbOziipuWk0o5t39wKQumVI74uVUlUX78j6dyzS9X8x/3Pseu9hj/+uZjR+2KvD3d/Xz6jqf5y/NNvOf02fzbBYteMfheRm5yVQmXnzqLVVtbeXzzHrbt6aF88dm4e87OiY7eOA1tvZx+1MQxdx7qX+YoKYmGmT85PTYslkge/AkiklNbWrq5+LuP8mJTFze9awkfWXbMfgdnz5tcyV0fPYN/mF/Pv/12Ld/80wbcfdhjx5vnd3dy+Y3LcYdfXH1oAWxAcSTMdy4/ife+Zg63PLqZT/ziqVH5vfnU1lYuuP4RHtvUwtfeuphrL1qsADYKwiHj1Dm1vPPVs6irKKbuzZ/mt0/voKMvnpP3f253epLZhYfxby3f6V/nKDp+RjWJlLNuR0fQpYjIEA2tPVx50+P0JZL88p9O5w2LDt5aXVkS5aZ3L+GK02bxvYc38V/3PV8wQSxbi0UX1c/h7H+/m107d7D62+9j/pSqw36tcDjEtRctpvWhH3HPmp3Mef/1RGunv+K4SLTooK8VKiplwrL3cfF3/srWLS+x5Ycf56qlcwJf6DpfFr0eLRPKirjk5Om03Pc9drT38rPlW1jT0JbV88LdeW5nB9NrSqkqHTtj7waoO3IUTa4qYVp1CSu2tHLctGqKIsq4IkHb2d7LFTctp7Mvzs8/uJTjpo187qZQyPjaxYsB54aHNmIG15wzP+//qB7OYtEH09QZ4zdPNRAOGZecPIMJFz8waq+9sbGLP0dCVHz4Jl43r55F06oIZX7G17xxwX6/l2TKeX53J3/f1EJXLMGiqVWcOW8+JZfcO2q1jZaxMmejmdH11L1c9eVv8OfndvPQhiZeaOzi7IWTqCkb/fFauztjtPbEOXnWyK5qLTQKYaPstfPq+OWKBlZubeV0DegVCVRjRx9X3vQ4rd1xfnaAq/cOJB3EXoU7fOfBjRjw6QIIYqNpc3M3f1y7i6JwiEtOnj7qf2yPmVTB5Kpi7lu3mweea2TFllZOmlXDwsmv7H5KpZymrhibm7t5Zns7Pf1J6iqKOG/xDKbVjK0r5/JZVWmUt544nbU7O3jk+WZue3wrrzl6IifMrBkM0KPhuZ0dhEPGvMnBXZmZTQpho2xqdSnzJlWwaksrr5pWTUWJfsQiQWjuinHlzY+zu6OPn77/NE6cWXPYrxUKGf/x1nQQu/7BjWDGNefMH8Vq85O78+RLrTz2Ygv1FcVccPzUrHUJVZZEedtJ09nY1MWqLW08vKGJhzc0Mf1DN3PX09tJuRNPOHu6++lPpgCYPbGMk2bWMKs2/9c9HIvMjMXTqpldW8aDzzXy1xeaeaGxi7MWTBqVWQLae+Os29nB0fXlFEfGziz5QykhZMEZx9SxqamLx15s4ZwRjD0RkdHV2NHHu295gobWHn78vtM4ZXbtQZ8zsvVijYnnfZzrga/+59dpe+hHQHDjxEIlFRRNnU9R/WzClfWEKycSihYz6dIv87vVOygvjlBZEqGmNMqkqhKqSiIjCivuzrbWXv6+qZndHTEWTKnk7IWT9nshw2gxM+ZNquSY+gp2dfTR0NrL/ff8ld6ZswmHjOJIiPlTKphRU8aMCaWUa1LdvFBZEuXCE6axYVcnf3mhiduf2Mri6dWcfvTEw15iKOXOfet2YRhnFNAKBIdK/4KzoLo0ygkza3hqaxsLp1Qe1gSBInJ4Xmzq4t23PMGe7n5ufvepI57na6TjqNydh59vYg1v45QL3s2bFk3O2RV4sUSShtZetu7pYeueHtp6Xr46rSgcoqI4QjRibN3wDB19cXa099IXTw0eUxwJDS7iXFdeTHVplMrSCCEzUimnsy9BQ1sPW1p62NneR2VJhDcumpzzBbjNjKnVpUytLuX2u7/BFz72/py9txweM2Ph1Crm1JWz/MUW1mxvZ8OuTk6aVcNJs2oOuSXr6a1t7Gjr45xFk6kahTne8pVCWJYsnTuRrS093PvMTi47dWZWBizK+ODubGrq4pnt7TTs6aWhtZd4KkU0FKKsOMzR9RXMm1TB4unV475lYOWWPXzw1pUA3P7BpZxwBF2Q+2NmLJtfT3VJlEc2NtPZF+dNi6YwoXz0z/Fkytnd0ce2TOja2dGHO0TDxvSaUo6bWsXkqhLqK4spGdLicM3/PYfPZgJlPJliT3c/jZ0xmjpjNHb2saahnWRq/y14dRVF/MP8ehZPr9J6uHJISqJhli2YxKumV/PYiy08vnkPq7e18aoZ1bxqevWIJs1taO3h75taOKqunGPH4LQUQ43v39hZVBQJ8ZYTpvGLJ7byu9U7ecepM8Zsn7aMvvbeOA9vaOT+dbtZ/mILzV39g/vqKoopiYaIJ1OD66lB+g/zaXNrOWvBJJYtqOfo+opxM04mnkxx/QMv8N2HNjJjQhk/+cfTmFtXnrX3MzNOnj2BqtIof16/m9ue2Mqr59Zy8qwJhI9g0lH39JinbZnWru2tvYPjnyZVFrNk9gRm1ZYxtbp0xO8TDYeYXFXC5CELMadSTkdfnPbeOJ19CRwIm1ESDTGtpnSvQCdyOCZWFHPB8dPY3dHHky/tYcVLrazY0srcieUcXV/B3LpySov2/ncWSyR5dGMLz2xvp6okwtnHThrzv8MUwrKoujTK+a+ayl1Pb+e3T+/gvMVTRmXpDBl9/YkU8WSKRMoJh4zSaPiI/pgejpeau3nwuUYefK6R5S+2kEg5dRXFnDmvnqVHpf/Az6wt2+sPpLuzs72PDbs7Wb6phYc2NPLvv1/Pv/9+PTMmlHLWgkmctbCe04+qe8UvvCPl7vT0J2nvjRMNhygrClOS459bKuU8+Fwj193/POt2dnDpKTP40lsW5ew8O2ZSBVOrS3h4QxN/39TC6oY2jptWzXHTqg7aheLudMUS7O6Isbujj10dfTR2xAZDV3VplPlTKpg1oYwZtWWHPbZmOKGQUVNWpBZ6ybrJVSVccPw0OnrjrGloZ8PuTl5s7gbS67RWl0YpioRo743T2hMnlXJOmlXD0rkTx8U0T1kNYWZ2LvDfQBi42d3/v332FwO3AqcALcBl7v5SNmvKtZm1ZZx73BTuX7+bnz+xlTcumpLV/0MvRMmUE0+m6E+miCdSxJPpx7FMMAKIhIxIOJT5ahSFQxRHw5REQsOOxxkaEAZuzV0xGjti7O7soynzdXdHjMaOPjr6Eq94jdJomMlVxemxKTUlTKsuZVpNKdMnlDJjQinTj7DFoLkrxjPb2/n7xmYeeK6RF5vSv5jmTarg/WfO5U3HTeHEGTUHXM7FzJhWk67rrAWT+Pz5x7K9rZeHNzTy0HON3LmygZ8u30JRJMTx06s5efYEFk9PX800s7aMqpLI4M8vlXJiiRStPf20dPXT0h17+Wt3ZlvXkPvdsb3GGw0oiqQDWU1plAnlRdSWFVFbnr4NfTwhs622rIiq0pENGIf0Z7uxsYtHXmjmF09u5fndXUyvKeUHV53CuYunHMYncWTKiyO8+fipvNTSzdPb2nhi8x6e2LyHiuIIkyqLqSiOEAmnJ+yMxZP0xpN09iVo64kPBq6QpVs4F06pZHJ1CTPG6MSUMn5VlUZ57bw6zjhmIo2dMV5q6WZPdz/tvXG6YglqyoqYWVvGgsmVe7XajnVZC2FmFga+C5wDNABPmtnd7r5uyGHvB1rd/Rgzuxz4OnBZtmoKyrzJldRVFHPvszu5e/UOZtSUcvyMao6qr8h5a8u+3NN/eHv7k/Qlkumv8RR9iSR9g9tS9MVf3j94fOYPSl88RSyRJJ58OUANtCwNPk5mHid8MGwNbDvA0JSRfQ/JBJ7ox1MJcIfMDNoWHv6PmCfjJLv2kOxqJdHVQrKrlVRP2+BrWCiMRUvoKCqjuXIiz1XWEamqI1wxEQvtHboSnS0kO5pI9rSR6uva6+bJzKBpCxEqLiNUVEa4opZI9WQitdOIVKav+PFEnL5tz9C78Ql6Nz3Jlvbd/Bn4wpH9WNLCUUpmLqZ07sk8Ov1Yntx0NBbZ++fiyTieShKK7v8Xnyf6Sfa0kexuJ9XbTrK7nWRPG5VFIb7x1S8ST6bojaf/rfTGk/T0p0NGa08/uzr6WLezg5bufvoTrwxtkA7ZNWVF1JZHmVBWxMSKIiaUFVFVGiWVcuJJp7Wnn+2tvWxu6aapMwaklzG57h0n8JYTpmX9yr2DmTOxnDkTy2nvjbOpqYvGjvT4qx1tvSRSTsqdkmi6tbCiOMKxU0uYUFbEpKpi6iuKtbyOjAtm9oru8fEsmy1hpwEb3f1FADP7BXARMDSEXQR8OXP/TuAGMzMvlLVBDsGE8iIuWzKTpxvaeKahnXuf3UU0bNRXFjO5soSq0ihlRWGKIyHCocxSHMbgpHcpd1IpSLqTSvleXxMpJ5EJO4mkE0+lBu/XXfx53nPLE5k/kMm9vvb0J4ZtyRiJkKUHYJZm/qgURUIUhUNEI0Y0nL5fXhwhGg4RDb+8LTr0mMiQbZnjiiL7PA6HMIN40rniyndyxWe/ScqdZCrzfadS6fvJ9M9iQLqlLERJJP0zLY6GKSsKU14coSQSOqxxBqmU092foKMvQWdvnI6+BB19VXT0zhgMo/2ZcDkcMygrClNVEuWFR+/lvIvfwaSqYiZVllAUWUQu/v8jkUzR1hunI9M6OBCSU+5EMy2NJdEwpUXpn1dpNExZUYRoePilV65507Fccd8PR/z+Fi0mVFpFuKw6/bW0ilBZFeHSalrLqtgyuK06/bWkHE+lIJUk1ddJor2RRHsjse3r6N38FFs6GvnTaP6ARkF1aXTMzu4tIqMrmyFsOrBtyOMG4NX7O8bdE2bWDkwEmrNYV2Ai4RBLZqfH9mxp6WFLSze7O2Ks2X7gK5UOVTRsREKZ8DNxJvc9/CieiJGK9+HxGJ75mor3pVt/4jE8ERu8nxpyf9/9qcx9kq/svsuFRdOqAnlfSI+jqSyJpscbHWBm7oFuvYFQaKS76CKhl4PMNV+4jpM/+aFclL2XSDiUXoC34sgnUgTAU6O+PE5QxsqyMiJSOCxbjU5mdilwrrt/IPP4XcCr3f1jQ455NnNMQ+bxpswxzfu81tXA1ZmHC4Bc/NavY4yGwQKmzyT/6DPJT/pc8o8+k/yUi89ltrvXD7cjmy1h24GZQx7PyGwb7pgGM4sA1aQH6O/F3W8EbsxSncMysxXuviSX7ykHps8k/+gzyU/6XPKPPpP8FPTnks2RoE8C88xsrpkVAZcDd+9zzN3AezL3LwUeHIvjwURERET2lbWWsMwYr48BfyI9RcUt7r7WzL4CrHD3u4EfAj81s43AHtJBTURERGTMy+o8Ye5+L3DvPtu+OOR+H/D2bNZwBHLa/Skjos8k/+gzyU/6XPKPPpP8FOjnkrWB+SIiIiKyf5odUERERCQACmH7MLNzzWyDmW00s88FXc94ZGYzzewhM1tnZmvN7JOZ7bVmdr+ZvZD5qhkxA2BmYTN7yszuyTyea2aPZ86ZOzIX4kiOmFmNmd1pZs+Z2XozO13nSvDM7NOZ31/PmtntZlaicyX3zOwWM2vMTIk1sG3Y88PSrs98PmvM7ORs16cQNsSQpZbOAxYBV5jZomCrGpcSwGfcfRGwFPho5nP4HPCAu88DHsg8ltz7JLB+yOOvA99292OAVtLLkUnu/DfwR3dfCJxA+rPRuRIgM5sOfAJY4u6LSV+cNrA0n86V3PoxcO4+2/Z3fpwHzMvcrga+n+3iFML2NrjUkrv3AwNLLUkOuftOd1+Vud9J+o/KdNKfxU8yh/0EuDiYCscvM5sBvBm4OfPYgNeTXnYM9LnklJlVA68jfaU57t7v7m3oXMkHEaA0MwdmGbATnSs55+5/JT37wlD7Oz8uAm71tOVAjZlNzWZ9CmF7G26ppekB1SKAmc0BTgIeBya7+87Mrl3A5IDKGs/+H/BZYGCBzIlAm7sPrGOlcya35gJNwI8yXcQ3m1k5OlcC5e7bgW8BW0mHr3ZgJTpX8sX+zo+cZwCFMMlbZlYB/Br4lLt3DN2XmdRXl/bmkJldADS6+8qga5FBEeBk4PvufhLQzT5djzpXci8zxugi0iF5GlDOK7vEJA8EfX4ohO1tJEstSQ6YWZR0ALvN3X+T2bx7oGk487UxqPrGqTOAC83sJdJd9a8nPR6pJtPlAjpncq0BaHD3xzOP7yQdynSuBOsNwGZ3b3L3OPAb0uePzpX8sL/zI+cZQCFsbyNZakmyLDPO6IfAene/bsiuoctcvQf4ba5rG8/c/fPuPsPd55A+Nx5093cCD5Fedgz0ueSUu+8CtpnZgsyms4F16FwJ2lZgqZmVZX6fDXwuOlfyw/7Oj7uBd2euklwKtA/ptswKTda6DzM7n/S4l4Gllr4WcEnjjpm9FngEeIaXxx59gfS4sF8Cs4AtwDvcfd8Bl5IDZrYM+Gd3v8DMjiLdMlYLPAVc5e6xIOsbT8zsRNIXShQBLwLvI/0/2DpXAmRm1wKXkb7a+yngA6THF+lcySEzux1YBtQBu4EvAXcxzPmRCcw3kO467gHe5+4rslqfQpiIiIhI7qk7UkRERCQACmEiIiIiAVAIExEREQmAQpiIiIhIABTCRERERAKgECYi44KZuZn9bMjjiJk1mdk9mceXmNlaM3vEzCZmth1tZncEVbOIjG0KYSIyXnQDi82sNPP4HPaeDfvjwKnA/wBXZrb9O/CvOatQRMYVhTARGU/uBd6cuX8FcPuQfSmgGCgD4mZ2JrDL3V/IbYkiMl4ohInIePIL4HIzKwGOJ70Kw4D/BP4MvIV0OPs34Ks5r1BExg2FMBEZN9x9DTCHdCvYvfvsu9/dT3H3twAXZfbPN7M7zewmMyvLecEiMqYphInIeHM38C327ooclAlb7wW+C1xLeoHfvwHvzFF9IjJORIIuQEQkx24B2tz9mcxC5Pv6F+B6d49nBvE76fFiagkTkVGlECYi44q7NwDXD7fPzKYBp7n7tZlN3wGeBNqAi3NToYiMF+buQdcgIiIiMu5oTJiIiIhIABTCRERERAKgECYiIiISAIUwERERkQAohImIiIgEQCFMREREJAAKYSIiIiIBUAgTERERCcD/D3WkSxDQujPfAAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 720x360 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmEAAAFNCAYAAABIc7ibAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXzcdb3v8ddntuxLk6ZtmqYrpdDWQqEUEBdAQVQEvQLCPUfFDZfjfvRc9Xrdjp57Pa4HN0RB8aCoBzlaEVAQFDxHlrSU7qWF0r1N2uzNOpnP/WMmJcS0nbb5zW+SvJ+PxzwyvyUzn2Tym7zn+/3+vj9zd0REREQktyJhFyAiIiIyESmEiYiIiIRAIUxEREQkBAphIiIiIiFQCBMREREJgUKYiIiISAgUwkREjpOZPWdmr8zc/5yZ3R52TSIy9iiEici4YmazzeweM2sxs31m9m0zi4Vdl4jIcAphIjLefBdoBGqBM4GXA+8LtSIRkREohInIeDMH+KW797j7PuA+YNFIO5pZlZn9yMz2ZFrOfj1k2+VmttrMWs3sv81sSY7qF5EJQiFMRMabbwLXmlmxmdUBryYdxEby70Ax6ZA2BfgGgJktBW4F3g1UA98HVphZQcC1i8gEMiZDmJndamaNZrYuy/2vMbMNZrbezH4WdH0iEqqHSYeqdmAX0AD8evhOZlZLOqC9x91b3L3f3f+c2XwD8H13f8zdB9z9NqAXOC8nP4GITAhjMoQBPwYuy2ZHM5sPfBK4wN0XAR8OsC4RCZGZRUi3et0FlACTgUnAl83sJjPrzNw+BdQDze7eMsJDzQL+MdMV2WpmrZn9p+fmJxGRiWBMnjHk7g+b2eyh68xsHvAdoAboAt7l7puAdwHfGXyjdffG3FYrIjlUBcwEvu3uvUCvmf0I+KK7LwbeM7hjpiWsyswq3b112OPsBL7k7l/KVeEiMvGM1ZawkdwMfMDdzwY+RvoMKYBTgVPN7L/M7FEzy6oFTUTGHnc/AGwD3mtmMTOrBN4KrBlh373AvcB3zWySmcXN7GWZzT8A3mNm51paiZm91szKcvWziMj4Ny5CmJmVAi8G/sPMVpMeRFub2RwD5gMXAtcBP8i8MYvI+PQ/SA9XaAK2Av3AR46w75sz2zeRntbiwwDu3kC6Ff3bQEvmca4PsmgRmXjM3cOu4YRkuiPvdvfFZlYObHb32hH2uwl4zN1/lFn+I/AJd38il/WKiIiIDDUuWsLcvR3YZmZXA2S6D87IbP416VYwzGwy6e7JZ8OoU0RERGRQYCHMzArN7HEzeyozNcTnR9jnejNrykyIuNrM3pnlY98B/BVYYGa7zOwdwN8B7zCzp4D1wJWZ3X8PHDSzDcBDwMfd/eBo/IwiIiIiJyqw7kgzM6DE3TvNLA78BfiQuz86ZJ/rgWXu/v5AihARERHJU4FNUeHpdNeZWYxnbmNzAJqIiIjIKAt0TJiZRTNnKzYC97v7YyPs9kYzW2Nmd5pZfZD1iIiIiOSLnJwdmZkS4j9Jz+O1bsj6aqDT3XvN7N3Am9z94hG+/wbSlxGhpKTk7NNOOy3wmkVERERO1sqVKw+4e81I23I2RYWZfQbocvevHmF7lPQlRCqO9jjLli3zhoaGIEoUERERGVVmttLdl420LcizI2sGJ0U1syLgEtITIg7dZ+i8XlcAG4OqR0RERCSfBHntyFrgtkwLVwT4pbvfbWZfABrcfQXwQTO7AkgCzWhGahEREZkgxtyM+eqOFBERkbEilO5IERERETkyhTARERGRECiEiYiIiIRAIUxEREQkBAphIiIiIiFQCBMREREJgUKYiIiISAgUwkRERERCoBAmIiIiEgKFMBEREZEQKISJiIiIhEAhTERERCQECmEiIiIiIVAIExEREQmBQpiIiIhICBTCREREREKgECYiIiISAoUwERERkRAohImIiIiEQCFMREREJAQKYSIiIiIhUAgTERERCYFCmIiIiEgIFMJEREREQqAQJiIiIhIChTARERGRECiEiYiIiIRAIUxEREQkBAphIiIiIiFQCBMREREJQWAhzMwKzexxM3vKzNab2edH2KfAzH5hZlvN7DEzmx1UPSIiIiL5JMiWsF7gYnc/AzgTuMzMzhu2zzuAFnc/BfgG8OUA6xERERHJG4GFME/rzCzGMzcfttuVwG2Z+3cCrzAzC6omERERkXwR6JgwM4ua2WqgEbjf3R8btksdsBPA3ZNAG1AdZE0iIiIi+SDQEObuA+5+JjADWG5mi0/kcczsBjNrMLOGpqam0S1SREREJAQ5OTvS3VuBh4DLhm3aDdQDmFkMqAAOjvD9N7v7MndfVlNTE3S5IiIiIoEL8uzIGjOrzNwvAi4BNg3bbQXw1sz9q4AH3X34uDERERGRcScW4GPXAreZWZR02Pulu99tZl8AGtx9BXAL8O9mthVoBq4NsB4RERGRvBFYCHP3NcDSEdZ/Zsj9HuDqoGoQERERyVeaMV9EREQkBAphIiIiIiFQCBMREREJgUKYiIiISAgUwkRERERCoBAmIiIiEgKFMBEREZEQKISJiIiIhEAhTERERCQECmEiIiIiIVAIExEREQmBQpiIiIhICBTCREREREKgECYiIiISAoUwERERkRAohImIiIiEQCFMREREJAQKYSIiIiIhUAgTERERCYFCmIiIiEgIFMJEREREQqAQJiIiIhIChTARERGRECiEiYiIiIRAIUxEREQkBAphIiIiIiFQCBMREREJgUKYiIiISAgUwkRERERCoBAmIiIiEoLAQpiZ1ZvZQ2a2wczWm9mHRtjnQjNrM7PVmdtngqpHREREJJ/EAnzsJPCP7r7KzMqAlWZ2v7tvGLbfI+5+eYB1iIiIiOSdwFrC3H2vu6/K3O8ANgJ1QT2fiIiIyFiSkzFhZjYbWAo8NsLm883sKTO718wW5aIeERERkbAF2R0JgJmVAr8CPuzu7cM2rwJmuXunmb0G+DUwf4THuAG4AWDmzJkBVywiIiISvEBbwswsTjqA/dTd7xq+3d3b3b0zc/8eIG5mk0fY72Z3X+buy2pqaoIsWURERCQngjw70oBbgI3u/vUj7DMtsx9mtjxTz8GgahIRERHJF0F2R14AvBlYa2arM+s+BcwEcPebgKuA95pZEugGrnV3D7AmERERkbwQWAhz978Adox9vg18O6gaRERERPKVZswXERERCYFCmIiIiEgIFMJEREREQqAQJiIiIhIChTARERGRECiEiYiIiIRAIUxEREQkBAphIiIiIiFQCBMREREJgUKYiIiISAgUwkRERERCoBAmIiIiEgKFMBEREZEQKISJiIiIhEAhTERERCQECmEiIiIiIVAIExEREQmBQpiIiIhICBTCREREREKgECYiIiISAoUwERERkRAohImIiIiEQCFMREREJAQKYSIiIiIhUAgTERERCYFCmIiIiEgIFMJEREREQqAQJiIiIhIChTARERGRECiEiYiIiIQgsBBmZvVm9pCZbTCz9Wb2oRH2MTO70cy2mtkaMzsrqHpERERE8kkswMdOAv/o7qvMrAxYaWb3u/uGIfu8GpifuZ0LfC/zVURERGRcC6wlzN33uvuqzP0OYCNQN2y3K4GfeNqjQKWZ1QZVk4iIiEi+yMmYMDObDSwFHhu2qQ7YOWR5F38b1DCzG8yswcwampqagipTREREJGey7o40s9cCi4DCwXXu/oUsvq8U+BXwYXdvP5Ei3f1m4GaAZcuW+Yk8hoiIiEg+yaolzMxuAt4EfAAw4GpgVhbfFycdwH7q7neNsMtuoH7I8ozMOhEREZFxLdvuyBe7+1uAFnf/PHA+cOrRvsHMDLgF2OjuXz/CbiuAt2TOkjwPaHP3vVnWJCIiIjJmZdsd2Z352mVm04GDwLEG0F8AvBlYa2arM+s+BcwEcPebgHuA1wBbgS7gbdmXLiIiIjJ2ZRvC7jazSuArwCrAgR8c7Rvc/S+kuy6Pto8D/5BlDSIiIiLjRlYhzN3/OXP3V2Z2N1Do7m3BlSUiIiIyvmU7MH/V4H1371UAExERETk52Q7MP2q3ooiIiIgcn2zHhC0wszVDlo30kK4lAdQkIiIiMu5lG8K2Aa8LshARERGRiSTbENbn7tsDrURERERkAsl2TNgHAq1CREREZILJNoStNbNvDF5E28y+ZmYVgVYmIiIiMo5lG8JuBdqBazK3duBHQRUlIiIiMt5lOyZsnru/ccjy54dcikhEREREjlO2LWHdZvaSwQUzu4DnrycpIiIiIscp25aw9wK3DRkH1gK8NZiSRERERMa/bEPYPnc/w8zKAdy9PcCaRERERMa9bLsj74F0+FIAExERETl52YYwERERERlF2XZHLjGzoS1gg9eOLA+gJhEREZFxL9sQttbdlwZaiYiIiMgEou5IERERkRBkG8LeeOxdRERERCRb2Yawz5pZ5eCCmU0ys1sDqklERERk3Ms2hC1x99bBBXdvATRGTEREROQEZRvCImY2aXDBzKrIflC/iIiIiAyTbZD6GvBXM/sP0tNTXAV8KbCqRERERMa5rEKYu//EzFYCF2VW/Q933xBcWSIiIiLjW9Zdiu6+3syagEIAM5vp7jsCq0xERERkHMtqTJiZXWFmW4BtwJ+B54B7A6xLREREZFzLdmD+PwPnAU+7+xzgFcCjgVUlIiIiMs5lG8L63f0g6bMkI+7+ELAswLpERERExrVsx4S1mlkp8AjwUzNrBA4FV5aIiIjI+JZtS9gVQBfwYeA+YCtweVBFiYiIiIx3R20JM7NtgA9fnfn6EWDuUb73VtJBrdHdF4+w/ULgN6QH+wPc5e5fyK5sERERkbHtWN2RQ8d9GfAgz88Vdiw/Br4N/OQo+zzi7mpRExERkQnnqCEsMxj/MDNLDl93lO992Mxmn3hpIiIiIuNXtmPCMLO5PN8VOVrON7OnzOxeM1t0lOe+wcwazKyhqalplEsQERERyb1jjQlbS3pMWAFQDLx7FJ97FTDL3TvN7DXAr4H5I+3o7jcDNwMsW7Zs+Bg1ERERkTHnWGPCBsdr9bj7/tF8YndvH3L/HjP7rplNdvcDo/k8IiIiIvnoWGPCtgf1xGY2Ddjv7m5my0l3jWY13kxERERkrMv6At7Hy8zuAC4EJpvZLuCzQBzA3W8CrgLea2ZJoBu41t3V1SgiIiITQmAhzN2vO8b2b5OewkJERERkwsn67EgRERERGT0KYSIiIiIhUAgTERERCYFCmIiIiEgIFMJEREREQqAQJiIiIhIChbARDKQ0XZmIiIgESyFsmO0HD3HJN/7MY89q8n4REREJjkLYMLFoBAPefOvj/H79vrDLERERkXFKIWyYusoi7nzPi1lYW857b1/JLxt2hl2SiIiIjEMKYSOYVJLgZ+86l3PnVPPZ36ynszcZdkkiIiIyziiEHUFxIsbHL1tAd/8Av1uzJ+xyREREZJxRCDuKpfWVnDKllF827Aq7FBERERlnFMKOwsy4ZtkMVm5vYWtjZ9jliIiIyDiiEHYMb1g6g2jE+A8N0BcREZFRpBB2DDVlBVx82hR+tWo3/QOpsMsRERGRcUIhLAvXLKvnQGcvf97cFHYpIiIiMk4ohGXhwgU1lCSi/PlphTAREREZHQphWYhHIyybXcVfdSkjERERGSUKYVk6b241Wxs7aeroDbsUERERGQcUwrJ0/rxqAB5Va5iIiIiMAoWwLC2eXk5pQUwhTEREREaFQliWYtEI58yepBAmIiIio0Ih7DicN7eaZ5oO0djeE3YpIiIiMsYphB2Hw+PCtjWHXImIiIiMdQphx2HR9ArKCmL89Rl1SYqIiMjJUQg7DtGIsXxOlcaFiYiIyElTCDtO58ypYtuBQzQf6gu7FBERERnDFMKO05IZFQCs3d0WciUiIiIylgUWwszsVjNrNLN1R9huZnajmW01szVmdlZQtYymF9WlQ9iana0hVyIiIiJjWZAtYT8GLjvK9lcD8zO3G4DvBVjLqCkrjDO3poSndqklTERERE5cYCHM3R8GjjaXw5XATzztUaDSzGqDqmc0nTGjkjW71BImIiIiJy7MMWF1wM4hy7sy6/LekhkVNHb0sq9Nk7aKiIjIiRkTA/PN7AYzazCzhqamprDLYcmMSgCeUmuYiIiInKAwQ9huoH7I8ozMur/h7je7+zJ3X1ZTU5OT4o5mYW050YixVuPCRERE5ASFGcJWAG/JnCV5HtDm7ntDrCdrRYkop04tU0uYiIiInLBYUA9sZncAFwKTzWwX8FkgDuDuNwH3AK8BtgJdwNuCqiUIZ8yo4L71+3B3zCzsckRERGSMCSyEuft1x9juwD8E9fxBWzKjkp8/sZMdzV3Mqi4JuxwREREZY8bEwPx8NDhzvuYLExERkROhEHaCFkwrIxGLsFbjwkREROQEKISdoHg0wmnTyli/pz3sUkRERGQMUgg7CYumV7Budxvp4W0iIiIi2VMIOwmL68pp70myq6U77FJERERkjFEIOwmLp6cH56/brcH5IiIicnwUwk7CgmllRCPGuj0KYSIiInJ8FMJOQmE8yvwppazbrcH5IiIicnwUwk7S4joNzhcREZHjpxB2khZPL+fgoT72t/eGXYqIiIiMIQphJ2lxnQbni4iIyPFTCDtJp9eWY4YG54uIiMhxUQg7gvqZszCzY95KC+P0HdjJv3zv37PaP9e3+pmzwv5VioiIyAhiYReQr3bt3MHX/7A5q33vW7eP3a1z+OR7sts/lz566YKwSxAREck7je09AEwpLwytBrWEjYIpZQV09ibp6kuGXYqIiIgcxcHOXv7lno289F8f4iu/D7fxRC1ho6CmrACApo5eZlXrVyoiIpKPHtiwnw/+/El6+gd4/dI63n/xKaHWo8QwCqZkQlhjRy+zqktCrkZERESGa+ro5eN3PsXs6hJuvG4pp0wpDbskdUeOhoJ4lIqiOE0dmitMREQk37g7n7xrLV19A3kTwEAhbNTUlBXQqBAmIiKSd+5atZsHNu7n469akDcBDBTCRs2UsgLauvvp7R8IuxQRERHJaO/p5/O/Xc/y2VW87YI5YZfzAgpho+Tw4PxOtYaJiIjki188vpP2niSfed1CohELu5wXUAgbJTWlzw/OFxERkfAlB1L86L+2cd7cqsOXGcwnCmGjpKQgRmlBTCFMREQkT9y7bh972np450vmhl3KiBTCRlFNWQFN7QphIiIiYXN3fvjIs8yZXMLFp00Ju5wRKYSNoillBTR39dE/kAq7FBERkQlt5fYWntrVxttfModIno0FG6QQNoqmDJk5X0RERMJz+6PbKS+M8caz6sIu5YgUwkZRjUKYiIhI6Lr6kvxhw35eu2Q6xYn8vTiQQtgoKi2IURSPsr+jJ+xSREREJqwHNjbS1TfAlWdOD7uUo1IIG0VmxrSKQva3qSVMREQkLCtW72ZaeSHLZ1eFXcpRKYSNsmnlhTR39dGjmfNFRERyruVQH3/a3MQVZ07P2wH5gwINYWZ2mZltNrOtZvaJEbZfb2ZNZrY6c3tnkPXkQm1FIQD729UlKSIikmv3rttHMuVccUZ+d0UCBDZazcyiwHeAS4BdwBNmtsLdNwzb9Rfu/v6g6si1qeXpELa3rYdZ1SUhVyMiIjKx/Gb1bubVlLBoennYpRxTkC1hy4Gt7v6su/cBPweuDPD58kIiFqG6NME+tYSJiIjk1J7Wbh5/rpkrz6zDLL+7IiHYEFYH7ByyvCuzbrg3mtkaM7vTzOoDrCdnassL2dfWg7uHXYqIiMiEcfeaPbgzJroiIfyB+b8FZrv7EuB+4LaRdjKzG8yswcwampqaclrgiZhWUUhvMkVLV3/YpYiIiEwYv1m9hzPqK5k9eWwMBwoyhO0GhrZszcisO8zdD7r74HwOPwTOHumB3P1md1/m7stqamoCKXY01VYUAbCvTV2SIiIiubC1sYP1e9q5coy0gkGwIewJYL6ZzTGzBHAtsGLoDmZWO2TxCmBjgPXkzKTiOAWxCHvbu8MuRUREZEJYsXoPEYPLl9Qee+c8EdjZke6eNLP3A78HosCt7r7ezL4ANLj7CuCDZnYFkASageuDqieXzIxpmXFhIiIiEix35zdP7eHF8yYzJTNLwVgQ6Jgwd7/H3U9193nu/qXMus9kAhju/kl3X+TuZ7j7Re6+Kch6cmlaRSEHO/voS6bCLkVERGTU1M+chZnl1a2w7jS2H+zi1zd++ri+r37mrFB/l/l7VcsxrraiEAf2tnVrvjARERk3du3cwdf/sDnsMl7gz083sXZ3G5/+8o0UxKJZf99HL10QYFXHFvbZkePW9MoiombsbNa4MBERkaCk3Hl6fwezq4uPK4DlA4WwgMSjEaZVFLKzpSvsUkRERMatXS3ddPUNsGBaWdilHDeFsADVVxXR2NGri3mLiIgEZPO+DhLRCHPG4NAfhbAA1U8qBlBrmIiISACSAym2NnUyb0oJsejYizQamB+gqeWFxKPpcWHzp4y9ZlKR8a7huWZufvhZth9Mf1AqTES56qw6rl5WT2F8bI0tEZmInjvYRV8yxYKpY/N/7NiLjWNINGLUVRapJUwkz2xt7OCa7/+Vq276Kw3bW5g9uZg5k0tIpZz/85v1vPRfH+Lnj+8Iu0wROYbN+zsoTkQP9zyNNWoJC1h9VTHPbTlAR08/ZYXxsMsRmfAe39bMO297gng0wmdft5A3nVNPcSL9VujuPPpsM9944Gk+cddath04xP+67DQiEQu5ahEZrjc5wLYDh3jR9Ioxe4wqhAXs8Liw5m4WTlcIEwnTfev28sGfr2bGpCJue9ty6qte+OnZzDh/XjXL55zH51as5/sPP8veth6+evUZJGLqOBDJJ1v2dzKQ8jF5VuQghbCATS5NUBSPsqOli4XTy8MuR2TC+q+tB3j/z55kyYwKbnnrOUwqSRxx32jE+MKVi6itLORf79tMIhbhK1ctwWxsftoWGY/W7WmjuiTB1PKCsEs5YQphATMzZk8u5tmmQwyknOgYbTIVGcueaerkvbevZG5NCbe9fXlWQwPMjPddeAo9/Slu/OMW5k8p5d0vn5eDakXkWA509rK/vZeXzZ88pj8cqX09B+ZPKaM3mWJHswboi+Raa1cf77ytgXg0wi1vPee4x2Z++BXzee2SWv7ffZv4w/p9AVUpIsdj/Z52omacNm1s9zAphOXAzKpiErEIWxo7wi5FZEJxdz7yi9Xsbunm5rec/TdjwLIRiRhfu/oMltRV8JFfrGbbgUMBVCoi2UoOpNi0t525NSUUJcb2VDIKYTkQjRjzJpcc7pIUkdy45S/beGhzE5++/HTOnlV1wo9TGI/yvb8/m3gswvt+ukpXwRAJ0TNNh+hJplg0DsZZK4TlyClTS+lNptipLkmRnFizq5Uv37eJSxdO5c3nzTrpx5teWcTXrj6DjXvb+eLvNoxChSJyItbtbqOsMMbME2jZzjcKYTkys6qYRDTC0+qSFAlcR08/7//Zk9SUFvCvo3hW4ytOn8oNL5vL7Y/u4O41e0blMUUke/vbe9jV2s0ZMyrH9ID8QQphORKLRJhboy5JkaC5O5/6z3Xsbu3mxuuWUll85KkoTsTHX7WApTMr+cSv1vKcxoeJ5FTD9hYSsQiL68Z+VyQohOXU/EyXpAb2igTnPxp28dun9vDRS05l2ezsx4HVz5yFmR3zlohF+e0n3kBbawvnf+yHWCyR1ffl8lY/8+S7X0XyTWtXH1sbO1lSV0FBbGwPyB+kecJyaHZVCWWFMZ7c2cIpU0rDLkdk3Nna2MFnVqzjglOqec9xzum1a+cOvv6HzVnv/2xTJ79dU8brvvkgFy2YcrylBuqjly4IuwSRUbdyRwtRM86srwy7lFGjlrAcikTSfzx7WnvY194Tdjki40pnb5L33L6KkkSMb1xzZuATI8+tKeWsmZWs2dXGlv0a6ykSpEO9STbu7eD02jJKCsZP+5FCWI4tml5OIhrhye0tYZciMm64O//rzjU829TJt65bypTywpw874vnTWZaeSEPbGyktasvJ88pMhE9+uxB3J2zZk0Ku5RRNX7i5BhREIuyuK6cJ3e2ckF3P+VFuqi3ZKetu59nmjp5prGTvoEU8WiE8sIYC2srqK8qGhdnCp2oW/6yjd+t3csnX30aLz5lcs6eNxoxXr14Gj97fAf3rtvH1ctmEIvos63IaNrf3sO6Pe0sra9k0iifaBM2hbAQnFlfyeqdraze2crLTq0JuxzJY880dXLPmr3cu24fG/a2H3G/yuI4F8ybzJVnTufCBVNIxCZOEHhw037+772bePXiadzwsrk5f/7yojiXLpzKb9fs5cFNjVxy+tQJHYhFRpO786fNTRQnopw798QnXM5XCmEhKCuMs2BqGWt2t/GiugomlYyvZC8nx915eMsBfvjIszyy5QAAZ82s5OOvWsCCqWXMm1JKcSJKXzJFa1c/a3e3sXpnC3/c2Mjv1u6lsjjOW86fzdsvmD3q0zPkm5Xbm3nfT1exaHo5X7n6jNDCz9yaUs6dU8Vj25opScS4IIetcfkgOZBize42Nu3tYPO+dpo6e+lLOil3ppYXUF9VzKlTyjh3btVxX7tTJraN+zrY197DJadPHTdnRA6lEBaSC06ZzLMHDvHHTY288aw6fXIW3J37N+znmw9sYcPedqaUFfCxS0/lqrPrmVYx8hin+ip40YwK/ue5M+kfSPGXrQe447Ed3PjHLdzyyLP8/XmzeMdL5zClLDdjpHJpy/4O3v7jBmorirj1+nMoDXmw7rlzqjjUm6RhewslBbFxdQbXSPoHUjy4qZF71+7loc1NtHX3A1CSiFJbWUQiGsEsfeWCA53p8XKxiLF0ZiWvfVEtV5xZR5U+gMpRtHf388iWJqaVF3J6bVnY5QRCISwkJQUxXjp/Mg9sbGTdnnZeVFcRdkkSklTKeWDjfr714FbW7m5jdnUxX7lqCVeeWXdc3YrxaISLFkzhogVT2Lyvg+/+aSs/eORZfvzfz3HtOfW8++XzmF5ZFOBPkjtP7mjhHbc1kIhF+MnblzO5tCDskjAzLjptCt39A/z56SYGUs5ZM8fHrN5D7W3r5o7Hd/KLJ1VtKJQAABH4SURBVHawv72XqpIErzx9KhefNoUlMyqYMelvxyd29iZZu6uNR7Y08dDmJj732w186Z6NvPL09CWlzp9XPe5+T3JykgMp7l67l1QKLl00frv4FcJCtLC2nE37OvjL1gPMri5WM/0Ek0o5967bx7ce3MKmfR3MrEqHrzcsrSMWPbkxXQumlfFv1y7lI688le/96Rl+9vgO7nh8J1cvm8H7LjqFujEcxh7YsJ/337GKqeWF/Phty6nPo+vHRcy4bPE07l+/n79sPUBnT5KXnjqZyBj/BzKQch7Z0sRPH9vBHzfux4GXn1rDF18/i4sW1Bzz77W0IMb586o5f141/3TZaWzc286dK3dx16pd3LtuH6dMKeUt58/iDUvr9D4ouDsPbmqkqaOX151RO+4G4w+lEBYiM+MVp03hZ4/v4D+f3M1VZ8+gOKGXZLzrS6Y45eJrSJ76ChKTZ9J/cCdtf/0l2zf8mUc8FchzRstrqDjvGm7veyW3/3UbXZseof2J39C3b8tJPe6M+pns3LF9lKo8ut7kAN98YAvf//MzLK6r4Nbrz8mLFrDhYpEIly2eRsmWAzy5s5WWrj5ecfqU3IYLi4xKy0G0rJrSF11C6ZJLiVVMYeBQC51r7qfzqd9zW9t+bjvpJ4hTcvpL6V16OZ9p7OTTv3yCQ+sfpGPV7+g/uBPI7d+YhM/d+e9nDrJxXwfnzqli7uTxPbG5/uOHrLI4wZVn1PHr1bv5zyd388azZlAYH3+DD/NJciBFY0cve9t6aD7UR0dPP4f6BkilHHcnFo1QUhCltCDOlLICplUUMrm04KQm/xxIOU/tamXF6j38ZvVuIi9+G7UlCZbPqeKUKacQuebiUfwJj6y9p5+ndrayLn4xJQsvpKa0gNNqy1gw9cQmQMzVzOwNzzXzybvWsqWxk6vPnsHnrliU1xM2mhkvO7WGyuI4j2w5wO2P7eBl8yezsLY8N90qnjqu2f+H6h9IsaO5i/V72nnuwCEcqK8qYvH0CubVnEL0iuXA/x7VcgH2tfWwZlcrTxe9jrKzLmfGpCIWTS/nB+98+ag/l+SngczQjE37Olg8vZxz54y/syGHy993sQmkblIRly+p5bdr9nLnql28auE0asry7xP+WNGbHGBPaw87m7vY1dLNzpb01z2t3ext7WZ/R+9xX0Q9GjGmZgJZbUVR5msh0yoKmVZeSFlhnKJ4lGjU6OpN0tmbZHdrN880HmL9njYeffYg7T1JEtEIlyyayq2ffhcfvOlnOR/nUF4Y56Xza1g+p4pNezvYsLedR7Yc4JEtB6gpLWBmdTHTKwqpKSugtCAW6jiMgZTz8JYmbvrTMzy2rZnaikJ+/LZzuDDPLhF0NEtmVDKzqpj7N+7ngY2NrNzewrJZVSyYVhb4jP7ZSqWcps70h5KdzV3saO4imXKKE1HOnjWJRdPLc3KW7bSKQqZVTOMl85Os39PO2t1t/H79fuo/8FOu/9HjnD+3mnPmVLGwtvy4PqgmB1I0d/XRfOj5W8uhPrr7B0imnIEBJ5lykqkUiWiUssIYZYUxyovilBXGmFScoKasgEnFibx5zcaj5kN9PLipkd2t3enXevakcTsObKhAQ5iZXQb8GxAFfuju/2/Y9gLgJ8DZwEHgTe7+XJA15atZ1SW8bkktv1+/n58/sYNls6pYNnsS8ZMcGzTWuTu9yRSdvUmWLr+AxuZWLFFMpKiUaHEl0eIKIsWVRIvLiZZVE6uYSrS0CrPnf28+kCTZcYCBtsb01/ZGku0HSHYcINXVSqq3i1RfF6TSXYEWiWKJIiIFxURLqtKPWzaZtrJqnimrIVZWRbSshkji2GccuqdItu6nZ8caep5bTc+2VWzpTV/APcw3mIJYlDPqKzmjvpLmQ31sbepkx8EuntzRwspMPi2MRSgvilNeGKc4EaUgHqEgFqUgFqEgFiEei1AwYxFP7WylMB6lKB6lMB6hMBGlMBYlHrXj+hl7+gfY2tjJ5n0d/NfWAzy0uZGWrn5qKwr59GtP57rlM/O69etIKosTXHXWDJ7e30nD9mbu37ifh7c0Mbu6hDmTS5haXkBFUXzU/h7cHYsV0N03QDKVIplKH0PdfQN09w8c/trR058OJF39hz+UlBXGWDS9nDmTS5gxqTiU0FGciHHO7CqWzZrEvvYefvjdb7F92t/zp81NAJjB9Ioi6quKqCxKUFoYIxGLkBxIkRxw2rr7ae5KB63mQ3209ySzet5oxI764SxiUF1aQE1pAZPLBr8mqCktoCazXFNWwOTSAsqL4gpsWerqS9LwXAtP7WolFo3wqoVTOa22POyycsbcj69FIOsHNosCTwOXALuAJ4Dr3H3DkH3eByxx9/eY2bXAG9z9TUd73GXLlnlDQ0MgNQ9lZifcnH8yuvsHePjpJjbt6yARjbBgWhmnTStjannhCR3UH710AUG9xkfj7vT0p+jo7aezJ90y1NGTvnX2Juns6U+v600e3t7Z8/zy0O/rHzh6/QWxCMWJKMWJGOVFMcoL45QXxakojFNWFKO0IDbqA6Pdnb5kKl1vb5L+ZPqf3YA7iWiEeDRCaUGMyuL4iEH6o5cuCOXv61j6B1I0dfTS1NHLwUN9tPf0097dT3f/AL3JFMfzpxSNGIWxCIXxaOYWoehwQIswkHL6Uynau/tp6eqnpavv8ONXFse5aMEUXnn6VC5ZODUnk8/m4ph3d7Y3d/H0/g6eO9BFd/8AkJ66oaIoTlEiHWYjEWPwL3bwT3cg5QyknORA+veWHMgspzwdQDL3s2nljZpRUhBlUkmCqpIEU8sKmV5ZmJeD4gffw5o6eml4rpnN+zvYfrCLnc1dtPek3yf6BlLEIhFi0fTvsaokwaTi9M9WWRynuiRx+GetLilgUkmc4kSMWMSIRYxoJP2BITmQ/sDX3p1M/+339NPa1U9TRy8HOnsPHxuH73f2HvH9qTCefg8oKYhRkohRUhBN3y+IURiLUpSIZD64PH8riqfXF8aiFCaih7cPfsApikdP+EPOaDqZY8Xdae9Jsre1O/1aNnfhDounl3P+vOqcj4vOxf9IM1vp7stG2hbkT7sc2Oruz2aK+DlwJbBhyD5XAp/L3L8T+LaZmYeRGvJEUTzKqxZN40V1Fazd3caGvelm+WjEmFJWQFVJgrLMgRyPRohHjVg0cvjNBMAzNxwStaeycnsL4Lin16cyb9b9Ayn6Bwa/pu8nM/f7ht3vH0h/ku7pf+Gn6aHLPf2p5+8nB7L6h52IRSgriFFamA5LZYUxplcWUVZYRumw9e9719u5/lNfzbTERA//wwrjE6eZURCPUhCP5uXg8BMVj0aYXlk04lQW7k7/gNObTL/WyVSKb33seiwWx+IFWCyBxQoO348MuZ9en/kaK8CiMUgl8dRAuiWyu52BQy30H9xJf9N2tjfv5ilP8c0QfgdBMjNmV5cwu7qEVCZYHOjs5UBn3+Gw29TZmz5WMweQA+7pUBuLpo/zeCRCUSJz3A85/gfv33fr13j9u/8ps85IRCOHj5eiRDQzh9cYaakZpZMMghIpKCFSOolo8SSipZOIllQSKSjBEkU0xYuIJIqwRBFF5ZNYctY57GjuovcE3iuHG/yQU5QYFuLig63WEWKRCNGoEc/8bcSjdjisDv69DG6LRdN/V9GIveD/SsQMMw5/BSg+/WU8vb/j8AcFDIznt/dnWiWH/o8Z/CDenOkKhvRZs2fNnMTC2vIJO2dckCGsDtg5ZHkXcO6R9nH3pJm1AdXAgQDrGhMG/xFeeOoAO5q72Nvew762HrYdOERX30DWj1P7lq/zxu/990nX4wP9eH8vqf5ePNmLD/n6gnX9vaSG3u89hPd1p//R9h4i1deF93aR6utOdwEOZNdVMGhezfg+UyafmRmJmKWDc6Yntmf76rxs0TsRuTrJYFDEjKnlhUwN4GLjv3z8V5z5xX8Z9ccNxUmcZJBPPvqq09nx4yOc/RyNE4klsHjh8x9c/ubDTMELth15feZ7o3GIxLBIJP01Gs0sR9MfgiyKRaPp/Y5TzRX/xL3r9mW9f8TSc2OWFcSYPbmYaeXpsbSTywrG/PQtJyvI7sirgMvc/Z2Z5TcD57r7+4fssy6zz67M8jOZfQ4Me6wbgBsyiwuAXByRk1EYzDd6TfKPXpP8pNcl/+g1yU+5eF1mufuIF4oOsiVsN1A/ZHlGZt1I++wysxhQQXqA/gu4+83AzQHVOSIzazhSH66EQ69J/tFrkp/0uuQfvSb5KezXJcjRrk8A881sjpklgGuBFcP2WQG8NXP/KuDBiTweTERERCaOwFrCMmO83g/8nvQUFbe6+3oz+wLQ4O4rgFuAfzezrUAz6aAmIiIiMu4Fei6ou98D3DNs3WeG3O8Brg6yhpOQ0+5PyYpek/yj1yQ/6XXJP3pN8lOor0tgA/NFRERE5Mgm9nTsIiIiIiFRCBvGzC4zs81mttXMPhF2PRORmdWb2UNmtsHM1pvZhzLrq8zsfjPbkvk6KexaJyIzi5rZk2Z2d2Z5jpk9ljlmfpE5EUdyxMwqzexOM9tkZhvN7HwdK+Ezs49k3r/WmdkdZlaoYyX3zOxWM2vMTIk1uG7E48PSbsy8PmvM7Kyg61MIGyJzqaXvAK8GFgLXmdnCcKuakJLAP7r7QuA84B8yr8MngD+6+3zgj5llyb0PARuHLH8Z+Ia7nwK0AO8IpaqJ69+A+9z9NOAM0q+NjpUQmVkd8EFgmbsvJn1y2rXoWAnDj4HLhq070vHxamB+5nYD8L2gi1MIe6HDl1py9z5g8FJLkkPuvtfdV2Xud5D+p1JH+rW4LbPbbcDrw6lw4jKzGcBrgR9mlg24mPRlx0CvS06ZWQXwMtJnmuPufe7eio6VfBADijJzYBYDe9GxknPu/jDp2ReGOtLxcSXwE097FKg0s9og61MIe6GRLrVUF1ItApjZbGAp8Bgw1d33ZjbtA6aGVNZE9k3gn4DB669UA63uPnj9KR0zuTUHaAJ+lOki/qGZlaBjJVTuvhv4KrCDdPhqA1aiYyVfHOn4yHkGUAiTvGVmpcCvgA+7e/vQbZlJfXVqbw6Z2eVAo7uvDLsWOSwGnAV8z92XAocY1vWoYyX3MmOMriQdkqcDJfxtl5jkgbCPD4WwF8rmUkuSA2YWJx3Afurud2VW7x9sGs58bQyrvgnqAuAKM3uOdFf9xaTHI1VmulxAx0yu7QJ2uftjmeU7SYcyHSvheiWwzd2b3L0fuIv08aNjJT8c6fjIeQZQCHuhbC61JAHLjDO6Bdjo7l8fsmnoZa7eCvwm17VNZO7+SXef4e6zSR8bD7r73wEPkb7sGOh1ySl33wfsNLMFmVWvADagYyVsO4DzzKw48342+LroWMkPRzo+VgBvyZwleR7QNqTbMhCarHUYM3sN6XEvg5da+lLIJU04ZvYS4BFgLc+PPfoU6XFhvwRmAtuBa9x9+IBLyQEzuxD4mLtfbmZzSbeMVQFPAn/v7r1h1jeRmNmZpE+USADPAm8j/QFbx0qIzOzzwJtIn+39JPBO0uOLdKzkkJndAVwITAb2A58Ffs0Ix0cmMH+bdNdxF/A2d28ItD6FMBEREZHcU3ekiIiISAgUwkRERERCoBAmIiIiEgKFMBEREZEQKISJiIiIhEAhTEQmBDNzM7t9yHLMzJrM7O7M8hvNbL2ZPWJm1Zl188zsF2HVLCLjm0KYiEwUh4DFZlaUWb6EF86G/QHgHOD7wP/MrPsi8OmcVSgiE4pCmIhMJPcAr83cvw64Y8i2FFAAFAP9ZvZSYJ+7b8ltiSIyUSiEichE8nPgWjMrBJaQvgrDoP8LPAC8jnQ4+z/AP+e8QhGZMBTCRGTCcPc1wGzSrWD3DNt2v7uf7e6vA67MbD/VzO40sx+YWXHOCxaRcU0hTEQmmhXAV3lhV+RhmbB1PfAd4POkL/D7F+DvclSfiEwQsbALEBHJsVuBVndfm7kQ+XAfB2509/7MIH4nPV5MLWEiMqoUwkRkQnH3XcCNI20zs+nAcnf/fGbVt4AngFbg9bmpUEQmCnP3sGsQERERmXA0JkxEREQkBAphIiIiIiFQCBMREREJgUKYiIiISAgUwkRERERCoBAmIiIiEgKFMBEREZEQKISJiIiIhOD/A8d39sPArD2IAAAAAElFTkSuQmCC\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 720x360 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmEAAAFNCAYAAABIc7ibAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXzcZbn38c81S2ayL13TNl0ohdIWSkvZUREFUTZFZJGj4kE5i/tynsftwQMeH/U8R845uHMUATdURKzK4gLKXmlZuwHdm65pm61ZZzLX88dMQihJM0nml0ma7/v1Cpnll8lFp7/0m/u+fvdt7o6IiIiIjKxQvgsQERERGY8UwkRERETyQCFMREREJA8UwkRERETyQCFMREREJA8UwkRERETyQCFMREREJA8UwkTkiGZmW8zszZnb1Wb2AzPbZWbNZrbezG4ws+LM825me80s0uvro5nHtKiiiOSUQpiIjAtmVgU8ARQCp7t7KXAuUAHM7XVoPfDWXvffmnlMRCSnFMJEZLz4JNAM/J27bwFw9+3u/jF3f77XcT8C3tvr/nuBO0asShEZNxTCRGS8eDNwt7unBjjuHuD1ZlZhZpXA64DfBF6diIw7YzKEmdmtmR6N1Vkef7mZrTWzNWb206DrE5FRaQKwK4vj2oHfAldkPpZnHhMRyanIwIeMSrcB3ySLKQIzmwd8FjjT3evNbHLAtYnI6LQfqM7y2DuArwAG/O/AKhKRcW1MjoS5+8PAgd6PmdlcM7vfzFaZ2SNmNj/z1AeBb7l7feZr945wuSIyOvwJeIeZZfNz7xHSgW0K8GigVYnIuDUmQ1g/bgE+4u4nAZ8Gvp15/BjgGDN7zMyeNLPz81ahiOTTTUAZcLuZzQIws+lmdpOZndD7QHd34CLg4sxtEZGcG6vTka9iZiXAGcAvzaz74VjmcwSYB5wNzAAeNrPj3b1hpOsUkfxx9wNmdgbwb8CKzNpgO4CfARv6OH7NCJcoIuPMERHCSI/oNbj7iX08VwuscPcEsNnMXiIdyp4ayQJFJD/cfXav2zuBvz/MsdbP4xtI94eJiOTMETEd6e5NpAPWuwAsbXHm6XtIj4JhZhNJT09uykedIiIiIt0CC2FmFjezv5nZc5mlIW7o45hrzKzOzJ7NfHwgy9f+GemVr481s1ozuxa4GrjWzJ4D1gCXZA5/ANhvZmuBh4B/cff9ufh/FBERERkqC6rn1NLNWcXuftDMoqSvMPqYuz/Z65hrgGXu/uFAihAREREZpQLrCctcUXQwczea+dBVRiIiIiIE3BNmZmEzexbYC/zR3Vf0cdg7zex5M7vLzGqCrEdERERktAhsOvJV38SsAvg16XW8Vvd6fAJw0N07zOwfgCvc/Zw+vv464DqA4uLik+bPn3/oISIiIiKjzqpVq/a5+6S+nhuREAZgZtcDre7+H/08HwYOuHv54V5n2bJlvnLlyiBKFBEREckpM1vl7sv6ei7IqyMnZUbAMLNC4Fxg/SHH9N7H7WJgXVD1iIiIiIwmQS7WWk16e5Aw6bD3C3f/nZndCKx09+XAR83sYiBJei/IawKsR0RERGTUGLHpyFzRdKSIiIiMFXmZjhQRERGR/imEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQpiIiIhIHiiEiYiIiOSBQtghNtYd5IN3rGTNzsZ8lyIiIiJHMIWwQxRGw/xx7R5WbqnPdykiIiJyBFMIO0R1eZyJJTGeq23IdykiIiJyBFMIO4SZcWJNOc9tVwgTERGR4CiE9eGEGRVs2tdCU3si36WIiIjIEUohrA+Laypwh9W1as4XERGRYCiE9WHxjHIAnlVfmIiIiAREIawPFUUFzJpQxPPbNRImIiIiwVAI68fiGRW6QlJEREQCoxDWj8U1FexqbGdvU3u+SxEREZEjkEJYP7r7wp5Tc76IiIgEQCGsHwunlRMOmdYLExERkUAohPWjsCDMMVNK1RcmIiIigVAIO4zjqkvZsPdgvssQERGRI1BgIczM4mb2NzN7zszWmNkNfRwTM7Ofm9kGM1thZrODqmcoJhQXUN/ame8yRERE5AgU5EhYB3COuy8GTgTON7PTDjnmWqDe3Y8G/hP4WoD1DFplcQHtiRRtnV35LkVERESOMIGFME/rnsuLZj78kMMuAW7P3L4LeJOZWVA1DVZVUQGARsNEREQk5wLtCTOzsJk9C+wF/ujuKw45ZDqwHcDdk0AjMCHImgajsjgdwg60KISJiIhIbgUawty9y91PBGYAp5jZoqG8jpldZ2YrzWxlXV1dbos8jEqNhImIiEhARuTqSHdvAB4Czj/kqR1ADYCZRYByYH8fX3+Luy9z92WTJk0KutweVcVRAOpbEyP2PUVERGR8CPLqyElmVpG5XQicC6w/5LDlwPsyty8DHnT3Q/vG8qZnJEzTkSIiIpJjkQBfuxq43czCpMPeL9z9d2Z2I7DS3ZcDPwB+ZGYbgAPAlQHWM2jlhVHM1BMmIiIiuRdYCHP354ElfTx+fa/b7cC7gqphuCLhEGXxqHrCREREJOe0Yv4AqooL1BMmIiIiOacQNoDKoqh6wkRERCTnFMIGUFVcoJ4wERERyTmFsAFUFGn/SBEREck9hbABVGkTbxEREQmAQtgAKou0ibeIiIjknkLYALpXzT+g0TARERHJIYWwAVRo1XwREREJgELYAKqKtYm3iIiI5J5C2AC694/UMhUiIiKSSwphA+gZCVMIExERkRxSCBtAzybe2rpIREREckghbADhkFFeGKVBPWEiIiKSQwphWagq0tZFIiIiklsKYVmo1Kr5IiIikmMKYVmoLIpyoEU9YSIiIpI7CmFZqCwqUE+YiIiI5JRCWBaqitM9Ye6e71JERETkCKEQloWKogI6kinaEtrEW0RERHJDISwLPZt46wpJERERyRGFsCx0b13UoAVbRUREJEcUwrLQvXWRRsJEREQkVxTCslCRGQnTWmEiIiKSKwphWagsSveEaTpSREREckUhLAsl8QgABzuSea5EREREjhQKYVmIRcIUREI0tyuEiYiISG4ohGWpNBahuV3TkSIiIpIbCmFZKolHNB0pIiIiOaMQlqWSWISDmo4UERGRHFEIy1JJLEKzRsJEREQkRwILYWZWY2YPmdlaM1tjZh/r45izzazRzJ7NfFwfVD3DVRqPaiRMREREciYS4GsngU+5+9NmVgqsMrM/uvvaQ457xN0vDLCOnChVT5iIiIjkUGAjYe6+y92fztxuBtYB04P6fkEriSmEiYiISO6MSE+Ymc0GlgAr+nj6dDN7zszuM7OFI1HPUJTE00tUuHu+SxEREZEjQOAhzMxKgF8BH3f3pkOefhqY5e6LgW8A9/TzGteZ2UozW1lXVxdswf0oiUVIdDkdyVRevr+IiIgcWQINYWYWJR3AfuLudx/6vLs3ufvBzO17gaiZTezjuFvcfZm7L5s0aVKQJferVFsXiYiISA4FeXWkAT8A1rn7Tf0cMzVzHGZ2Sqae/UHVNBwlsUwI0xWSIiIikgNBXh15JvAe4AUzezbz2OeAmQDu/l3gMuCfzCwJtAFX+ihtuiqNRwGNhImIiEhuBBbC3P1RwAY45pvAN4OqIZe6R8K0ibeIiIjkglbMz1J3T5g28RYREZFcUAjLUk9PmKYjRUREJAcUwrJUoqsjRUREJIcUwrKknjARERHJJYWwLMWjYQrCIY2EiYiISE4ohA1CSTyidcJEREQkJxTCBkGbeIuIiEiuKIQNQkksoiUqREREJCcUwgahJB5RY76IiIjkhELYIJRqOlJERERyRCFsEErjCmEiIiKSGwphg6CrI0VERCRXFMIGoSQWpVkjYSIiIpIDCmGDUBqP0JlM0ZHsyncpIiIiMsYphA1CzybempIUERGRYVIIG4SeEKYpSRERERkmhbBBKIlrE28RERHJDYWwQSiNayRMREREckMhbBBKY1FAPWEiIiIyfAphg1CikTARERHJEYWwQehuzNcm3iIiIjJcCmGD0N0TpgVbRUREZLgUwgYhFgkRCZl6wkRERGTYFMIGwcy0ibeIiIjkhELYIGkTbxEREckFhbBB0ibeIiIikgsKYYNUGovo6kgREREZNoWwQSpRT5iIiIjkgELYIJXE1BMmIiIiw6cQNkgaCRMREZFcCCyEmVmNmT1kZmvNbI2ZfayPY8zMbjazDWb2vJktDaqeXCmNR2jWSJiIiIgMU5AjYUngU+6+ADgN+JCZLTjkmLcC8zIf1wHfCbCenCiNRehIpuhMpvJdioiIiIxhgYUwd9/l7k9nbjcD64Dphxx2CXCHpz0JVJhZdVA15UL3/pEtmpIUERGRYRiRnjAzmw0sAVYc8tR0YHuv+7W8NqhhZteZ2UozW1lXVxdUmVkpiUcBNCUpIiIiwxLJ9kAzuwBYCMS7H3P3G7P4uhLgV8DH3b1pKEW6+y3ALQDLli3zobxGrnSPhDV3aK0wERERGbqsRsLM7LvAFcBHAAPeBczK4uuipAPYT9z97j4O2QHU9Lo/I/PYqFUaT4cwLVMhIiIiw5HtdOQZ7v5eoN7dbwBOB4453BeYmQE/ANa5+039HLYceG/mKsnTgEZ335VlTXnRE8LUEyYiIiLDkO10ZFvmc6uZTQP2AwM10J8JvAd4wcyezTz2OWAmgLt/F7gXeBuwAWgF3p996fnRPR2pECYiIiLDkW0I+52ZVQD/D3gacOB/DvcF7v4o6anLwx3jwIeyrGFUKMmMhKkxX0RERIYjqxDm7l/K3PyVmf0OiLt7Y3BljV6lMV0dKSIiIsOXbWP+09233b1jvAYwgHg0RDhkHNTVkSIiIjIM2TbmH3ZacTwxM23iLSIiIsOWbU/YsWb2fK/7Rrql64QAahr1SmIRmtWYLyIiIsOQbQjbDFwUZCFjSWlcI2EiIiIyPNmGsE533xpoJWNIaTyiJSpERERkWLLtCftIoFWMMSUxhTAREREZnmxD2Atm9p/dm2ib2dfNrDzQykaxknhUS1SIiIjIsGQbwm4FmoDLMx9NwA+DKmq0K4lFFMJERERkWLLtCZvr7u/sdf+GXlsRjTvpnjCtEyYiIiJDl+1IWJuZndV9x8zO5JX9JMed0liE9kSKRFcq36WIiIjIGJXtSNg/Abf36gOrB94XTEmjX/f+kS0dSSqKCvJcjYiIiIxF2Yaw3e6+2MzKANy9KcCaRr2S2CubeCuEiYiIyFBkOx15L6TD13gPYJDuCQNt4i0iIiJDl20Ik15KYlEArRUmIiIiQ5btdOQJZtZ7BKx778iyAGoa9bp7wnSFpIiIiAxVtiHsBXdfEmglY0jvnjARERGRodB05BCU9YyEKYSJiIjI0GQbwt458CHjR890pEbCREREZIiyDWFfNLOK7jtmVmlmtwZU06hXGA0TMo2EiYiIyNBlG8JOcPeG7jvuXg+M2x4xM9P+kSIiIjIs2YawkJlVdt8xsyqyb+o/IpXGowphIiIiMmTZBqmvA0+Y2S9JL09xGfDlwKoaA0pi2sRbREREhi6rEObud5jZKuCNmYcudfe1wZU1+pXGI+oJExERkSHLekrR3deYWR0QBzCzme6+LbDKRrmSeIT6ls58lyEiIiJjVFY9YWZ2sZm9DGwG/gpsAe4LsK5RryQWoVkjYSIiIjJE2Tbmfwk4DXjJ3ecAbwKeDKyqMaA0HtE6YSIiIjJk2YawhLvvJ32VZMjdHwKWBVjXqKclKkRERGQ4su0JazCzEuAR4CdmthdoCa6s0a8kFqUt0UWyK0UkrN2fREREZHCyTQ8XA63Ax4H7gQ3AhUEVNRaUZrYuaunoynMlIiIiMhYddiTMzDYDfujDmc+fAI46zNfeSjqo7XX3RX08fzbwG9LN/gB3u/uN2ZWdf937RzZ3JCgviua5GhERERlrBpqO7N33ZcCDvLJW2EBuA74J3HGYYx5x9zE5olYay2zirSskRUREZAgOG8Iyzfg9zCx56GOH+dqHzWz20Esb3bpHwnSFpIiIiAxF1h3lZnYUr0xF5srpZvacmd1nZgsP872vM7OVZrayrq4uxyUMTUlmJExXSIqIiMhQDNQT9gLpnrAYUAT8Qw6/99PALHc/aGZvA+4B5vV1oLvfAtwCsGzZskN71PKitKcnTCFMREREBm+gnrDufq12d9+Ty2/s7k29bt9rZt82s4nuvi+X3ycoJbF0M76mI0VERGQoBuoJ2xrUNzazqcAed3czO4X01GhW/WajQfdI2MGORJ4rERERkbEo6w28B8vMfgacDUw0s1rgi0AUwN2/C1wG/JOZJYE24Ep3HxVTjdkoKggTMvWEiYiIyNAEFsLc/aoBnv8m6SUsxiQzozQeVQgTERGRIdF+O8NQVhihqU3TkSIiIjJ4CmHDUBaP0tSuECYiIiKDpxA2DGXxKI0aCRMREZEhUAgbhvR0pHrCREREZPAUwoZB05EiIiIyVAphw1BWGFVjvoiIiAyJQtgwlMWjtHR2kexK5bsUERERGWMUwoahrFCbeIuIiMjQKIQNQ1k8vX+k+sJERERksBTChqG8MBPCdIWkiIiIDJJC2DCUFWokTERERIZGIawfNTNnYWaH/TjrlKUAnHfBJQMem6+Pmpmz8vwnKSIiIn0JbAPvsa52+zZu+sOLhz2muT3BrY9t4crP/ReLppWPUGWD88nzjs13CSIiItIHjYQNQywSBqAzoSUqREREZHAUwoYhGjYMaE925bsUERERGWMUwobBzIhFQnQkNRImIiIig6MQNkyxaFghTERERAZNIWyYYpEQHQlNR4qIiMjgKIQNU4GmI0VERGQIFMKGKRYJ0akQJiIiIoOkEDZMsYh6wkRERGTwFMKGKRYN0aElKkRERGSQFMKGKRYJkehyulKe71JERERkDFEIG6Z496r5mpIUERGRQVAIG6ZYJP1HqClJERERGQyFsGEqiHaHMI2EiYiISPYUwoapexPvdi3YKiIiIoOgEDZM3dOR6gkTERGRwVAIG6ZXesIUwkRERCR7gYUwM7vVzPaa2ep+njczu9nMNpjZ82a2NKhagtQ9HakQJiIiIoMR5EjYbcD5h3n+rcC8zMd1wHcCrCUw0bBhpqsjRUREZHACC2Hu/jBw4DCHXALc4WlPAhVmVh1UPUExM2KREB0JjYSJiIhI9vLZEzYd2N7rfm3msTFH+0eKiIjIYI2Jxnwzu87MVprZyrq6unyX8xqxiPaPFBERkcHJZwjbAdT0uj8j89hruPst7r7M3ZdNmjRpRIobjHQI00iYiIiIZC+fIWw58N7MVZKnAY3uviuP9QyZpiNFRERksCJBvbCZ/Qw4G5hoZrXAF4EogLt/F7gXeBuwAWgF3h9ULUGLRTUdKSIiIoMTWAhz96sGeN6BDwX1/UeSro4UERGRwRoTjfmjXSwSJplyulKe71JERERkjFAIy4FYNP3HqE28RUREJFsKYTlQFE1vXdTaqRAmIiIi2VEIy4GignRrXZtGwkRERCRLCmE5UFiQHglr00iYiIiIZEkhLAe6Q1hrZzLPlYiIiMhYoRCWA/FICDNNR4qIiEj2FMJywMwojIY1HSkiIiJZUwjLkcKCsEbCREREJGsKYTlSGA1riQoRERHJmkJYjhQVKISJiIhI9hTCcqQwqulIERERyZ5CWI4UFUToTKZIprSRt4iIiAxMISxHutcKa+9UCBMREZGBKYTlSGH3/pEJLdgqIiIiA1MIy5EibV0kIiIig6AQliPaP1JEREQGQyEsR4p6piMVwkRERGRgCmE5UhAJETKNhImIiEh2FMJyxMwo1IKtIiIikiWFsBwqika0YKuIiIhkRSEshwoLwpqOFBERkawohOVQejpS64SJiIjIwBTCckj7R4qIiEi2FMJyqKggTKLLSXRp6yIRERE5PIWwHNKCrSIiIpIthbAc6l6wVVOSIiIiMhCFsBzqHgnTWmEiIiIyEIWwHCrUSJiIiIhkSSEsh4oKIoB6wkRERGRggYYwMzvfzF40sw1m9pk+nr/GzOrM7NnMxweCrCdo0bARDpnWChMREZEBRYJ6YTMLA98CzgVqgafMbLm7rz3k0J+7+4eDqmMkmRlFBWFaNBImIiIiAwhyJOwUYIO7b3L3TuBO4JIAv9+oUBaP0tSWyHcZIiIiMsoFGcKmA9t73a/NPHaod5rZ82Z2l5nVBFjPiCgrjCiEiYiIyIDy3Zj/W2C2u58A/BG4va+DzOw6M1tpZivr6upGtMDBKi+M0tLZRVKr5ouIiMhhBBnCdgC9R7ZmZB7r4e773b0jc/f7wEl9vZC73+Luy9x92aRJkwIpNlfK41EAmtrVnC8iIiL9CzKEPQXMM7M5ZlYAXAks732AmVX3unsxsC7AekZEWWEmhGlKUkRERA4jsKsj3T1pZh8GHgDCwK3uvsbMbgRWuvty4KNmdjGQBA4A1wRVz0jpDmGN7QphIiIi0r/AQhiAu98L3HvIY9f3uv1Z4LNB1jDSigvChEOmkTARERE5rEBD2HhkZpTFIzQqhImMel0p5wv3rGbF5v1AeomZf7/sBI6ZUprnykRkPMj31ZFHpLLCqBrzRcaAf79/PT/72zZmTyhmQXUZtfWtfOD2ldS3dOa7NBEZBxTCAlCuBVtFRr1frarlew9v4j2nzeLWa07mm+9eyi3vXcbupnb++SdPk9AyMyISMIWwAJQVRulIpuhIaPsikdFozc5GPnv3C5x+1ASuv2hBz+NLZ1bylXcczxOb9vO1+9bnsUIRGQ8UwgJQVphutdMVkiKj001/eImiWJhvX72UaPjVPwbfedIMrjqlhtse38KOhrY8VSgi44FCWAC6F2xVc77I6LN2ZxN/Xr+Xvz9zDpXFBX0e8+Fz5gHwPw9vGsnSRGScUQgLwCsLtqo5X2S0+fZfNlASi/C+02f3e8z0ikLesWQ6dz61jX0HO/o9TkRkOBTCAhCPholFQmrOFxllNtUd5Pcv7OLvTptFeVH0sMf+49lz6UimuPXRzSNUnYiMN1onLCBlhVH1hEnO7W5s599+v5a2zi6OnlLCkpoK3rJwKmaW79LGhO/+dSMF4RDXnjVnwGPnTirhbcdX86MntvIPb5hLeeHhQ5uIyGBpJCwgZfGIRsIkpx5cv4e33fwIf163l9r6Nn746Bb+8cdP89X71uPu+S5v1Nt3sINfP7ODy5fVMKk0ltXX/PPZc2nuSPKLp7YHXJ2IjKT2RBc/f2obT2zcn9c6NBIWkPLCKFv2t+LuGqWQYfvxk1v5wj2rmT+1lG9dvZS5k0pIdqW44bdr+d7Dm4iEjU+fd6z+rh3GL1ZuJ9HlvO+MWVl/zcJp5SybVclPVmzl2rPmEArpz1dkLGvpSHLro5u5/Ymt7DvYwZUn13D63Al5q0chLCBl8ShdKaels4uSmP6YZehe2tPMjb9by+uPmcQt7zmJeDQMQCQc4oaLF5JMOd96aCPlhVGue/3cPFc7OqVSzk9XbOO0o6o4evLgtiT6u9Nm8fGfP8vjG/dz1ryJAVUoIkFLdqX4xx+v4pGX93H2sZP44OuO4ow8BjDQdGRgJpSkL32va9aVVTJ0HckuPnbns5TGInz9XYt7Ali3UMj48tsX8dZFU/mPB15iw96Deap0dPvry3XU1rdx9anZj4J1O3/RVCqLovz4ya0BVCYiI+Ur963nkZf38dVLj+e295/CmUdPzPvsgUJYQKaUxTGDXY1a7FGG7qY/vMS6XU187Z0n9NvHFAoZN16yiHg0xOd//YL6w/rwkye3MrEkxlsWTh3018ajYS5fVsMf1+1hT1N7ANWJSNB+uXI7P3h0M9ecMZsrT5mZ73J6KIQFJBoOMakkxq5G/dCWoXmhtpFbHtnEVafM5M0Lphz22EmlMT77tuNYsfkAv1xVO0IVjg07Gtp4cP1erjh5BgWRof3Ie/epM+lKOXf+TQ36ImPN1v0tfP6e1Zx59AS+cMFx+S7nVRTCAlRdHmdPUzuplEYmZHBSKef65auZUBzjc2+bn9XXXLGshpNnV/J/713HgZbOgCscO366YisOXDWM335nTSjmdfMm8rO/bdPG3iJjzJd+t45oyLjp8hOJhEdX7Bld1RxhqssLSXQ5+1rUFzaauTtrdzZx3wu7uOXhjdz9dC3ted58/e5ndvDMtgY+89b5lMazW58qFDK+/I7jaW5P8l9/eingCseG9kQXP12xjXOPm8KMyqJhvdY1Z8xmd1M796/enaPqRCRof3lxL39at4ePvGkeU8ri+S7nNXTZXoCqy9Nv+K7GdiaXjr43X2Dzvhau/81qHnl536sev+G3a7ni5Bo+9MajR3yRzqb2BF+9bz1LZlZw6ZLpg/raY6aU8u5TZvKTFdt47+mzBn0l4JHmnmd2UN+a4P1nDrw460DeeOxkZk8o4tbHNnPR4mk5qE5EgtSZTHHjb9cyZ2Ix7z9zdr7L6ZNGwgJUGo9QXBBWX9go5O585y8bect/Pcyz2xr4wgXH8fuPnsVzXzyPn33wNM46eiI/eHQzl377MbbubxnR2m7+08vsb+ngxosXDWldqo+/eR5FBWG+/Pt1AVQ3drg7P3xsC8dVl3HaUVXDfr1QyLjmjNk8s62BZ7c35KBCEQnSbY9vZtO+Fq6/cAGxSHjgL8gDhbAAmRnV5YXsatAVkqNJsivFZ371Al+7fz1vmj+ZP3/qDXzgdUexcFo55YVRTp87gW9dvZSffOBU9rd08vZvPcaKTSOzqvKGvc3c9vgWrjy5huNnlA/pNSaUxPjIOUfz0It1PPxSXY4rHDse37ifF/c08/4zZ+fsMvTLltVQGovww8e0n6TIaLa3qZ3//tPLnDN/Mm+cPznf5fRLISxg1eVxmtqTtHQk812KAK2dST54x0p+vnI7Hz3naL599VIm99MncNpRE7jnn8+ksriA9976Nx7fsK/P43LF3fnX5WspKgjz6fOOHdZrve+M2cysKuLG362lMzk+G8l/+NhmJhQXcHGWU4c1M2dhZof9KI1H2fHoXdyzahuR0gkDHlUxU3kAABevSURBVJ+vj5qZg18PTeRI8rX7XyTR5Vx/4YJ8l3JY6gkLWHXFK31hR08uyXM149u+gx1ce9tTvLCjkS+/Y1FWC3fOnljMXf94Blfd8iTX3r6S2//+FE6ZM/yprb48sGY3j27Yxw0XL2RCSXZ7G/YnFgnzrxcv4O9vW8n3/rqRj7xpXo6qHBvW7Gzkz+v38pFz5r1mgdv+1G7fxk1/eHHA4xrbEtz++BYu/Opy3nDMpOGWGohPDjPEi4xlT2+r51dP1/JPZ89l9sTifJdzWBoJC9ik0hhhMy3aOgTuzvYDrTy7vYFnttXzQm0jbZ1Du2px6/4WLvvO47y4p5nvvWfZoFZOryou4McfOJVpFXHe/8O/8fS2+iHVcDgtHUm+9Lt1zJ9aytWn5mYhwXPmT+GC46v5xkMb2FQ3vlbS/+p96ykvjHLtWcNvyD9UeWGUBdPKeL62gca2RM5fX0SGrivl/OvyNUwpi/HhNx6d73IGpJGwgEVCIaaWx9lY18JZo2CLhNFu+4FW/rB2Dw+/VMfztQ3Ut776HzlPdZGo20L79tW0vvg4HbVrgcOvw1a86Bwqz/kgeIq9v/oS5/3b+iHVFi6pYspVX+XtNzWy587P07ln45BeByAcidKVfOX/bcKFn6J4wRvY89PPEv3EmiG/7mu+T3El0z74Xc746M3sufPzOXvdbjNqZrJ92+jazufhl+p45OV9/J8LFwR2Zetpcybw4u5mnti4n/MXDX4V/iPV3qZ2frmqlrU7m2hqT9Ce6GLxjArOOW4yJ8+uIjrK1miSI89tj2/h+dpG/vvKEykeA/s2j/4KjwAnzCjnvtW72byvhaMmaUqyt1TKWburiT+s3cMf1uxm/e5mAI6eXMK5C6ZwwowKqsvTW0BddMmlXPCJ/2DXxBJ2Vh9N2bJLKC4IM3dyCfMmlzCtopBQJuQmu1LsbGxn1dZ6th1opbo8zrkLplB50W+GVW9ze4K7VtVS+IFv8M6lM/rdSmggnzzv2J6prxd2NPLg+r2cdlQVp95697Dq68vztQ08FCvm777/JEtnVub0tUfbtFdXyvnKfeuZWVXEe04Lri+qJB5hycwKntpSz5KZFaNy/aGRtGFvM//vgRf507q9dKWcOROLKS+MEgkZdzyxle8/upkpZTH+5S3zuXTJ9CFd9SujR83MWdRu35bvMl4jUjmN6vffTPvW53n7kguz+pp8/yKpEDYCjp5UQkkswjPbG8Z1COtMpth2oJXN+1rYWHeQp7fW89SWA9S3JggZLJtVxRcuOI5zF0xh1oTXzuO3vvQ4p2d2vO9Mpti8r4WX9zazZmcTz9c2EgkZRQVh4tEwB1o6SaacaNg4+9hJnDC9PCejkKXxKO9cOoNfrqrl7qdredvx1dRUDX0R0L3N7fz1pTpmVRVxyuxges2On17OtgOtPPryPiqLCpgzynskhuNXq2pZt6uJb1y1ZMhbFGXrpFmVrN7RxKMv7+PSpdPH5Sh3e6KLbz64ge89vJHCaJgPvG4OV54881V/x1o6kjzych3f+esmPv3L57j98S18+R2LOGFGRR4rl+HItn9yJKXcuWtVLQdaOvngNVdQ8g9XZ/V1+f5FUiFsBIRCxok1FTy6YR91zR1DHj0ZKxrbEqzb1cTLe5rZtK+FzZmP7Qda6b2D06wJRbz5uCmcdtQEzj520qCa0QsiIY6dWsqxU0vpTKbYsr+F3U3ttHZ20dbZxaJp5dRMKGRGRVHO/zEuK4xy2UkzWP7sTn797A7ecMzQQt6uxjZ++9wuCqNhzls4JbB/xM2Mtyycyi9X1XL/6t28a9kMJg6z8b8r5XQmU4RLJ7CjoY3JpbG8TzWt393EF5ev4ZTZVVxwfHXg3y8WCXP63Ak8uH4vT22tDyxEj1bPbKvnU794jk37Wrh0yXQ+d8Fxff69Ko5FOH9RNectmMry53bylfvW8Y5vP86Hzp7Lh8+ZF3hYlvHhmW0N7Gps57wFUygZA9OQ3cZOpWPcomllrNi8n2e21XPewiOjh8Td2dnYztqdTazZ2cjanU2s3dVEbf0rFyEUFYSZM7GY46eXc8niacyZVMyciSXMmVBMeVFu+nUKIiGOmVLKMVNGbnX48sIol588gwfW7OEvL9ax/UArr5s3KesepMKjT+Xup3dQHItwyYnTKCoI9lSMhkNcdEI1dz61nV8/s4Pz+hlt7E9nMsXW/S1sO9DK3uYO9h/spMudGf98O2d+9UHCIWNGZSHHTS3jzHkTef28iYN6/eFqaO3kujtWURqP8M13Lxmx6a5F08rYUd/GExv3M7k0xuwR/H/Ol0RXim88uIFvPbSBKaUxfnztqZw1b+KAXxcKGW9fMp03zp/Mjb9dy80PbuDP6/fy9csXM39q2QhULkeql/c28+iGfRw9qYT5U8fWLiEKYSMkFg2zoLqMF3Y0cupRE0Z8K5zhak90sWHvQdbvbmbdrqaewNV9dZgZzJlYzIk1Fbz71JksqC5j/tQyppTFjthpmlgkzIUnVLMqM636oye2csKMco6rLmNiSUGf/997m9tZuaWeSZd+ngkl6TWsgg5g3UrjUS5dMp17V+/mnmd3ctKsSk6ZXdXvSERzeyI9klnXQm19G13uxCIhJpfGOLGmgpJ4hLtvvp5bbrmFHfVtbN7fwrPbGrh/TXpvxXmTSzh/0VTesnAqC6eVBfb3oD3RxUfvfJZdjW3ced3p/a77FgQz403HTWZ/Swf3r97NFSfXUFlUMGLff6St3HKAz/36BV7ac5BLl07nXy9eSFmWe5t2Ky+M8vXLF3Pewil8/tcvcPE3HuMT5x7Dda8/irB6xWSQdtS38cCaPVSXxwOdUQiKQtgIWjqzknW7m/n1Mzt459LpWW/MPJKSXSm217fx4u4m1u9u5sXMx5b9LT1TifFoiPlTy7jghGoWVJexYFoZ86eWjliYGE1CZpw8u4rjqst4fOO+9HIa2xsoL4wypSxGcSxCNByiqS3BgZZO9jZ3UBAO0bTiV3zoM58Z8Sm8CSUxrjy5hodfqmPV1nqe3d7AzKoippXHMTNS7uw/2MnupvaegF1eGGVxTTlHTSyhujz+qlGmO557gKtOeWVJDXdn874WHn6pjgfW7OFbD23gGw9uYEZlIecvnMob50/mpFmVWa/dNZBNdQf50E+fYd2uJr5y6fGcNCu3Fx5kIxoOccHx1fz8qe384qntnLdw6oj33SVTKToSKTq7UrhDpKKaSEkVXa2N4MNfrDdSNZ3y095FyfFvJtm4lwN//C7/+bW/8Z/DfN1QYRlVb/kQX+tK8aUf3U/DX26jbdPKVx2T68bpzmSKjXUH2dXYxoGWBAfbE0QjIeKRMFUlBcyqKmJGZe7bGCT3tu5v4d7VuymLR7ho8bS8t0QMhbkf/vL+Yb242fnAfwNh4Pvu/tVDno8BdwAnAfuBK9x9y+Fec9myZb5y5crDHZITZhZI4+HuxnZ+/cwOigrCXHbSjMAvof3kW4577Q/hUIRI+WQildOIVlYTqagmWjmNSGU1kfIpWDhdk3uKZP0uOuu2kKjbSmLfVjrrtpCs35WTH+xDMdqaQQ/V2plkU10LG+oO0tCaoKUjSTLllMQiVBZHqaks4oTp5Xz2ggV5/3/Z2dDGy3sPsrHuIM3tr+zoUBwLM7UsTnV5IXMmFlNZFO33t8tPnncsh/sZcqClkz+t3cP9a3bz6Mv76OxKEY+GWDarihNrKjixpoJjppQyrSJOZBA/QPcf7ODup3fwX396iYJIiK9fvphz5k/J/n++D8M95xtaO/n9C7vYd7CTk2ZWsnRWRc5+MXF32hMpGto6aWxL0NiaoKEtkb7dlqC1n/XzjHRPVmk8QmVRAZXFUaqKCqgsKqCsMHrYkaf2RBfb61t5cXczG+taCIeMxTPKOXXOhJwGFHdnw96DPLZxP41tCaaVxzl+RjlzJ5UQDYcG/Dt2OE3tCdbvau5pl1izs4mX9zaT6Dr864VDxtGTSlg4Lf1L5sJp5SyYVjbmZjCCEtS/j9nqSjlPbNzPqm31VBUXcMniaZQN8b0Zzt+vbJnZKndf1tdzgSUAMwsD3wLOBWqBp8xsubuv7XXYtUC9ux9tZlcCXwOuCKqm0WBqeZxLTpzGPc/u4McrtrKguoxF08uHNYXRlXI6kl20J1KvfE500Z5MUX7GFZx1+T/R0tnFwY709kmH/sAuCIcoL4pSURiloihKRWEBE0oKqCouIBoePUsQ5PsqlmwUFURYNL2cRdPT+z66OylnVE6zTKsoZFpFIa+fN7HnHyUzBvfbpIWyHv63gkLiMxYSn7OEB2sX8cik2VgoPSLmXQmSDXtINOwiWb+TruYDdLU1keo4CKkUmBEuLCVSPpXo5DkUzlmKhSO0bXmO/ffexJu+ODJ7ex5ORVEBVyyr4S8v1bFqWz3PbK9n7qQS5kwspqo4HXyiYev583J3kikn0ZUi2eV0JFO0diZp6eyitTN9nh5sT9LYlqChNUFn16t/8SmJRagojDJnYjFl8SixSIhYJP1+/Pjf/xeXffIrtHZ00dyRoKktyZb9Lazd9cq5H7L0RSaVRQXEIiEiYQOHls4umtsT7D/YiQPxSIhTZlexuKY8kNFuM2PelFKOmlTC6h2NrNpWzwNr9lAQrmNmVRGlJ7+DFZv2M62ikMllsVdtxOye/nOrb+1kd2M7OxvaeXF3E2t3pdsmdvTat3dCcQELppXx+mOOYsG0MmoqC6kqLqAkFiGZctoTXext7mDr/lY27zvI2p1NPLphH3c/s6PnNWZUFrJwWhnHTCllSlmcKWVxSmIRCgvCFIRDdJ8KZmBYz22A7rPkldPFiEVCFBWEKY5Fet476V9Hoou1u9JXwze0JTh+ejmvmzdxTI6AdQtyGOYUYIO7bwIwszuBS4DeIewS4F8zt+8Cvmlm5kHH0jybVlHIZUtn8NSWep7Z3sDT2xqIRUKUF0YpKgj3nIjdp6OTnm5IdjnJLifRfbvnc/9/XBVnXc3GuhZKYhGKY2GmlKanyMoLo5RnQldhNKyTPyBmRniU/9GaGQWRIRbpqSH/RpzoSlHX3EF9aycNrQka2ippbJ1LQ1tnvyMV3cFh7sQSjqsuZcKb5sG1lw2t9kPkIuRHwiHefNwUltRUsHpnE+t3NfHy3lfvVhDJBPLDnbfdxxXHIlQURZlaHqeiMJr5ZamAsnjksCOHLasfZHEfS0B0JLqob01Q39qZ/mhJ0NDWyf5kqqee4oIIJbEIR00qYVZVEVPK4iPyS0Q4ZCyuqeCEGeXU1rexfncztfWtVJ1zLVfc8mTPcalEB6S6AMcisZ6R+26e6iJZv5POvZt7PhJ7N7H14AGeHkJdoaIKCqYcRcGUubRMOYrNk+cSqZyKWW7/4Q9nltgpLohQFEt/Ls58jmdCXjRsRMMhouEQBZH0/Ujolds9z4VDRCPp+5FQiHDICIe6fx4Z4ZARMiNk6e8bCqUfD5kRCqUfC2f2IX3ldiZcmhEunUhze6YfmFdS5qF/S3r/s2KZA3ofYz3/eeWYrlT637bOZIrWzi5aOpPsO9jJnsZ2dje1k0w5U8viXDRv4hGx5FOQIWw6sL3X/Vrg1P6OcfekmTUCE4Bgd0oeBSaXxbnghGpaOpK8tKeZ+tYETb2mFdzBMyvBG0YknP6IR8NEQ0YknP7NNRIyYpEwsUiIeDRMLJrubej+/PmLFnHTA+vy+b8q0qdoONQzGtebu5PoSo9MdCRTPedBLBKmNBYZEwt9TiiJ8YZjJvG6oyfSkOkHbGjrJJFM/xJlpANbNGxEQ+lzORYJU1TQ/REJpCcpFg0ztTzM1PLRu7ismVFTVdSz/t6n334qH/7Bgz0j+Ymkk/L034pIyCjI9HOVxNPhsbIoSiQ8P9AaUymnNdGVrqcrRaLL6cqEWMdfs4mHA3d8+RO9/yfTn8IRQgVFWDROqCBOfUEhoYJCLBrHet0OReMQjmChcDp0hqNYKIKFI68JoSNhxj/fxq2PbRmx7xey9BaAC6eVsaC6bEQvvglaYD1hZnYZcL67fyBz/z3Aqe7+4V7HrM4cU5u5vzFzzL5DXus64LrM3WOBkZiMnsg4CINjjN6T0Ufvyeik92X00XsyOo3E+zLL3Sf19USQEXoHUNPr/ozMY30dU2tmEaCcdIP+q7j7LcAtAdXZJzNb2V8jneSH3pPRR+/J6KT3ZfTRezI65ft9CbKb7SlgnpnNMbMC4Epg+SHHLAfel7l9GfDgkd4PJiIiIgIBjoRlerw+DDxAeomKW919jZndCKx09+XAD4AfmdkG4ADpoCYiIiJyxAu0o8/d7wXuPeSx63vdbgfeFWQNwzCi05+SFb0no4/ek9FJ78voo/dkdMrr+xLoYq0iIiIi0rexu8KZiIiIyBimEHYIMzvfzF40sw1m9pl81zMemVmNmT1kZmvNbI2ZfSzzeJWZ/dHMXs58HvmNAgUzC5vZM2b2u8z9OWa2InPO/DxzIY6MEDOrMLO7zGy9ma0zs9N1ruSfmX0i8/NrtZn9zMziOldGnpndamZ7M0tidT/W5/lhaTdn3p/nzWxp0PUphPXSa6ultwILgKvMbEF+qxqXksCn3H0BcBrwocz78Bngz+4+D/hz5r6MvI8BvVcA/hrwn+5+NFBPejsyGTn/Ddzv7vOBxaTfG50reWRm04GPAsvcfRHpi9O6t+bTuTKybgPOP+Sx/s6PtwLzMh/XAd8JujiFsFfr2WrJ3TuB7q2WZAS5+y53fzpzu5n0PyrTSb8Xt2cOux14e34qHL/MbAZwAfD9zH0DziG97RjofRlRZlYOvJ70lea4e6e7N6BzZTSIAIWZNTCLgF3oXBlx7v4w6dUXeuvv/LgEuMPTngQqzKw6yPoUwl6tr62WpuepFgHMbDawBFgBTHH3XZmndgNT8lTWePZfwP8CuneTngA0uHsyc1/nzMiaA9QBP8xMEX/fzIrRuZJX7r4D+A9gG+nw1QisQufKaNHf+THiGUAhTEYtMysBfgV83N2bej+XWdRXl/aOIDO7ENjr7qvyXYv0iABLge+4+xKghUOmHnWujLxMj9ElpEPyNKCY106JySiQ7/NDIezVstlqSUaAmUVJB7CfuPvdmYf3dA8NZz7vzVd949SZwMVmtoX0VP05pPuRKjJTLqBzZqTVArXuviJz/y7SoUznSn69Gdjs7nXungDuJn3+6FwZHfo7P0Y8AyiEvVo2Wy1JwDJ9Rj8A1rn7Tb2e6r3N1fuA34x0beOZu3/W3We4+2zS58aD7n418BDpbcdA78uIcvfdwHYzOzbz0JuAtehcybdtwGlmVpT5edb9vuhcGR36Oz+WA+/NXCV5GtDYa9oyEFqs9RBm9jbSfS/dWy19Oc8ljTtmdhbwCPACr/QefY50X9gvgJnAVuBydz+04VJGgJmdDXza3S80s6NIj4xVAc8Af+fuHfmsbzwxsxNJXyhRAGwC3k/6F2ydK3lkZjcAV5C+2vsZ4AOk+4t0rowgM/sZcDYwEdgDfBG4hz7Oj0xg/ibpqeNW4P3uvjLQ+hTCREREREaepiNFRERE8kAhTERERCQPFMJERERE8kAhTERERCQPFMJERERE8kAhTETGBTNzM/txr/sRM6szs99l7r/TzNaY2SNmNiHz2Fwz+3m+ahaRI5tCmIiMFy3AIjMrzNw/l1evhv0R4GTge8C7M4/9G/CFEatQRMYVhTARGU/uBS7I3L4K+Fmv51JADCgCEmb2OmC3u788siWKyHihECYi48mdwJVmFgdOIL0LQ7evAH8CLiIdzv4P8KURr1BExg2FMBEZN9z9eWA26VGwew957o/ufpK7XwRcknn+GDO7y8z+x8yKRrxgETmiKYSJyHizHPgPXj0V2SMTtq4BvgXcQHqD30eBq0eoPhEZJyL5LkBEZITdCjS4+wuZjcgP9S/Aze6eyDTxO+l+MY2EiUhOKYSJyLji7rXAzX09Z2bTgFPc/YbMQ98AngIagLePTIUiMl6Yu+e7BhEREZFxRz1hIiIiInmgECYiIiKSBwphIiIiInmgECYiIiKSBwphIiIiInmgECYiIiKSBwphIiIiInmgECYiIiKSB/8fwn/dhnCxHbsAAAAASUVORK5CYII=\n"
          },
          "metadata": {
            "needs_background": "light"
          }
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Ð‘Ð¾Ð½ÑƒÑ"
      ],
      "metadata": {
        "id": "Q4fcmmqYUqPA"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Ð£ÑÑ‚Ð°Ð½Ð¾Ð²ÐºÐ¸"
      ],
      "metadata": {
        "id": "GriiAfMLLA9h"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b6k9REsM6NXn",
        "outputId": "5b37a758-d069-41de-8626-c4726cfd810d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting pyGenomeTracks\n",
            "  Downloading pyGenomeTracks-3.8-py2.py3-none-any.whl (113 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m113.6/113.6 KB\u001b[0m \u001b[31m8.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pytest in /usr/local/lib/python3.8/dist-packages (from pyGenomeTracks) (3.6.4)\n",
            "Collecting future>=0.17.0\n",
            "  Downloading future-0.18.3.tar.gz (840 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m840.9/840.9 KB\u001b[0m \u001b[31m39.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting bx-python>=0.8.13\n",
            "  Downloading bx_python-0.9.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m4.8/4.8 MB\u001b[0m \u001b[31m64.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: matplotlib<=3.6.2,>=3.1.1 in /usr/local/lib/python3.8/dist-packages (from pyGenomeTracks) (3.2.2)\n",
            "Requirement already satisfied: tqdm>=4.20 in /usr/local/lib/python3.8/dist-packages (from pyGenomeTracks) (4.64.1)\n",
            "Collecting gffutils>=0.9\n",
            "  Downloading gffutils-0.11.1.tar.gz (1.6 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m52.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting hicmatrix>=15\n",
            "  Downloading HiCMatrix-15-py3-none-any.whl (37 kB)\n",
            "Collecting pysam>=0.14\n",
            "  Downloading pysam-0.20.0-cp38-cp38-manylinux_2_24_x86_64.whl (16.1 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m16.1/16.1 MB\u001b[0m \u001b[31m65.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting pybedtools>=0.8.1\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m71.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: intervaltree>=2.1.0 in /usr/local/lib/python3.8/dist-packages (from pyGenomeTracks) (2.1.0)\n",
            "Requirement already satisfied: numpy>=1.20 in /usr/local/lib/python3.8/dist-packages (from pyGenomeTracks) (1.21.6)\n",
            "Collecting pyBigWig>=0.3.16\n",
            "  Downloading pyBigWig-0.3.18.tar.gz (64 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m64.8/64.8 KB\u001b[0m \u001b[31m7.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pyfaidx>=0.1.3\n",
            "  Downloading pyfaidx-0.7.1.tar.gz (103 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m103.2/103.2 KB\u001b[0m \u001b[31m6.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.8/dist-packages (from gffutils>=0.9->pyGenomeTracks) (1.15.0)\n",
            "Collecting argh>=0.26.2\n",
            "  Downloading argh-0.26.2-py2.py3-none-any.whl (30 kB)\n",
            "Collecting argcomplete>=1.9.4\n",
            "  Downloading argcomplete-2.0.0-py2.py3-none-any.whl (37 kB)\n",
            "Collecting simplejson\n",
            "  Downloading simplejson-3.18.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (135 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m135.5/135.5 KB\u001b[0m \u001b[31m14.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: tables>=3.5.* in /usr/local/lib/python3.8/dist-packages (from hicmatrix>=15->pyGenomeTracks) (3.7.0)\n",
            "Collecting intervaltree>=2.1.0\n",
            "  Downloading intervaltree-3.1.0.tar.gz (32 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: pandas>=0.25.* in /usr/local/lib/python3.8/dist-packages (from hicmatrix>=15->pyGenomeTracks) (1.3.5)\n",
            "Requirement already satisfied: scipy>=1.2.* in /usr/local/lib/python3.8/dist-packages (from hicmatrix>=15->pyGenomeTracks) (1.7.3)\n",
            "Collecting cooler>=0.8.9\n",
            "  Downloading cooler-0.9.1-py2.py3-none-any.whl (103 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m103.9/103.9 KB\u001b[0m \u001b[31m12.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: sortedcontainers<3.0,>=2.0 in /usr/local/lib/python3.8/dist-packages (from intervaltree>=2.1.0->pyGenomeTracks) (2.4.0)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.8/dist-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (2.8.2)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.8/dist-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (1.4.4)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.8/dist-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (0.11.0)\n",
            "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/local/lib/python3.8/dist-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (3.0.9)\n",
            "Requirement already satisfied: setuptools>=0.7 in /usr/local/lib/python3.8/dist-packages (from pyfaidx>=0.1.3->pyGenomeTracks) (57.4.0)\n",
            "Requirement already satisfied: pluggy<0.8,>=0.5 in /usr/local/lib/python3.8/dist-packages (from pytest->pyGenomeTracks) (0.7.1)\n",
            "Requirement already satisfied: py>=1.5.0 in /usr/local/lib/python3.8/dist-packages (from pytest->pyGenomeTracks) (1.11.0)\n",
            "Requirement already satisfied: more-itertools>=4.0.0 in /usr/local/lib/python3.8/dist-packages (from pytest->pyGenomeTracks) (9.0.0)\n",
            "Requirement already satisfied: atomicwrites>=1.0 in /usr/local/lib/python3.8/dist-packages (from pytest->pyGenomeTracks) (1.4.1)\n",
            "Requirement already satisfied: attrs>=17.4.0 in /usr/local/lib/python3.8/dist-packages (from pytest->pyGenomeTracks) (22.2.0)\n",
            "Collecting cytoolz\n",
            "  Downloading cytoolz-0.12.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.8 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m1.8/1.8 MB\u001b[0m \u001b[31m56.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting multiprocess\n",
            "  Downloading multiprocess-0.70.14-py38-none-any.whl (132 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m132.0/132.0 KB\u001b[0m \u001b[31m7.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pyyaml in /usr/local/lib/python3.8/dist-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (6.0)\n",
            "Requirement already satisfied: click>=7 in /usr/local/lib/python3.8/dist-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (7.1.2)\n",
            "Collecting asciitree\n",
            "  Downloading asciitree-0.3.3.tar.gz (4.0 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: h5py>=2.5 in /usr/local/lib/python3.8/dist-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (3.1.0)\n",
            "Requirement already satisfied: pytz>=2017.3 in /usr/local/lib/python3.8/dist-packages (from pandas>=0.25.*->hicmatrix>=15->pyGenomeTracks) (2022.7.1)\n",
            "Requirement already satisfied: numexpr>=2.6.2 in /usr/local/lib/python3.8/dist-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (2.8.4)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.8/dist-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (23.0)\n",
            "Requirement already satisfied: toolz>=0.8.0 in /usr/local/lib/python3.8/dist-packages (from cytoolz->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.12.0)\n",
            "Requirement already satisfied: dill>=0.3.6 in /usr/local/lib/python3.8/dist-packages (from multiprocess->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.6)\n",
            "Building wheels for collected packages: future, gffutils, intervaltree, pybedtools, pyBigWig, pyfaidx, asciitree\n",
            "  Building wheel for future (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for future: filename=future-0.18.3-py3-none-any.whl size=492036 sha256=c699bd029393480bb4c6253ea4ee2414c1fe28c11dc72fd112a3f119e3dfa972\n",
            "  Stored in directory: /root/.cache/pip/wheels/a0/0b/ee/e6994fadb42c1354dcccb139b0bf2795271bddfe6253ccdf11\n",
            "  Building wheel for gffutils (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for gffutils: filename=gffutils-0.11.1-py3-none-any.whl size=1619366 sha256=56b11f1d16e8bbc747191b4da5d0795243c0bdcb6e8803fd983db2dcf9e16de2\n",
            "  Stored in directory: /root/.cache/pip/wheels/fd/d2/4f/a441971b5d0e5fdb43cc6c4c9305c172e418fae6da18309ee0\n",
            "  Building wheel for intervaltree (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervaltree: filename=intervaltree-3.1.0-py2.py3-none-any.whl size=26118 sha256=4818516ec90a71438bf374e7676123aec779fa6301c9703b99083bc04868ef2a\n",
            "  Stored in directory: /root/.cache/pip/wheels/45/23/de/5789a92962483fd33cb06674792b9697c1b3766d7c7742830e\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp38-cp38-linux_x86_64.whl size=14295447 sha256=80678133dba36fde887da23593ee1086f9edd5e7dd3411c2882c5bc03cdc2c3b\n",
            "  Stored in directory: /root/.cache/pip/wheels/7b/10/cc/219c98e11fc3cbf65aa83bfe028d72b24043d6c921a3c19b50\n",
            "  Building wheel for pyBigWig (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pyBigWig: filename=pyBigWig-0.3.18-cp38-cp38-linux_x86_64.whl size=217653 sha256=8723957aaf2d59d7fc04ce59802e976e8691403aeae0b903f95bdcc496fd5cd1\n",
            "  Stored in directory: /root/.cache/pip/wheels/1f/d0/64/fa4977556a0cfae19b8b1cf546f1928e81c45a8f19f73aba9b\n",
            "  Building wheel for pyfaidx (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pyfaidx: filename=pyfaidx-0.7.1-py3-none-any.whl size=27748 sha256=1a419bdaed2b5717090b61bcaa77d3ef7a0acbc838fb67046cbe37a49b6f3a69\n",
            "  Stored in directory: /root/.cache/pip/wheels/0e/6c/f4/2560f2678a35d8b1997bb693336743308ef32b78bc9ce9aaf4\n",
            "  Building wheel for asciitree (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for asciitree: filename=asciitree-0.3.3-py3-none-any.whl size=5050 sha256=c68c852ecae4892cd38da8eada6677306afbdc3f603399007a7c4f025b297e6e\n",
            "  Stored in directory: /root/.cache/pip/wheels/a3/d7/75/19cd0d2a893cad4bb0b2b16dd572ad2916d19c0d5ee9612511\n",
            "Successfully built future gffutils intervaltree pybedtools pyBigWig pyfaidx asciitree\n",
            "Installing collected packages: pysam, pyBigWig, asciitree, argh, simplejson, pyfaidx, pybedtools, multiprocess, intervaltree, future, cytoolz, bx-python, argcomplete, gffutils, cooler, hicmatrix, pyGenomeTracks\n",
            "  Attempting uninstall: intervaltree\n",
            "    Found existing installation: intervaltree 2.1.0\n",
            "    Uninstalling intervaltree-2.1.0:\n",
            "      Successfully uninstalled intervaltree-2.1.0\n",
            "  Attempting uninstall: future\n",
            "    Found existing installation: future 0.16.0\n",
            "    Uninstalling future-0.16.0:\n",
            "      Successfully uninstalled future-0.16.0\n",
            "Successfully installed argcomplete-2.0.0 argh-0.26.2 asciitree-0.3.3 bx-python-0.9.0 cooler-0.9.1 cytoolz-0.12.1 future-0.18.3 gffutils-0.11.1 hicmatrix-15 intervaltree-3.1.0 multiprocess-0.70.14 pyBigWig-0.3.18 pyGenomeTracks-3.8 pybedtools-0.9.0 pyfaidx-0.7.1 pysam-0.20.0 simplejson-3.18.3\n"
          ]
        }
      ],
      "source": [
        "! pip install pyGenomeTracks"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fcsJWNmHo1i-",
        "outputId": "7b7e7580-0c4b-457f-aeef-6e53e5153aca"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: ipykernel in /usr/local/lib/python3.8/dist-packages (5.3.4)\n",
            "Collecting ipykernel\n",
            "  Downloading ipykernel-6.21.1-py3-none-any.whl (149 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m149.8/149.8 KB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting ipython>=7.23.1\n",
            "  Downloading ipython-8.9.0-py3-none-any.whl (783 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m783.1/783.1 KB\u001b[0m \u001b[31m35.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting comm>=0.1.1\n",
            "  Downloading comm-0.1.2-py3-none-any.whl (6.5 kB)\n",
            "Requirement already satisfied: jupyter-client>=6.1.12 in /usr/local/lib/python3.8/dist-packages (from ipykernel) (6.1.12)\n",
            "Collecting nest-asyncio\n",
            "  Downloading nest_asyncio-1.5.6-py3-none-any.whl (5.2 kB)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.8/dist-packages (from ipykernel) (5.4.8)\n",
            "Collecting tornado>=6.1\n",
            "  Downloading tornado-6.2-cp37-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (423 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m424.0/424.0 KB\u001b[0m \u001b[31m33.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pyzmq>=17 in /usr/local/lib/python3.8/dist-packages (from ipykernel) (23.2.1)\n",
            "Requirement already satisfied: jupyter-core!=5.0.*,>=4.12 in /usr/local/lib/python3.8/dist-packages (from ipykernel) (5.2.0)\n",
            "Requirement already satisfied: traitlets>=5.4.0 in /usr/local/lib/python3.8/dist-packages (from ipykernel) (5.7.1)\n",
            "Collecting debugpy>=1.6.5\n",
            "  Downloading debugpy-1.6.6-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m62.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting matplotlib-inline>=0.1\n",
            "  Downloading matplotlib_inline-0.1.6-py3-none-any.whl (9.4 kB)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.8/dist-packages (from ipykernel) (23.0)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.8/dist-packages (from ipython>=7.23.1->ipykernel) (0.2.0)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.8/dist-packages (from ipython>=7.23.1->ipykernel) (4.4.2)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.8/dist-packages (from ipython>=7.23.1->ipykernel) (0.7.5)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.8/dist-packages (from ipython>=7.23.1->ipykernel) (4.8.0)\n",
            "Collecting jedi>=0.16\n",
            "  Downloading jedi-0.18.2-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m46.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting prompt-toolkit<3.1.0,>=3.0.30\n",
            "  Downloading prompt_toolkit-3.0.36-py3-none-any.whl (386 kB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m386.4/386.4 KB\u001b[0m \u001b[31m25.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pygments>=2.4.0 in /usr/local/lib/python3.8/dist-packages (from ipython>=7.23.1->ipykernel) (2.6.1)\n",
            "Collecting stack-data\n",
            "  Downloading stack_data-0.6.2-py3-none-any.whl (24 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.8/dist-packages (from jupyter-client>=6.1.12->ipykernel) (2.8.2)\n",
            "Requirement already satisfied: platformdirs>=2.5 in /usr/local/lib/python3.8/dist-packages (from jupyter-core!=5.0.*,>=4.12->ipykernel) (2.6.2)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.8/dist-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel) (0.8.3)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.8/dist-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.8/dist-packages (from prompt-toolkit<3.1.0,>=3.0.30->ipython>=7.23.1->ipykernel) (0.2.6)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.1->jupyter-client>=6.1.12->ipykernel) (1.15.0)\n",
            "Collecting executing>=1.2.0\n",
            "  Downloading executing-1.2.0-py2.py3-none-any.whl (24 kB)\n",
            "Collecting pure-eval\n",
            "  Downloading pure_eval-0.2.2-py3-none-any.whl (11 kB)\n",
            "Collecting asttokens>=2.1.0\n",
            "  Downloading asttokens-2.2.1-py2.py3-none-any.whl (26 kB)\n",
            "Installing collected packages: pure-eval, executing, tornado, prompt-toolkit, nest-asyncio, matplotlib-inline, jedi, debugpy, comm, asttokens, stack-data, ipython, ipykernel\n",
            "  Attempting uninstall: tornado\n",
            "    Found existing installation: tornado 6.0.4\n",
            "    Uninstalling tornado-6.0.4:\n",
            "      Successfully uninstalled tornado-6.0.4\n",
            "  Attempting uninstall: prompt-toolkit\n",
            "    Found existing installation: prompt-toolkit 2.0.10\n",
            "    Uninstalling prompt-toolkit-2.0.10:\n",
            "      Successfully uninstalled prompt-toolkit-2.0.10\n",
            "  Attempting uninstall: debugpy\n",
            "    Found existing installation: debugpy 1.0.0\n",
            "    Uninstalling debugpy-1.0.0:\n",
            "      Successfully uninstalled debugpy-1.0.0\n",
            "  Attempting uninstall: ipython\n",
            "    Found existing installation: ipython 7.9.0\n",
            "    Uninstalling ipython-7.9.0:\n",
            "      Successfully uninstalled ipython-7.9.0\n",
            "  Attempting uninstall: ipykernel\n",
            "    Found existing installation: ipykernel 5.3.4\n",
            "    Uninstalling ipykernel-5.3.4:\n",
            "      Successfully uninstalled ipykernel-5.3.4\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "google-colab 1.0.0 requires ipykernel~=5.3.4, but you have ipykernel 6.21.1 which is incompatible.\n",
            "google-colab 1.0.0 requires ipython~=7.9.0, but you have ipython 8.9.0 which is incompatible.\n",
            "google-colab 1.0.0 requires tornado~=6.0.4, but you have tornado 6.2 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed asttokens-2.2.1 comm-0.1.2 debugpy-1.6.6 executing-1.2.0 ipykernel-6.21.1 ipython-8.9.0 jedi-0.18.2 matplotlib-inline-0.1.6 nest-asyncio-1.5.6 prompt-toolkit-3.0.36 pure-eval-0.2.2 stack-data-0.6.2 tornado-6.2\n"
          ]
        }
      ],
      "source": [
        "!pip install -U ipykernel"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install conda=22.9.0"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MWPew6u4Jctv",
        "outputId": "1527887d-05f0-4f4b-c4cd-5ff3b8ce5344"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - conda=22.9.0\n",
            "\n",
            "\n",
            "The following packages will be DOWNGRADED:\n",
            "\n",
            "  conda                              22.11.1-py38h578d9bd_1 --> 22.9.0-py38h578d9bd_2 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\bdone\n",
            "Executing transaction: | \b\b/ \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UhIKyKz_t7yy",
        "outputId": "47c5a491-11d2-47a0-e618-c0a96234dcf2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bfailed with initial frozen solve. Retrying with flexible solve.\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bfailed with repodata from current_repodata.json, will retry with next repodata source.\n",
            "Collecting package metadata (repodata.json): \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-fetchchromsizes\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    libgcc-7.2.0               |       h69d50b8_2         304 KB  conda-forge\n",
            "    libpng-1.6.39              |       h753d276_0         276 KB  conda-forge\n",
            "    mysql-connector-c-6.1.6    |                2         9.6 MB  bioconda\n",
            "    ucsc-fetchchromsizes-357   |       h5eb252a_2           5 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        10.2 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  libgcc             conda-forge/linux-64::libgcc-7.2.0-h69d50b8_2 None\n",
            "  libpng             conda-forge/linux-64::libpng-1.6.39-h753d276_0 None\n",
            "  mysql-connector-c  bioconda/linux-64::mysql-connector-c-6.1.6-2 None\n",
            "  ucsc-fetchchromsi~ bioconda/linux-64::ucsc-fetchchromsizes-357-h5eb252a_2 None\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  conda                               22.9.0-py38h578d9bd_2 --> 22.11.1-py38h578d9bd_1 None\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00,  6.20it/s]\n",
            "ucsc-fetchchromsizes | 5 KB      | : 100% 1.0/1 [00:00<00:00, 23.21it/s]\n",
            "libpng-1.6.39        | 276 KB    | : 100% 1.0/1 [00:00<00:00, 19.77it/s]\n",
            "mysql-connector-c-6. | 9.6 MB    | : 100% 1.0/1 [00:02<00:00,  2.04s/it]               \n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\bdone\n",
            "Retrieving notices: ...working... done\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda ucsc-fetchchromsizes"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! apt-get install bedtools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GnH3Ks2TJDhS",
        "outputId": "f9851680-6ef5-4a56-d9ae-0640fe0ef7a6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "bedtools is already the newest version (2.27.1+dfsg-4ubuntu1).\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'apt autoremove' to remove it.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "svn9iAuGtVxX",
        "outputId": "45c19f17-9e21-4232-e51d-42f26b154e00"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-bedgraphtobigwig\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ucsc-bedgraphtobigwig-357  |                0         972 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         972 KB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  ucsc-bedgraphtobi~ bioconda/linux-64::ucsc-bedgraphtobigwig-357-0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "                                                                        \n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda ucsc-bedgraphtobigwig"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb\n",
        "!sudo apt-get install /content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "42fVGJlZLudQ",
        "outputId": "ece4a54d-9650-4567-b006-120223acdfa7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-02-08 13:20:06--  http://archive.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb\n",
            "Resolving archive.ubuntu.com (archive.ubuntu.com)... 91.189.91.39, 185.125.190.36, 185.125.190.39, ...\n",
            "Connecting to archive.ubuntu.com (archive.ubuntu.com)|91.189.91.39|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1088952 (1.0M) [application/x-debian-package]\n",
            "Saving to: â€˜libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.debâ€™\n",
            "\n",
            "\r          libssl1.0   0%[                    ]       0  --.-KB/s               \rlibssl1.0.0_1.0.2n- 100%[===================>]   1.04M  --.-KB/s    in 0.09s   \n",
            "\n",
            "2023-02-08 13:20:06 (12.1 MB/s) - â€˜libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.debâ€™ saved [1088952/1088952]\n",
            "\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "Note, selecting 'libssl1.0.0' instead of '/content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb'\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-510\n",
            "Use 'sudo apt autoremove' to remove it.\n",
            "The following NEW packages will be installed:\n",
            "  libssl1.0.0\n",
            "0 upgraded, 1 newly installed, 0 to remove and 28 not upgraded.\n",
            "Need to get 0 B/1,089 kB of archives.\n",
            "After this operation, 3,501 kB of additional disk space will be used.\n",
            "Get:1 /content/libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb libssl1.0.0 amd64 1.0.2n-1ubuntu5.10 [1,089 kB]\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libssl1.0.0:amd64.\n",
            "(Reading database ... 129563 files and directories currently installed.)\n",
            "Preparing to unpack .../libssl1.0.0_1.0.2n-1ubuntu5.10_amd64.deb ...\n",
            "Unpacking libssl1.0.0:amd64 (1.0.2n-1ubuntu5.10) ...\n",
            "Setting up libssl1.0.0:amd64 (1.0.2n-1ubuntu5.10) ...\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)\n",
            "debconf: falling back to frontend: Readline\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Ð—Ð°Ð´Ð°Ð½Ð¸Ðµ\n"
      ],
      "metadata": {
        "id": "0ki8QRk9LLw7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! fetchChromSizes mm10 > m.chrom.sizes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SKi-AaQyLLaT",
        "outputId": "dc0a246e-d6b0-49e0-d3e3-e39fb36fcdf9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO: trying WGET /usr/bin/wget for database mm10\n",
            "url: http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig s_SRR3824222_1_bismark_bt2_pe.deduplicated.bedGraph m.chrom.sizes epiblast.bigWig.bw\n",
        "! bedGraphToBigWig s_SRR5836475_1_bismark_bt2_pe.deduplicated.bedGraph m.chrom.sizes icm.bigWig.bw\n",
        "! bedGraphToBigWig s_SRR5836473_1_bismark_bt2_pe.deduplicated.bedGraph m.chrom.sizes _8cell.bigWig.bw"
      ],
      "metadata": {
        "id": "j3dEA3TMLZI3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedtools genomecov -bg -ibam s_SRR3824222_1_bismark_bt2_pe.deduplicated.bam -g m.chromSizes  >  epiblast.cover.bedGraph\n",
        "! bedtools genomecov -bg -ibam s_SRR5836475_1_bismark_bt2_pe.deduplicated.bam -g m.chromSizes  >  icm.cover.bedGraph\n",
        "! bedtools genomecov -bg -ibam s_SRR5836473_1_bismark_bt2_pe.deduplicated.bam -g m.chromSizes  >  _8cell.cover.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JZXfC0TkLZGB",
        "outputId": "1b65489b-b4ab-4d1f-8a97-3efa39a2e2d6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n",
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n",
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig epiblast.cover.bedGraph m.chrom.sizes epiblast_coverage.bigWig.bw\n",
        "! bedGraphToBigWig icm.cover.bedGraph m.chrom.sizes icm_coverage.bigWig.bw\n",
        "! bedGraphToBigWig _8cell.cover.bedGraph m.chrom.sizes _8cell_coverage.bigWig.bw"
      ],
      "metadata": {
        "id": "9ZA5L2b0LZC4"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!pyGenomeTracks --tracks coverage.ini.txt --region chr11:3000000-4000000 -out cov.png"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5qCunKKmO6J9",
        "outputId": "3038d4f6-4e83-4204-e43a-8cd2f3558819"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "WARNING:pygenometracks.tracksClass:In section 1. [bigwig file test], these parameters are unused:['bins'].\n",
            "\n",
            "WARNING:pygenometracks.tracksClass:In section 2. [bigwig file test], these parameters are unused:['bins'].\n",
            "\n",
            "WARNING:pygenometracks.tracksClass:In section 3. [bigwig file test], these parameters are unused:['bins'].\n",
            "\n",
            "INFO:pygenometracks.tracksClass:initialize 1. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.00436091423034668\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 14.760638297872342. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [x-axis]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pyGenomeTracks --tracks meth.ini.txt --region chr11:3000000-4000000 -out meth.png"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c4atCbxxP1Mo",
        "outputId": "6a86d39b-2fae-4e98-9798-889e8ab543b4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "WARNING:pygenometracks.tracksClass:In section 1. [bigwig file test], these parameters are unused:['bins'].\n",
            "\n",
            "WARNING:pygenometracks.tracksClass:In section 2. [bigwig file test], these parameters are unused:['bins'].\n",
            "\n",
            "WARNING:pygenometracks.tracksClass:In section 3. [bigwig file test], these parameters are unused:['bins'].\n",
            "\n",
            "INFO:pygenometracks.tracksClass:initialize 1. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.00321197509765625\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 14.760638297872342. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [x-axis]\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}